---
title: "zoops_mb_analysis"
output: html_document
date: "2023-02-13"
---
```{r loading packages}

install.packages("ggcorrplot")
install.packages("FactoMineR")
install_github("kassambara/factoextra")
remotes::install_github("microbiome/microbiome")
devtools::install_github("microbiome/mia")
BiocManager::install("DESeq2")
install.packages("corrr")
install.packages("patchwork")
install.packages("picante")


library(tidyverse)
library(ggplot2)
library(phyloseq)
library(vegan)
library(cowplot)
library(gridExtra)
library(ggpubr)
library(dplyr)
library(doBy)
library(RColorBrewer)
library(glue)
library(ecodist)
library(geosphere)
library(data.table)
library(patchwork)
library(ggpubr)
library(car)
library(ggcorrplot)
library(FactoMineR)
library(devtools)
library(factoextra)
library(knitr)
library(magrittr)
library(BiocManager)
library(decontam)
library(plyr)
library(multcompView)
library(lattice)
library(Hmisc)
library(sciplot)
library(reshape)
library(doBy)
library(car)
library(readr)
library(corrr)
library(picante)
library(factoextra)
library(microbiome)
library(mia)
library(MASS)


```


```{r loading in data and modifying metadata}

# load in CSV files for phyloseq object
metadata <- read.csv('phyloseq analysis/metadata_mp.csv')
ASV_table <- read.csv('phyloseq analysis/otu_table.csv')
tax_table <- read.csv('phyloseq analysis/tax_table.csv')  

# modify metadata
str(metadata)
table(metadata$organism)

# set row names
row.names(metadata) <- metadata$X

metadata$date <- as.Date(metadata$date)

# factors
metadata$sample_ID <- as.factor(metadata$sample_ID)
metadata$sequencing_ID <- as.factor(metadata$sequencing_ID)
metadata$sample_type <- as.factor(metadata$sample_type)
metadata$organism_water <- as.factor(metadata$organism_water)
metadata$time_point <- as.factor(metadata$time_point)
metadata$date <- as.factor(metadata$date)
metadata$lake <- as.factor(metadata$lake)

# numerics
metadata$number <- as.numeric(metadata$number)
metadata$latitude <- as.numeric(metadata$latitude)
metadata$longitude <- as.numeric(metadata$longitude)
metadata$elevation_m <- as.numeric(metadata$elevation_m)
metadata$temp_C <- as.numeric(metadata$temp_C)
metadata$chla_ug.L <- as.numeric(metadata$chla_ug.L)
metadata$pH <- as.numeric(metadata$pH)
metadata$DO_perc <- as.numeric(metadata$DO_perc)
metadata$spc <- as.numeric(metadata$spc)
metadata$cond <- as.numeric(metadata$cond)
metadata$DOC <- as.numeric(metadata$DOC)
metadata$TDN <- as.numeric(metadata$TDN)
metadata$TDP <- as.numeric(metadata$TDP)

#check
class(metadata$sample_ID)
class(metadata$elevation_m)

# make new column for cladocerans, copepoda, water
metadata$phy_group <- ifelse(metadata$organism == "Calanoid" | metadata$organism == "Cyclopoid" | metadata$organism =="Large Calanoid", "copepoda", ifelse(metadata$organism == "Water", "water", "cladocera"))

#make new column for elevation category
metadata$elev.cat <- as.factor(ifelse(metadata$elevation_m < 2900, "Below", "Above"))

# make a new column for basin category
metadata$basin <- as.factor(ifelse(metadata$lake == "Eastern Brook" | metadata$lake == "Serene", "South", ifelse(metadata$lake == "Cooney" | metadata$lake == "Blue" | metadata$lake == "Virginia", "North", "Middle")))

#check that levels are right
levels(as.factor(metadata$phy_group))
levels(as.factor(metadata$elev.cat))

metadata$phy_group <- as.factor(metadata$phy_group)
metadata$elev.cat <- as.factor(metadata$elev.cat)


```


```{r making phyloseq objects}

PS.fin<-
  read_phyloseq(
    otu.file = "phyloseq analysis/otu_table.csv",
    taxonomy.file = "phyloseq analysis/tax_table.csv",
    metadata.file = "phyloseq analysis/metadata_mp.csv",
    type = c("simple"),
    sep = ","
  ) 
str(PS.fin)

nsamples(PS.fin)
ntaxa(PS.fin)
head(taxa_sums(PS.fin))

# make sure to remove all mitochondria and chloroplast taxonomic IDs
PS.fin <- subset_taxa(PS.fin, Family!= "Mitochondria" | 
                                is.na(Family) & Class!="Chloroplast" | is.na(Class))

PS.zoops <- subset_samples(PS.fin, sample_type == "Zooplankton")
sample_data(PS.zoops)$time_point <- as.factor(sample_data(PS.zoops)$time_point)
sample_data(PS.zoops)$phy_group <- ifelse(sample_data(PS.zoops)$organism_water == "Calanoid" | sample_data(PS.zoops)$organism_water == "Cyclopoid" | sample_data(PS.zoops)$organism_water =="Large Calanoid", "copepoda", "cladocera")
PS.water <- subset_samples(PS.fin, sample_type == "Water")
sample_data(PS.water)$time_point <- as.factor(sample_data(PS.water)$time_point)


# change sample names to match 
sample_names(PS.fin) <- metadata$sequence_ID
rownames(metadata) <- metadata$sequence_ID
sample_data(PS.fin) <- metadata


########### let's inspect
df.fin <- as.data.frame(sample_data(PS.fin))
df.fin$LibrarySize <- sample_sums(PS.fin) # this is the # of reads
df.fin$Index <- seq(nrow(df.fin))

########### now we have final phyloseq object with correct metadata 
saveRDS(PS.fin, file="phyloseq analysis/output/PS.fin.rds")
PS.fin <- readRDS("phyloseq analysis/output/PS.fin.rds")

# Make a data frame with a column for the read counts of each sample
sample_data(PS.fin)$read_depth <- sample_sums(PS.fin)
sample_sum_df <- data.frame(read.sum = sample_sums(PS.fin))
View(sample_sum_df)
median(sample_sum_df$read.sum)

# make row names the sample names
colnames(sample_sum_df) <- "read.sum"
sample_sum_df$sequence_ID <- rownames(sample_sum_df)

# merge in the reads
run.metaD <- merge(metadata, sample_sum_df, by="sequence_ID", all.y=TRUE)
write.csv(run.metaD, "phyloseq analysis/output/run.metaD.final.csv")
```


```{r taxa compositions}

# finding average abundance of each class of bacteria for zoops overall
glom <- tax_glom(PS.zoops, taxrank = 'Class')
CGroup <- transform_sample_counts(glom, function(x)100* x / sum(x))
OTUg <- otu_table(CGroup)
TAXg <- tax_table(CGroup)[,"Class"]
AverageD <- as.data.frame(rowMeans(OTUg))
names(AverageD) <- c("Mean")
GTable <- merge(TAXg, AverageD, by=0, all=TRUE)
GTable$Row.names = NULL
GTable <- GTable[order(desc(GTable$Mean)),]
head(GTable, 18)




```


```{r total reads by species boxplots}

# Box plots showing spread of total reads by species

reads <- ggplot(sample_data(PS.fin), aes(x=organism_water, y=read_depth)) + geom_boxplot() +
  theme(axis.text.x = element_text(angle = 90)) + labs(y= "Read Depth", x = "Organism")
reads
dev.copy(pdf, "phyloseq analysis/figures/reads.sample.pdf", height=4, width=7)
dev.off() 

log.reads <- ggplot(sample_data(PS.fin), aes(x=organism_water, y=log(read_depth))) + geom_boxplot() +
  theme(axis.text.x = element_text(angle = 90)) + labs(y= "log(Read Depth)", x = "Organism")
log.reads
dev.copy(pdf, "phyloseq analysis/figures/log.reads.sample.pdf", height=4, width=7)
dev.off() 

```


```{r Read depth plots and rarefaction curves}

readcount = data.table(as(sample_data(PS.fin), "data.frame"),
                       TotalReads = sample_sums(PS.fin),
                       keep.rownames = T)
ggplot(readcount, aes(TotalReads)) + geom_histogram() + ggtitle("Sequencing Depth")


# Histogram of sample read counts
hist.depth <- ggplot(sample_sum_df, aes(x = read.sum)) + 
  geom_histogram(color = "black", fill = "gold2", binwidth = 2500) +
  ggtitle("Distribution of sample sequencing depth") + 
  xlab("Read counts") +
  theme(axis.title.y = element_blank()) + theme_classic() + geom_vline(xintercept=1000, lty=2)
hist.depth
dev.copy(pdf, "phyloseq analysis/figures/hist.depth.pdf", height=4, width=5)
dev.off() 

# a different view
read_depth_violin <- ggplot(sample_sum_df, aes(x=1, y=read.sum)) +
  geom_violin() + ggtitle("Distribution of sample sequencing depth") + scale_y_log10() +
  xlab("Samples") + ylab("Read Depth")

read_depth_violin <- read_depth_violin + geom_boxplot(width=0.1)
read_depth_violin
dev.copy(pdf, "phyloseq analysis/figures/read_depth_violin.pdf", height=4, width=5)
dev.off()

# Rarefaction curve
count_table_filt <- as.data.frame(otu_table(PS.fin))
rarecurve((count_table_filt), step=50, cex=0.5, xlim=c(0,100000), ylab = "ASVs", label=FALSE)
abline(v = 5000, lty = "dotted", col="red", lwd=2)

dev.copy(pdf, "phyloseq analysis/figures/rare.raw.pdf", height=4, width=5)
dev.off() 
```


```{r exploring what low read count samples do and what to rarefy to}

ord <- ordinate(PS.fin, method = 'PCoA', distance = 'bray')
plot_ordination(PS.fin, ord, color = "sample_type")
sample_data(PS.fin)$read_depth <- sample_sums(PS.fin)
plot_ordination(PS.fin, ord, shape = "sample_type") +
  geom_point(aes(color = read_depth > 500))

ord <- ordinate(PS.rare, method = 'PCoA', distance = 'bray')
plot_ordination(PS.rare, ord, shape = "sample_type") +
  geom_point(aes(color = read_depth))
sample_data(PS.rare)$read_depth <- sample_sums(PS.rare)

ord <- ordinate(PS.rare, method = 'PCoA', distance = 'bray')
plot_ordination(PS.rare, ord, color = "time_point") +
  geom_point() +
  stat_ellipse(level=0.9, linetype = 2)
```


```{r rarefy}

table(sample_data(PS.fin)$sample_type)

# rarefying to 500 reads (both the zoops and water)
set.seed(10000)
PS.500 <- rarefy_even_depth(PS.fin, rngseed=1e4, sample.size = 500)
sort(rowSums(otu_table(t(PS.500))))

# filtering metadata by sample names with 500 reads or more, 49 samples lost
metadata_500 <- as.data.frame(sample_data(PS.500))
metadata_500$sequence_ID <- rownames(PS.500) 
table(sample_data(PS.500)$sample_type)

# rarefying to 500 reads for just zoops
set.seed(10000)
PS.zoops <- rarefy_even_depth(PS.zoops, rngseed=1e4, sample.size = 500)
table(sample_data(PS.zoops)$phy_group, sample_data(PS.zoops)$time_point, sample_data(PS.zoops)$lake)
table(sample_data(PS.zoops)$sample_type)


```


```{r alpha diversity plots - plot_richness}


#richness by Lake for zoops only
richness.plot.location <- plot_richness(PS.zoops, x="lake", measures=c("Observed", "Shannon")) + 
  labs(y= "Alpha Diversity Measure", x = "Lake") + theme_bw() + geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90)) + stat_compare_means(method = "anova") +
  theme(axis.text.x=element_text(size=9)) + theme(strip.text.x = element_text(size = 12)) +
  ggtitle("Zooplankton Samples- Microbial Community Alpha Diversity")
richness.plot.location
dev.copy(pdf, "phyloseq analysis/figures/richness.plot.location.pdf", height=5, width=7)
dev.off()

#richness by Lake for water only
water.richness.plot.location <- plot_richness(PS.water, x="lake", measures=c("Observed", "Shannon")) + 
  labs(y= "Alpha Diversity Measure", x = "Lake") + theme_bw() + geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90)) + stat_compare_means(method = "anova") +
  theme(axis.text.x=element_text(size=9)) + theme(strip.text.x = element_text(size = 12)) + 
  ggtitle("Water Samples- Microbial Community Alpha Diversity")
water.richness.plot.location
dev.copy(pdf, "phyloseq analysis/figures/water.richness.plot.location.pdf", height=5, width=7)
dev.off()

#richness by organism (or sample type i.e, water)
richness.plot.organism <- plot_richness(PS.500, x="sample_type", measures=c("Observed", "Shannon")) + 
  labs(y= "Alpha Diversity Measure", x = "Organism") + theme_bw() + geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90)) + stat_compare_means(method = "anova")
richness.plot.organism
dev.copy(pdf, "phyloseq analysis/figures/richness.plot.organism.pdf", height=4, width=12)
dev.off() 


#richness by phy_group (or sample type i.e, water)
richness.plot.phy <- plot_richness(PS.rare, x="phy_group", measures=c("Observed", "Shannon")) +
  labs(y= "Alpha Diversity Measure", x = "Sample Type") + theme_bw() + geom_boxplot() +
  theme(axis.text.x = element_text(angle = 90)) + stat_compare_means(method = "anova")
richness.plot.phy
dev.copy(pdf, "phyloseq analysis/figures/richness.plot.phy.pdf", height=4, width=12)
dev.off() 


#richness by Time.point
richness.plot.time.point <- plot_richness(PS.zoops, x="time_point", measures=c("Observed", "Shannon")) + 
  labs(y= "Alpha Diversity Measure", x = "Time Point") + theme_bw() + geom_boxplot() +
  stat_compare_means(method = "anova")
richness.plot.time.point
dev.copy(pdf, "phyloseq analysis/figures/richness.plot.time.point.pdf", height=4, width=10)
dev.off() 

```


```{r ASV Richness and Diversity Plots}

######### both sample types combined
# measure richness and shannon diversity

rich.rare <- richness(otu_table(PS.500), index ="observed")
div.rare <- diversity(otu_table(PS.500), index="shannon")

# split	(Optional). Logical. Should a separate set of richness estimates be performed for each sample? Or alternatively, pool all samples and estimate richness of the entire set.

a.diversity.PS500 <- estimate_richness(PS.fin, measures = c("Observed", "Shannon"))

# combine into one data frame
rare.a.diversity <- cbind(rich.rare, div.rare)
rare.a.diversity$sequencing_ID <- rownames(div.rare)

#now combine with metadata
rare.a.diversity <- cbind(rare.a.diversity, sample_data(PS.500)[,c(3,4,7,8,9,10:19,22:24)])

# lake and sample type are significant
anova(lm(shannon ~ lake + time_point + sample_type + phy_group + temp_C + DOC + elevation_m + chla_ug.L + pH + DO_perc + cond + basin, data = rare.a.diversity))
anova(lm(observed ~ lake + time_point + sample_type + phy_group + temp_C + DOC + elevation_m + chla_ug.L + pH + DO_perc + cond + basin, data = rare.a.diversity))

# summaryBy for averages and lm for regression model to conduct anova for significance
# rare.sum_div <- summaryBy(shannon ~ lake, data = rare.a.diversity, FUN = c(mean, sd))
# rare.sum_rich <- summaryBy(observed ~ lake, data = rare.a.diversity, FUN = c(mean, sd))

# plots

bargraph.CI(sample_type, shannon, data = rare.a.diversity,
                     xlab = "Sample Type", ylab = "Shannon Diversity", cex.lab = 1.5, x.leg = 2,
                     col = "black", angle = 45, cex.names = 1.25,
                     density = c(0,20), legend = TRUE)

bargraph.CI(lake, shannon, data = rare.a.diversity,
                     xlab = "Lake", ylab = "Shannon Diversity", cex.lab = 1, x.leg = 2,
                     col = "black", angle = 45, cex.names = 1,
                     density = c(0,20), legend = TRUE)

bargraph.CI(sample_type, observed, data = rare.a.diversity,
                     xlab = "Sample Type", ylab = "Species Richness", cex.lab = 1.5, x.leg = 2,
                     col = "black", angle = 45, cex.names = 1.25,
                     density = c(0,20), legend = TRUE)

bargraph.CI(lake, observed, data = rare.a.diversity,
                     xlab = "Lake", ylab = "Species Richness", cex.lab = 1, x.leg = 2,
                     col = "black", angle = 45, cex.names = 1,
                     density = c(0,20), legend = TRUE)

rare.a.div.thru.time <- ggplot(rare.sum_div, aes(x=time_point, y=shannon.mean, group=lake))+
  facet_grid(rows="lake") +
  ylim(1,5) +
  geom_point(size = 1.2) +
  geom_line(size = 0.3) +
  xlab('Sampling Time Point') +
  ylab('Average Shannon Diversity') +
  geom_errorbar(aes(ymin=shannon.mean - shannon.sd, ymax=shannon.mean + shannon.sd), width=.05,
                 position=position_dodge(0.05)) +
  theme(strip.text.y = element_text(size = 8, hjust = 0.5,
          vjust = 0.5, face = 'bold'))
rare.a.div.thru.time
dev.copy(pdf, "phyloseq analysis/figures/rare.a.div.thru.time.pdf", height=6, width=8)
dev.off() 

rare.rich.thru.time <- ggplot(sum_rich, aes(x=time_point, y=observed.mean, group=lake))+
  facet_grid(rows="lake") +
  ylim(10,150) +
  geom_point(size = 1.2) +
  geom_line(size = 0.3) +
  xlab('Sampling Time Point') +
  ylab('Average Richness') +
  geom_errorbar(aes(ymin=observed.mean - observed.sd, ymax=observed.mean + observed.sd), width=.05,
                 position=position_dodge(0.05)) +
  theme(strip.text.y = element_text(size = 8, hjust = 0.5,
          vjust = 0.5, face = 'bold')) +
  stat_compare_means(method = "anova")
rare.rich.thru.time
dev.copy(pdf, "phyloseq analysis/figures/rare.rich.thru.time.pdf", height=6, width=8)
dev.off() 



#################### 
# zoops only

rich.zoops <- richness(otu_table(PS.zoops), index="observed")
div.zoops <- diversity(otu_table(PS.zoops), index="shannon")

a.diversity.zoops <- cbind(rich.zoops, div.zoops)
a.diversity.zoops$sequencing_ID <- rownames(div.zoops)

a.diversity.zoops <- cbind(a.diversity.zoops, sample_data(PS.zoops)[,c(6,8:18,21)])

# sum_div <- summaryBy(shannon ~ lake*time_point, data = a.diversity, FUN = c(mean, sd))
# sum_rich <- summaryBy(observed ~ lake*time_point, data = a.diversity, FUN = c(mean, sd))

# nothing is significant
anova(lm(shannon ~ lake + time_point + phy_group + temp_C + DOC + elevation_m + chla_ug.L + pH + DO_perc + cond, data = a.diversity.zoops))

# time point, DOC, and chlorophyll
anova(lm(observed ~ lake + time_point + phy_group + temp_C + DOC + elevation_m + chla_ug.L + pH + DO_perc + cond, data = a.diversity.zoops))


# plots
ggplot(data = rare.a.diversity, aes(x = DOC, y = observed)) +
  geom_point() + geom_abline()
duplicated_columns <- duplicated(names(rare.a.diversity))
print(duplicated_columns)
colnames(rare.a.diversity)[4] <- "sample"

bargraph.CI(chla_ug.L, observed, data = rare.a.diversity,
                     xlab = "Lake", ylab = "Species Richness", cex.lab = 1, x.leg = 2,
                     col = "black", angle = 45, cex.names = 1,
                     density = c(0,20), legend = TRUE)


a.div.thru.time <- ggplot(sum_div, aes(x=time_point, y=shannon.mean, group=lake))+
  facet_grid(rows="lake") +
  ylim(1,5) +
  geom_point(size = 1.2) +
  geom_line(size = 0.3) +
  xlab('Sampling Time Point') +
  ylab('Average Shannon Diversity') +
  geom_errorbar(aes(ymin=shannon.mean - shannon.sd, ymax=shannon.mean + shannon.sd), width=.05,
                 position=position_dodge(0.05)) +
  theme(strip.text.y = element_text(size = 8, hjust = 0.5,
          vjust = 0.5, face = 'bold'))
a.div.thru.time
dev.copy(pdf, "phyloseq analysis/figures/a.div.thru.time.pdf", height=6, width=8)
dev.off() 

rich.thru.time <- ggplot(sum_rich, aes(x=time_point, y=observed.mean, group=lake))+
  facet_grid(rows="lake") +
  ylim(10,150) +
  geom_point(size = 1.2) +
  geom_line(size = 0.3) +
  xlab('Sampling Time Point') +
  ylab('Average Richness') +
  geom_errorbar(aes(ymin=observed.mean - observed.sd, ymax=observed.mean + observed.sd), width=.05,
                 position=position_dodge(0.05)) +
theme(strip.text.y = element_text(size = 8, hjust = 0.5,
          vjust = 0.5, face = 'bold')) +
  stat_compare_means(method = "anova")
rich.thru.time
dev.copy(pdf, "phyloseq analysis/figures/rich.thru.time.pdf", height=6, width=8)
dev.off() 


######################### 
# Water
rich.water <- richness(otu_table(PS.water), index="observed")
div.water <- diversity(otu_table(PS.water), index="shannon")

water.a.diversity <- cbind(rich.water, div.water)
water.a.diversity$sequencing_ID <- rownames(div.water)

water.a.diversity <- cbind(water.a.diversity, sample_data(PS.water)[,c(6,8:18)])

# water_sum_div <- summaryBy(shannon ~ lake*time_point, data = water.a.diversity, FUN = c(mean, sd))
# water_sum_rich <- summaryBy(observed ~ lake*time_point, data = water.a.diversity, FUN = c(mean, sd))

# nothing is significant
anova(lm(shannon ~ lake + time_point + temp_C + DOC + elevation_m + chla_ug.L + pH + DO_perc + cond, data = water.a.diversity))
anova(lm(observed ~ lake + time_point + temp_C + DOC + elevation_m + chla_ug.L + pH + DO_perc + cond, data = water.a.diversity))

# plots
bargraph.CI(time_point, shannon, data = water.a.diversity,
                     xlab = "Sampling Time Point", ylab = "Shannon Diversity", cex.lab = 1.5, x.leg = 2,
                     col = "black", angle = 45, cex.names = 1.25,
                     density = c(0,20), legend = TRUE)

water.a.div.lakes <- ggplot(water_sum_div, aes(x=lake, y=shannon.mean))+
  ylim(1,5) +
  geom_point(size = 1.2) +
  geom_line(size = 0.3) +
  xlab('Sampling Time Point') +
  ylab('Average Shannon Diversity') +
  geom_errorbar(aes(ymin=shannon.mean - shannon.sd, ymax=shannon.mean + shannon.sd), width=.05,
                 position=position_dodge(0.05)) +
  theme(strip.text.y = element_text(size = 8, hjust = 0.5,
          vjust = 0.5, face = 'bold'))
water.a.div.lakes
dev.copy(pdf, "phyloseq analysis/figures/water.a.div.lakes.pdf", height=6, width=8)
dev.off() 

```


```{r testing effects of env. variables on richness and shannon diversity}
############ testing effects of env. variables on richness and shannon diversity
env_div <- anova(lm(shannon ~ chla_ug.L, data = rare.a.diversity))
summary(env_div)

# run these 2 lines together
plot(rare.a.diversity$chla_ug.L, rare.a.diversity$shannon)
abline(lm(rare.a.diversity$shannon~rare.a.diversity$chla_ug.L), col = 'black')

anova(lm(shannon ~ cond, data = rare.a.diversity))
summary(lm(shannon ~ cond, data = rare.a.diversity))

anova(lm(shannon ~ basin, data = rare.a.diversity))
summary(lm(shannon ~ basin, data = rare.a.diversity))
barchart = bargraph.CI(sample_type, shannon, data = rare.a.diversity,
                     xlab = "Lake", ylab = "Shannon Diversity", cex.lab = 1.5, x.leg = 2,
                     col = "black", angle = 45, cex.names = 1.25,
                     density = c(0,20), legend = TRUE)
plot(rare.a.diversity$cond, rare.a.diversity$shannon)
abline(lm(rare.a.diversity$shannon~rare.a.diversity$cond), col = 'black')
```


```{r testing effects of env. variables on rel. abund. and richness of dominant bacteria}
# Extract the abundance matrix
abundance_matrix <- as.data.frame(otu_table(PS.500))

# Extract the metadata
metaD <- as.data.frame(sample_data(PS.500))

# Extract the taxa table
taxa_table <- as.data.frame(tax_table(PS.500))

# Subset the abundance matrix to include only the class of interest
class_abundance <- abundance_matrix[taxa_table$Class == "Alphaproteobacteria",]
alphaproteobacteria <- subset(taxa_table, Class == "Alphaproteobacteria")
merged_data <- merge(alphaproteobacteria, t(abundance_matrix), metaD, by = "row.names")


Bacteroidia <- subset(Abund500, Class == "Bacteroidia")
Bacteroidia <- summaryBy(Abundance ~ sequence_ID, data = Bacteroidia, FUN = sum)
Bacteroidia <- data.frame(Bacteroidia, row.names = Bacteroidia[,1])
Bacteroidia <- merge(Bacteroidia, metadata_500, by = "row.names")
anova(lm(Abundance.sum ~ lake + time_point + phy_group + latitude +
               longitude + elevation_m + temp_C + chla_ug.L + pH + DO_perc +
               cond + DOC + elev.cat + basin, data = Bacteroidia))
# nothing significant

gammaproteobacteria <- subset(Abund500, Class == "Gammaproteobacteria")
gammaproteobacteria <- summaryBy(Abundance ~ sequence_ID, data = gammaproteobacteria, FUN = sum)
gammaproteobacteria <- data.frame(gammaproteobacteria, row.names = gammaproteobacteria[,1])
gammaproteobacteria <- merge(gammaproteobacteria, metadata_500, by = "row.names")
FitAll <- lm(Abundance.sum ~ lake + time_point + phy_group + latitude +
               longitude + elevation_m + temp_C + chla_ug.L + pH + DO_perc +
               cond + DOC + elev.cat + basin, data = gammaproteobacteria)
anova(FitAll) # Dissolved oxygen is important
step(FitAll, direction = "both")


alphaproteobacteria <- subset(Abund500, Class == "Alphaproteobacteria")
alphaproteobacteria <- summaryBy(Abundance ~ sequence_ID, data = alphaproteobacteria, FUN = sum)
alphaproteobacteria <- data.frame(alphaproteobacteria, row.names = alphaproteobacteria[,1])
alphaproteobacteria <- merge(alphaproteobacteria, metadata_500, by = "row.names")
anova(lm(Abundance.sum ~ lake, data = alphaproteobacteria))
# lake is important

# Merge the abundance matrix, metadata, and taxa table based on sample IDs
merged_data <- merge(t(class_abundance), metaD, by = "row.names", na.rm= TRUE)
merged_data <- t(merged_data)
model <- lm(class_abundance ~ env_var, data = merged_data)
summary(model)
```


```{r correlations}

# cor.test(metadata$elevation_m, metadata$DOC)

ggscatter(sample_data(PS.500), x = "elevation_m", y = "DOC", color = "lake",
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "elevation (m)", ylab = "DOC")

ggscatter(sample_data(PS.500), x = "elevation_m", y = "chla_ug.L", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "elevation (m)", ylab = "chla")

ggscatter(sample_data(PS.500), x = "elevation_m", y = "temp_C", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "elevation (m)", ylab = "Temperature (C)")

ggscatter(sample_data(PS.500), x = "latitude", y = "temp_C",
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "latitude", ylab = "Temperature")

ggscatter(sample_data(PS.500), x = "latitude", y = "DOC", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "latitude", ylab = "DOC")

ggscatter(sample_data(PS.500), x = "latitude", y = "chla_ug.L", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "latitude", ylab = "chla")

ggscatter(sample_data(PS.500), x = "DOC", y = "chla_ug.L", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "DOC", ylab = "chla")

ggscatter(sample_data(PS.500), x = "temp_C", y = "pH", color = "lake",
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "temp_C", ylab = "pH")

```


```{r environmental effects on bacterial comm. dissimilarities}

colSums(is.na(metadata))
# As stated early in the article, PCA only works with numerical values
numerical_data <- metadata[,10:19]
data_normalized <- scale(numerical_data)
head(data_normalized)
corr_matrix <- cor(data_normalized)
ggcorrplot(corr_matrix)
data.pca <- princomp(corr_matrix)
summary(data.pca)
fviz_eig(data.pca, addlabels = TRUE)
fviz_pca_var(data.pca, col.var = "black")

```


```{r Bray curtis distance matrix}

raw.asv.table <- as.data.frame(t(otu_table(PS.fin)))
str(raw.asv.table)
str(ASV_table)

# 3 methods for calculating bray curtis dissimilarity

set.seed(10000)
avg_bray_500 <- avgdist(raw.asv.table, sample = 500)

set.seed(10000)
bc_vegdist <- vegdist(t(otu_table(PS.fin)), method = "bray")

# will use these moving forward:
set.seed(10000)
bc <- phyloseq::distance(PS.fin, method = "bray")
set.seed(10000)
bc_500 <- phyloseq::distance(PS.500, method = "bray")
set.seed(10000)
bc_zoops <- phyloseq::distance(PS.zoops, method = "bray")
set.seed(10000)
bc_water <- phyloseq::distance(PS.water, method = "bray")

```


```{r metaMDS and PERMANOVAS from BC distance matrices}

set.seed(10000)
nmds1 <- metaMDS(bc,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)

species.scores1 <- as.data.frame(scores(nmds1, display="site"))
nmds.merge1 <- merge(sample_data(PS.fin), species.scores1, by = 'row.names', all = TRUE)
nmds.merge1$basin <- as.factor(ifelse(nmds.merge1$lake == "Eastern Brook" | nmds.merge1$lake == "Serene" | nmds.merge1$lake == "Convict", "South", "North"))

adonis2(bc ~ lake*time_point*sample_type, data = nmds.merge1, permutations = 1000)

#################
# rarefied
set.seed(10000)
nmds2 <- metaMDS(bc_500,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)

NMDS1=nmds2$points[,1]
NMDS2=nmds2$points[,2]


species.scores2 <- as.data.frame(scores(nmds2, display="site"))
nmds.merge2 <- merge(sample_data(PS.500), species.scores2, by = 'row.names', all = TRUE)
nmds.merge2$basin <- as.factor(ifelse(nmds.merge2$lake == "Eastern Brook" | nmds.merge2$lake == "Serene" | nmds.merge2$lake == "Convict", "South", "North"))

broad.model2 <- adonis2(bc_500 ~ lake*time_point*sample_type, data = nmds.merge2, permutations = 1000)
broad.model2
env.model2 <- adonis2(bc_500 ~ latitude + longitude + elevation_m + temp_C + pH + cond + DOC + chla_ug.L + DO_perc, data = nmds.merge2, permutations = 1000)
env.model2

#test for beta dispersion
anova(betadisper(bc_500, nmds.merge2$time_point)) # not significant
anova(betadisper(bc_500, nmds.merge2$sample_type)) # not significant


##################
# zoops only
set.seed(10000)
nmds3 <- metaMDS(bc_zoops,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)

species.scores3 <- as.data.frame(scores(nmds3, display="site"))
nmds.merge3 <- merge(sample_data(PS.zoops), species.scores3, by = 'row.names', all = TRUE)
nmds.merge3$basin <- as.factor(ifelse(nmds.merge3$lake == "Eastern Brook" | nmds.merge3$lake == "Serene" | nmds.merge3$lake == "Convict", "South", "North"))

broad.model3 <- adonis2(bc_zoops ~ time_point + lake + phy_group, data = nmds.merge3, permutations = 1000)
broad.model3
env.model3 <- adonis2(bc_zoops ~ latitude + longitude + elevation_m + temp_C + pH + cond + DOC + chla_ug.L + DO_perc, data = nmds.merge3, permutations = 1000)
env.model3


#######################
# water only
set.seed(10000)
nmds4 <- metaMDS(bc_water,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)

species.scores4 <- as.data.frame(scores(nmds4, display="site"))
nmds.merge4 <- merge(sample_data(PS.water), species.scores4, by = 'row.names', all = TRUE)
nmds.merge4$region <- as.factor(ifelse(nmds.merge4$lake == "Eastern Brook" | nmds.merge4$lake == "Serene" | nmds.merge4$lake == "Convict", "South", "North"))

broad.model4 <- adonis2(bc_water ~ time_point + lake + region, data = nmds.merge4, permutations = 1000)
broad.model4
env.model4 <- adonis2(bc_water ~ latitude + longitude + elevation_m + temp_C + pH + cond + DOC + chla_ug.L + DO_perc, data = nmds.merge4, permutations = 1000)
env.model4
```


```{r combining NMDS and PCA}
#Take environmental data, run PCA to compress to a single variable (PC1), and examine
#relationship between PC1 and NMDS1, as a way to examine beta diversity x environment. 
#Output here are testing NMDS relationship against PC1 and habitat using linear models.

numerical_data <- as.matrix(metadata_500[,c(9:18)])
env.PCA <- prcomp(numerical_data, center = TRUE, scale= TRUE)
summary(env.PCA)
pc_scores <- scores(env.PCA)
as.data.frame(pc_scores)
pc_loadings <- env.PCA$rotation
NMDSxPCA_data <- data.frame(NMDS1 = species.scores2$NMDS1, PC1 = pc_scores[, 1], time = metadata_500$time_point, lake = metadata_500$lake)
ggplot(NMDSxPCA_data, aes(x = PC1, y = NMDS1, color = lake)) +
  geom_point() +  
  geom_smooth(aes(color=lake, fill=lake), method = "lm", alpha = .2) +
  xlab("PC1 (Environment)") + ylab("NMDS (Bray-Curtis)") +
  theme_bw()

anova(lm(NMDS1 ~ PC1*lake, data = NMDSxPCA_data))
```



```{r envfit}

#Fit environmental vectors to ordination to see which environmental variables are correlated with the ordination

#environmental variables, continuous
env500 <- sample_data(PS.500)[ ,c(10,11,14,16,18)]
envzoops <- sample_data(PS.zoops)[ ,c(1, 10:19)]
envwater <- sample_data(PS.water)[ ,c(10:19)]

# factors, categorical
facs500 <- sample_data(PS.500)[,c(7,9)]
facszoops <- sample_data(PS.zoops)[,c(7,9)]

# run envfit functions
fit.env <- envfit(nmds1, env, na.rm = T)
fit.env500 <- envfit(nmds2, env500, na.rm = T)
fit.envzoops <- envfit(nmds3, envzoops, na.rm = T)
fit.envwater <- envfit(nmds4, envwater, na.rm = T)

NMDS.plot.df=data.frame(NMDS1=NMDS1,NMDS2=NMDS2, metadata_500[,6])
times <- NMDS.plot.df$time_point
levels(NMDS.plot.df$time_point)

# colors and groups for plots
time_colors <- scale_color_manual(values = c("royalblue1", "chartreuse3", "chocolate1", "deeppink2", "khaki4"))
lake_colors <- c("royalblue1", "chartreuse3", "chocolate1", "deeppink2", "khaki4", "orchid")
leg.col<-c("chartreuse3", "chocolate1", "orchid", "khaki4", "deeppink2", "royalblue1")
lake.group<- nmds.merge2$lake

# centroid
centroid <- nmds.merge2 %>% 
  group_by(time_point) %>%
  summarize(NMDS1 = mean(NMDS1), NMDS2 = mean(NMDS2))


### make plot by habitat with vectors for environment
environm_plot<-ordiplot(nmds2, type="n", main=substitute(paste("")), cex.main=1, display="sites", xlim=c(-0.25, 0.8), cex.lab=0.8, cex.axis=0.8)
abline(h = 0, lty = "dotted")
abline(v = 0, lty = "dotted")
points(nmds2, "sites", cex=0.8, pch=16, col=lake_colors[lake.group])
ordihull(environm_plot, groups=nmds.merge2$lake, draw="polygon", alpha=80, col = lake_colors)
legend("topright", legend=levels(lake.group), cex=1.5, pch=16, col=leg.col, pt.cex=1, bty="n")
par.new=T
plot(fit.env500, col="black", p.max=0.05, cex=0.9, lwd=1)
dev.copy(pdf, "phyloseq analysis/figures/environm_plot.pdf", height=5.5, width=6)
dev.off() 

data.scores <- as.data.frame(scores(nmds1))
data.scores.df <- cbind(facs, data.scores)
data.scores500 <- as.data.frame(scores(nmds2))
data.scores.df500 <- cbind(facs500, data.scores500)
data.scores1000 <- as.data.frame(scores(nmds3))
data.scores.df1000 <- cbind(facs1000, data.scores1000)

en_coord_cont = as.data.frame(scores(fit.env, "vectors")) * ordiArrowMul(fit.env)
en_coord_cat = as.data.frame(scores(fit.env, "factors")) * ordiArrowMul(fit.env)
en_coord_cont500 = as.data.frame(scores(fit.env500, "vectors")) * ordiArrowMul(fit.env500)
en_coord_cat500 = as.data.frame(scores(fit.env500, "factors")) * ordiArrowMul(fit.env500)
en_coord_cont1000 = as.data.frame(scores(fit.env1000, "vectors")) * ordiArrowMul(fit.env1000)
en_coord_cat1000 = as.data.frame(scores(fit.env1000, "factors")) * ordiArrowMul(fit.env1000)


envfit_plot = ggplot(data = data.scores.df, aes(x = NMDS1, y = NMDS2)) + 
     geom_point(data = data.scores.df, aes(colour = lake), size = 3, alpha = 0.5) + 
     scale_colour_manual(values = c("orange", "steelblue", "maroon", "mediumseagreen", "coral", "pink"))  + 
     geom_segment(aes(x = 0, y = 0, xend = NMDS1, yend = NMDS2), 
       data = en_coord_cont, size = 1, arrow = arrow(length = unit(0.25, "cm")),
       alpha = 0.5, colour = "black") +
     geom_point(data = en_coord_cat, aes(x = NMDS1, y = NMDS2), 
       shape = "diamond", size = 4, alpha = 0.6, colour = "navy") +
     geom_text(data = en_coord_cat, aes(x = NMDS1, y = NMDS2+0.04), 
       label = row.names(en_coord_cat), colour = "navy", fontface = "bold") + 
     geom_text(data = en_coord_cont, aes(x = NMDS1, y = NMDS2), colour = "black", 
       fontface = "bold", label = row.names(en_coord_cont)) + 
     stat_ellipse(aes(colour = lake)) +
     theme(axis.title = element_text(size = 10, face = "bold", colour = "black"), 
       panel.background = element_blank(), panel.border = element_rect(fill = NA, colour = "black"), 
       axis.ticks = element_blank(), axis.text = element_blank(), legend.key = element_blank(), 
       legend.title = element_text(size = 10, face = "bold", colour = "black"), 
       legend.text = element_text(size = 9, colour = "black")) + 
     labs(colour = "lake")
     
envfit_plot
dev.copy(pdf, "phyloseq analysis/figures/envfit_plot.pdf", height=4, width=5)
dev.off() 


envfit_plot500 = ggplot(data = data.scores.df500, aes(x = NMDS1, y = NMDS2)) + 
     geom_point(data = data.scores.df500, aes(colour = lake), size = 3, alpha = 0.5) + 
     scale_colour_manual(values = c("orange", "steelblue", "maroon", "mediumseagreen", "coral", "pink"))  + 
     geom_segment(aes(x = 0, y = 0, xend = NMDS1, yend = NMDS2), 
       data = en_coord_cont500, size = 1, arrow = arrow(length = unit(0.25, "cm")),
       alpha = 0.5, colour = "black") +
     geom_point(data = en_coord_cat500, aes(x = NMDS1, y = NMDS2), 
       shape = "diamond", size = 4, alpha = 0.6, colour = "navy") +
     geom_text(data = en_coord_cat500, aes(x = NMDS1, y = NMDS2+0.04), 
       label = row.names(en_coord_cat500), colour = "navy", fontface = "bold") + 
     geom_text(data = en_coord_cont500, aes(x = NMDS1, y = NMDS2), colour = "black", 
       fontface = "bold", label = row.names(en_coord_cont500)) + 
     stat_ellipse(aes(colour = lake)) +
     theme(axis.title = element_text(size = 10, face = "bold", colour = "black"), 
       panel.background = element_blank(), panel.border = element_rect(fill = NA, colour = "black"), 
       axis.ticks = element_blank(), axis.text = element_blank(), legend.key = element_blank(), 
       legend.title = element_text(size = 10, face = "bold", colour = "black"), 
       legend.text = element_text(size = 9, colour = "black")) + 
     labs(colour = "lake")
     
envfit_plot500
dev.copy(pdf, "phyloseq analysis/figures/envfit_plot500.pdf", height=4, width=5)
dev.off() 

```



```{r mantel test, geographic distance}
install.packages("fossil")
library(fossil)
install.packages("ade4")
library(ade4)

# to calculate geographic distances between lakes
# geographic distance dissim matrix - adapted from weinstein et al 
gps <- as.data.frame(cbind(sample_data(PS.500)$longitude, sample_data(PS.500)$latitude))
spatial_dist <- distm(gps, fun = distGeo)
rownames(spatial_dist) <- sample_data(PS.500)$lake
colnames(spatial_dist) <- sample_data(PS.500)$lake
spatial_dist <- as.dist(spatial_dist)

# converting distances to km 
spatial_dist_km <- spatial_dist/1000
dim(spatial_dist_km)
spatial_dist_km <- as.dist(spatial_dist_km) # make distance matrix

set.seed(10000)
bc_500 <- phyloseq::distance(PS.500, method = "bray")


#just to check dims
bc_dist <- as.matrix(bc_500)
dim(bc_dist)

mantel_test <- mantel.rtest(spatial_dist, bc_500, nrepet=999)
mantel_test

dist.df <- data.frame(Distance=spatial_dist[lower.tri(spatial_dist)],
                  BrayCurtis= bc_500[lower.tri(bc_500)])

##############
#Generate plots for all samples combined
dist_plot <- ggplot(dist.df, aes(x=log(Distance+1), y=BrayCurtis)) +
  geom_point(size=0.8,alpha=0.5) +
  geom_smooth(method = "glm", formula = y~x,
              method.args = list(family = gaussian(link = 'log')), lwd=0.7)+ 
  ggtitle("Bacterial Community Distances")+ 
  scale_y_continuous(name="Bray-Curtis Dissimilarity",breaks=seq(0,1,0.25),limits=c(0,1)) +
  scale_x_continuous(name="log(Distance (m)+1)",breaks=seq(0,8,2),limits=c(0,8)) +
  theme(plot.title = element_text(size=10, hjust = 0.5),
        axis.text.x=element_text(colour="black",size=7), 
        axis.title.x=element_blank(),
        axis.text.y=element_text(colour="black",size=7),
        axis.title=element_text(colour="black",size=9),
        legend.title=element_text(colour="black",size=7,face="bold"),
        legend.text=element_text(colour="black",size=7),
        panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
        panel.background=element_blank())


```


```{r MDS plots by time point}
# Time 1
T1 <- subset_samples(PS.fin, time_point=="1")
ORD.T1 <- ordinate(T1, method='MDS', distance='bray')

MDS.ord.T1 <- plot_ordination(
  physeq = T1,                                                   
  ordination = ORD.T1) +                                                
  geom_point(aes(color = lake, shape= phy_group), size = 3) +    
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  scale_color_brewer(palette = "Dark2") +
  scale_fill_brewer(palette = "Dark2") + ggtitle("Time1") + theme_classic()
MDS.ord.T1
dev.copy(pdf, "phyloseq analysis/figures/MDS.ord.T1.pdf", height=6, width=7)
dev.off() 

# Time 2
T2<- subset_samples(PS.fin, time_point=="2")
ORD.T2 <- ordinate(T2, method='MDS', distance='bray')
MDS.ord.T2<-plot_ordination(
  physeq = T2,                                                   
  ordination = ORD.T2) +                                                
  geom_point(aes(color = lake, shape= phy_group), size = 3) +    
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  scale_color_brewer(palette = "Dark2") +
  scale_fill_brewer(palette = "Dark2") + ggtitle("Time2") +
  theme_classic()     
MDS.ord.T2
dev.copy(pdf, "phyloseq analysis/figures/MDS.ord.T2.pdf", height=6, width=7)
dev.off() 

# Time 3
T3<- subset_samples(PS.fin, time_point=="3")
ORD.T3 <- ordinate(T3, method='MDS', distance='bray')
MDS.ord.T3<-plot_ordination(
  physeq = T3,                                                   
  ordination = ORD.T3) +                                                
  geom_point(aes(color = lake, shape= phy_group), size = 3) +    
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  scale_color_brewer(palette = "Dark2") +
  scale_fill_brewer(palette = "Dark2") + ggtitle("Time3") +
  theme_classic()     
MDS.ord.T3
dev.copy(pdf, "phyloseq analysis/figures/MDS.ord.T3.pdf", height=6, width=7)
dev.off() 

# Time 4
T4<- subset_samples(PS.fin, time_point=="4")
ORD.T4 <- ordinate(T4, method='MDS', distance='bray')
MDS.ord.T4<-plot_ordination(
  physeq = T4,                                                   
  ordination = ORD.T4) +                                                
  geom_point(aes(color = lake, shape= phy_group), size = 3) +    
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  scale_color_brewer(palette = "Dark2") +
  scale_fill_brewer(palette = "Dark2") + ggtitle("Time4") +
  theme_classic()     
MDS.ord.T4
dev.copy(pdf, "phyloseq analysis/figures/MDS.ord.T4.pdf", height=6, width=7)
dev.off() 

# Time 5
T5 <- subset_samples(PS.fin, time_point=="5")
ORD.T5 <- ordinate(T5, method='MDS', distance='bray')
MDS.ord.T5 <- plot_ordination(
  physeq = T5,                                                   
  ordination = ORD.T5) +                                                
  geom_point(aes(color = lake, shape= phy_group), size = 3) +    
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  scale_color_brewer(palette = "Dark2") +
  scale_fill_brewer(palette = "Dark2") + ggtitle("Time5") +
  theme_classic()     
MDS.ord.T5
dev.copy(pdf, "phyloseq analysis/figures/MDS.ord.T5.pdf", height=6, width=7)
dev.off() 


# now put them all together
combined.mds <- ggarrange(MDS.ord.T1 + theme(axis.title.y=element_text(size=8)),
          MDS.ord.T2 + theme(axis.title.y=element_text(size=8)),
          MDS.ord.T3 + theme(axis.title.y=element_text(size=8)),
          MDS.ord.T4 + theme(axis.title.y=element_text(size=8)),
          MDS.ord.T5 + theme(axis.title.y=element_text(size=8)),
          plot_layout(guides="collect"), labels=NULL, common.legend = TRUE, 
          legend="bottom", align="hv", 
          font.label = list(size=10, color="black", family=NULL, position="top", labels="auto"))
combined.mds
dev.print(pdf, "phyloseq analysis/figures/combined.mds.pdf", height=10, width=15)
dev.off
 
```


```{r NMDS plots broken down by lakes and times}
centroid <- nmds.merge2 %>% 
  group_by(time_point) %>%
  summarize(NMDS1 = mean(NMDS1), NMDS2 = mean(NMDS2))
str(nmds2.merge)


nmds.elev <- ggplot(nmds.merge3, aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = elev.cat), size = 2.5) +
  stat_ellipse(level=0.9, lwd = 1, linetype = 2, aes(color = elev.cat)) +
  theme_bw() + theme(legend.key.size = unit(2, 'cm'), legend.text = element_text(size=20), legend.title = element_text(size=22))
nmds.elev
dev.print(pdf, "phyloseq analysis/figures/nmds.elev.pdf", height=10, width=15)
dev.off

nmds.basin <- ggplot(nmds.merge3, aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = basin), size = 2.5) +
  stat_ellipse(level=0.9, lwd = 1, linetype = 2, aes(color = basin)) +
  theme_bw() + theme(legend.key.size = unit(2, 'cm'), legend.text = element_text(size=20), legend.title = element_text(size=22))
nmds.basin
dev.print(pdf, "phyloseq analysis/figures/nmds.basin.pdf", height=10, width=15)
dev.off

nmds.convict <- ggplot(nmds.merge3[nmds.merge3$lake == "Convict", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point), size = 2.5) +
  stat_ellipse(level=0.9, lwd = 0.7, linetype = 1, aes(color = time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) +
  theme_bw() + ggtitle("Convict")
nmds.convict
dev.print(pdf, "phyloseq analysis/figures/nmds.convict.pdf", height=10, width=15)
dev.off

nmds.convict.water <- ggplot(nmds.merge4[nmds.merge4$lake == "Convict", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point), size = 2.5) +
  stat_ellipse(level=0.9, lwd = 0.7, linetype = 1, aes(color = time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) +
  theme_bw() + ggtitle("Convict")
nmds.convict.water
dev.print(pdf, "phyloseq analysis/figures/nmds.convict.water.pdf", height=10, width=15)
dev.off

nmds.east.brook <- ggplot(nmds.merge3[nmds.merge3$lake == "Eastern Brook", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point, shape = sample_type), size = 2.5) +
  stat_ellipse(level=0.9, lwd = 0.7, linetype = 1, aes(color = time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) +
  theme_bw() + ggtitle("Eastern Brook")
nmds.east.brook
dev.print(pdf, "phyloseq analysis/figures/nmds.east.brook.pdf", height=10, width=15)
dev.off

nmds.serene <- ggplot(nmds.merge3[nmds.merge3$lake == "Serene", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point, shape = sample_type), size = 2.5) +
  stat_ellipse(level=0.9, lwd = 0.7, linetype = 1, aes(color = time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) +
  theme_bw() + ggtitle("Serene")
nmds.serene
dev.print(pdf, "phyloseq analysis/figures/nmds.serene.pdf", height=10, width=15)
dev.off

nmds.cooney <- ggplot(nmds.merge3[nmds.merge3$lake == "Cooney", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point, shape = sample_type), size = 2.5) +
  stat_ellipse(level=0.9, lwd = 0.7, linetype = 1, aes(color = time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) +
  theme_bw() + ggtitle("Cooney")
nmds.cooney
dev.print(pdf, "phyloseq analysis/figures/nmds.cooney.pdf", height=10, width=15)
dev.off

nmds.blue <- ggplot(nmds.merge3[nmds.merge3$lake == "Blue", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point, shape = sample_type), size = 2.5) +
  stat_ellipse(level=0.9, lwd = 0.7, linetype = 1, aes(color = time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) +
  theme_bw() + ggtitle("Blue")
nmds.blue
dev.print(pdf, "phyloseq analysis/figures/nmds.blue.pdf", height=10, width=15)
dev.off

nmds.virginia <- ggplot(nmds.merge3[nmds.merge3$lake == "Virginia", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point, shape = sample_type), size = 2.5) +
  stat_ellipse(level=0.9, lwd = 0.7, linetype = 1, aes(color = time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) +
  theme_bw() + ggtitle("Virginia")
nmds.virginia
dev.print(pdf, "phyloseq analysis/figures/nmds.virginia.pdf", height=10, width=15)
dev.off

combined.nmds.lakes <- ggarrange(nmds.convict + theme(axis.title.y=element_text(size=8)),
          nmds.east.brook + theme(axis.title.y=element_text(size=8)),
          nmds.serene + theme(axis.title.y=element_text(size=8)),
          nmds.cooney + theme(axis.title.y=element_text(size=8)),
          nmds.blue + theme(axis.title.y=element_text(size=8)),
          nmds.virginia + theme(axis.title.y=element_text(size=8)),
          plot_layout(guides="collect"), labels=NULL, common.legend = TRUE, 
          legend="bottom", align="hv", 
          font.label = list(size=10, color="black", family=NULL, position="top", labels="auto"))
combined.nmds.lakes
dev.print(pdf, "phyloseq analysis/figures/combined.nmds.lakes.pdf", height=10, width=15)
dev.off

nmds.cladocera.time <- ggplot(nmds.merge3[nmds.merge3$phy_group == "cladocera", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point), size = 4) +
  stat_ellipse(level=0.9, linetype = 2, aes(color = time_point)) +
  theme(axis.text.x= element_text(size = 15)) +
  theme(axis.text.y= element_text(size = 15)) +
  theme_bw() +
  ggtitle("Cladocera")
nmds.cladocera.time
dev.print(pdf, "phyloseq analysis/figures/nmds.cladocera.time.pdf", height=10, width=15)
dev.off

nmds.copepoda.time <- ggplot(nmds.merge3[nmds.merge3$phy_group == "copepoda", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point), size = 4) +
  stat_ellipse(level=0.9, linetype = 2, aes(color = time_point)) +
  theme_bw() +
  ggtitle("Copepoda")
nmds.copepoda.time
dev.print(pdf, "phyloseq analysis/figures/nmds.copepoda.time.pdf", height=10, width=15)
dev.off

nmds.water.time <- ggplot(nmds.merge4, aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(shape = time_point), size = 3) +
  stat_ellipse(level=0.9, linetype = 2, aes(color = time_point)) +
  theme_bw() +
  ggtitle("Water")
nmds.water.time
dev.print(pdf, "phyloseq analysis/figures/nmds.water.time.pdf", height=10, width=15)
dev.off

nmds.zoops.time.basin <- ggplot(nmds.merge3, aes(x=NMDS1, y=NMDS2, color = time_point)) +
  geom_point(aes(color = basin)) +
  stat_ellipse(level=0.9, linetype = 2) +
  theme_bw() +
  ggtitle("Zoops")
nmds.zoops.time.basin
  



# now by time point for rarefied 500
test_nmds11.500 <- ggplot(nmds.merge2[(nmds.merge2$time_point == "1"), ], aes(x=NMDS1, y=NMDS2, color = lake)) +
  geom_point(aes(color = lake, shape = phy_group), size = 3) + 
  stat_ellipse(level=0.9, linetype = 2, aes(color = lake)) +
  ggtitle("Time 1") +
  theme_bw()
test_nmds11.500
dev.print(pdf, "phyloseq analysis/figures/test_nmds11.500.pdf", height=10, width=15)
dev.off

test_nmds12.500 <- ggplot(nmds.merge2[(nmds.merge2$time_point == "2"), ], aes(x=NMDS1, y=NMDS2, color = lake)) +
   geom_point(aes(color = lake, shape = phy_group), size = 3) + 
   stat_ellipse(level=0.9, linetype = 2, aes(color = lake)) +
   ggtitle("Time 2") +
   theme_bw()
test_nmds12.500
dev.print(pdf, "phyloseq analysis/figures/test_nmds12.500.pdf", height=10, width=15)
dev.off

test_nmds13.500 <- ggplot(nmds.merge2[(nmds.merge2$time_point == "3"), ], aes(x=NMDS1, y=NMDS2, color = lake)) +
  geom_point(aes(color = lake, shape = phy_group), size = 3) + 
  stat_ellipse(level=0.9, linetype = 2, aes(color = lake)) +
  ggtitle("Time 3") +
  theme_bw()
test_nmds13.500
dev.print(pdf, "phyloseq analysis/figures/test_nmds13.500.pdf", height=10, width=15)
dev.off

test_nmds14.500 <- ggplot(nmds.merge2[(nmds.merge2$time_point == "4"), ], aes(x=NMDS1, y=NMDS2, color = lake)) +
   geom_point(aes(color = lake, shape = phy_group), size = 3) + 
   stat_ellipse(level=0.9, linetype = 2, aes(color = lake)) +
   ggtitle("Time 4") +
   theme_bw()
test_nmds14.500
dev.print(pdf, "phyloseq analysis/figures/test_nmds14.500.pdf", height=10, width=15)
dev.off

test_nmds15.500 <- ggplot(nmds.merge2[(nmds.merge2$time_point == "5"), ], aes(x=NMDS1, y=NMDS2, color = lake)) +
   geom_point(aes(color = lake, shape = phy_group), size = 3) + 
   stat_ellipse(level=0.9, linetype = 2, aes(color = lake)) +
   ggtitle("Time 5") +
   theme_bw()
test_nmds15.500
dev.print(pdf, "phyloseq analysis/figures/test_nmds15.500.pdf", height=10, width=15)
dev.off

combined.nmds.lake.500 <- ggarrange(test_nmds11.500 + theme(axis.title.y=element_text(size=8)),
          test_nmds12.500 + theme(axis.title.y=element_text(size=8)),
          test_nmds13.500 + theme(axis.title.y=element_text(size=8)),
          test_nmds14.500 + theme(axis.title.y=element_text(size=8)),
          test_nmds15.500 + theme(axis.title.y=element_text(size=8)),
          plot_layout(guides="collect"), labels=NULL, common.legend = TRUE, 
          legend="bottom", align="hv", 
          font.label = list(size=10, color="black", family=NULL, position="top", labels="auto"))
combined.nmds.lake.500
dev.print(pdf, "phyloseq analysis/figures/combined.nmds.lake.500.pdf", height=10, width=15)
dev.off


```


```{r General NMDS}
centroid <- nmds2.merge %>% 
  group_by(lake) %>%
  summarize(NMDS1 = mean(NMDS1), NMDS2 = mean(NMDS2))


nmds.time <- ggplot(nmds.merge4, aes(x=NMDS1, y=NMDS2, color = time_point)) +
  geom_point(aes(color = time_point), size = 3) + 
  stat_ellipse(level=0.9, linetype = 2, aes(color=time_point)) +
  theme_bw()
nmds.time
dev.print(pdf, "phyloseq analysis/figures/nmds.time.pdf", height=10, width=15)
dev.off

nmds.lake <- ggplot(nmds.merge3, aes(x=NMDS1, y=NMDS2, color = lake)) +
  geom_point(aes(color = lake), size = 3) + 
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  theme_bw()
nmds.lake
dev.print(pdf, "phyloseq analysis/figures/nmds.lake.pdf", height=10, width=15)
dev.off

nmds.basin <- ggplot(nmds.merge2, aes(x=NMDS1, y=NMDS2, color = basin)) +
  geom_point(aes(color = basin), size = 3) + 
  stat_ellipse(level=0.9, linetype = 2, aes(color=basin)) +
  theme_bw()
nmds.basin
dev.print(pdf, "phyloseq analysis/figures/nmds.basin.pdf", height=10, width=15)
dev.off

nmds.elevation <- ggplot(nmds.merge2, aes(x=NMDS1, y=NMDS2, color = elev.cat)) +
  geom_point(aes(color = elev.cat), size = 3) + 
  stat_ellipse(level=0.9, linetype = 2, aes(color=elev.cat)) +
  theme_bw()
nmds.elevation
dev.print(pdf, "phyloseq analysis/figures/nmds.elevation.pdf", height=10, width=15)
dev.off
  
nmds.sample_type <- ggplot(nmds.merge2, aes(x=NMDS1, y=NMDS2, color = sample_type)) +
  geom_point(aes(color = sample_type), size = 3) + 
  stat_ellipse(level=0.9, linetype = 2, aes(color=sample_type)) +
  theme_bw()
nmds.sample_type
dev.print(pdf, "phyloseq analysis/figures/nmds.sample_type.pdf", height=10, width=15)
dev.off

nmds.phy_group <- ggplot(nmds.merge2, aes(x=NMDS1, y=NMDS2, color = phy_group)) +
  geom_point(aes(color = phy_group), size = 3) + 
  stat_ellipse(level=0.9, linetype = 2, aes(color=phy_group)) +
  theme_bw()
nmds.phy_group
dev.print(pdf, "phyloseq analysis/figures/nmds.phy_group.pdf", height=10, width=15)
dev.off


```


```{r capscale and dbRDA}
# Trying to find environmental variables that signiificantly explained variation in the 
# 16S community composition

# Spatial variables were calculated using trend-surface analysis
# calculated second-order orthogonal polynomial terms (latitude, longitude, latitude  × longitude, 
# latitude2, longitude2) from UTM coordinates for  each lake using the “poly” function in R.
library(sp)
install.packages('rgdal')
library(rgdal)
gps <- as.data.frame(cbind(sample_data(PS.500)$longitude, sample_data(PS.500)$latitude))
coordinates(gps) <- c("V1", "V2")
# Define the coordinate reference system (CRS) for the latitude and longitude data
proj4string(gps) <- CRS("+proj=longlat +datum=WGS84")
# Convert latitude and longitude to UTM coordinates
utm_coords <- spTransform(gps, CRS("+proj=utm +zone=11 +datum=WGS84"))
# Extract the UTM coordinates from the transformed object
utm_easting <- utm_coords@coords[, 1]
utm_northing <- utm_coords@coords[, 2]
# Create a data frame for the UTM coordinates
coord_df <- data.frame(cbind(utm_easting, utm_northing))
# Calculate second-order orthogonal polynomial terms for latitude and longitude
poly_terms <- poly(coord_df$utm_easting, coord_df$utm_northing, degree = 2)
# Extract the polynomial terms from the matrix
coord_df$latitude_2 <- poly_terms[, 1]
coord_df$longitude_2 <- poly_terms[, 2]
coord_df$latitude_longitude <- poly_terms[, 3]
coord_df$latitude_squared <- poly_terms[, 4]
coord_df$longitude_squared <- poly_terms[, 5]


# CAPSCALE
set.seed(10000)
capscale_result <- capscale(bc_zoops ~ temp_C + DOC + chla_ug.L + cond + DO_perc + pH + longitude + latitude + elevation_m, data = nmds.merge3, distance = "bray")
capscale_model <- capscale(bc_zoops ~ 1, data = nmds.merge3, distance = "bray")
# fitting global model, then performing stepwise selection with ordiRstep
mod <- ordiR2step(capscale_model, scope = capscale_result)
anova(mod)

#dbRDA
set.seed(10000) 
upr <- dbrda(bc_zoops ~ I(temp_C)^2 + I(elevation_m)^2 + I(cond)^2 + I(chla_ug.L)^2 + I(DOC)^2 + I(longitude)^2 + I(latitude)^2 + I(DO_perc)^2 + I(pH)^2, data = nmds.merge3)
lwr <- dbrda(bc_zoops ~ 1, data = nmds.merge3, na.action = 'na.omit')
# fitting global model, then performing stepwise selection with ordiRstep
mod_selec <- ordiR2step(lwr, scope = formula(upr))
anova(mod_selec)# selecting site as most important factor - good stuff honestly simple semi strong model here 
anova(mod_selec, by = "axis", permu = 200)
summary(mod_selec)
RsquareAdj(mod_selec)
screeplot(mod_selec)


dbRDA <- dbrda(formula = bc_zoops ~ I(DOC), data = nmds.merge3, na.action = "na.omit")
plot(dbRDA)
```


```{r PCA}
all_pca <- ordinate(
  physeq = PS.500, 
  method = "PCA", 
  distance = "bray"
)

PCA.ord.plot <- plot_ordination(
  physeq = PS.500,                                                       
  ordination = all_pca, color = "time_point") +                                                
  geom_point(aes(color = phy_group), size = 3) +  
  stat_ellipse(type = "norm", linetype = 2, aes(color=time_point)) +
  scale_shape_manual(values = c(21, 22, 23, 24, 25)) +
  theme_classic() +                                                      
  theme(                             
    legend.text = element_text(size = 12),                               
    legend.title = element_blank(),                                      
    legend.background = element_rect(fill = "white", color = "NA"))+  
  theme(axis.text.y.left = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12))+
  guides(fill = guide_legend(override.aes = list(shape = 21))) 

PCoA.ord.plot
dev.copy(pdf, "phyloseq analysis/figures/PCoA.pdf", height=7, width=10)
dev.off() 

```


```{r bar plot for rare (<0.01 species)}

ASV_rel_abund <- transform_sample_counts(PS.500, function(x) x/sum(x))
ASV_dataframe <- psmelt(ASV_rel_abund)
rare_subset <- subset(ASV_dataframe, ASV_dataframe$Abundance < 0.01)

Abund_rare <- aggregate(Abundance ~ sequence_ID + time_point + lake + organism_water + phy_group + 
                     Phylum + Class + Order + Family +
                     Genus + Species, data=rare_subset, FUN=mean)

rare_genus <- ggplot(Abund_rare, aes(x = phy_group, y = Abundance, fill= Genus, color = Genus)) +
  geom_bar(aes(fill = Genus, color = Genus), stat ="identity", position ="fill") +
  xlab("Sampling Time Point") +
  theme_bw() + theme(axis.text.x=element_text(size=17)) +
  theme(axis.text.y=element_text(size=10)) +
  theme(strip.text.y = element_text(size = 15)) 
rare_genus
dev.copy(pdf, "phyloseq analysis/figures/rare_genus.pdf", height=8, width=7)
dev.off()

rare_class <- ggplot(Abund_rare, aes(x = phy_group, y = Abundance, fill= Class, color = Class)) +
  geom_bar(aes(fill = Class, color = Class), stat ="identity", position ="fill") +
  xlab("Sampling Time Point") +
  theme_bw() + theme(axis.text.x=element_text(size=17)) +
  theme(axis.text.y=element_text(size=10)) +
  theme(strip.text.y = element_text(size = 15)) 
rare_class


```


```{r another way to categorize & plot rare taxa}
# testing something out from online
physeq2 = filter_taxa(PS.500, function(x) mean (x) > 0.05, T)
physeq3 = transform_sample_counts(physeq2, function(x) x/sum(x))
abundance_df <- psmelt(physeq3)
glom <- tax_glom(physeq3, taxrank = 'Family')
glom_melt <- psmelt(glom)
glom_melt$Family <- as.character(glom_melt$Family)
glom_melt$Family[glom_melt$Abundance < 0.05] <- "<5% abund."
rare_taxa <- subset(glom_melt, Family == "<5% abund.")
num_unique_families <- length(unique(glom_melt$Family))
unique_families <- unique(glom_melt$Family)
num_rare <- length(unique(glom_melt$Family[glom_melt$Abundance < 0.05]))
cat("Number of unique families:", num_unique_families, "\n")
cat("Unique family names:", unique_families, "\n")
glom_melt$Family <- factor(glom_melt$Family, levels = c("Candidatus Hepatincola", "Comamonadaceae", "Flavobacteriaceae", "Aeromonadaceae", 
                                                        "Chitinibacteraceae", "Pseudomonadaceae", "Methylophilaceae", "Sporichthyaceae", "Spirosomaceae",
                                                        "Chitinophagaceae", "Weeksellaceae", "T34", "Mycobacteriaceae", "Exiguobacteraceae", 
                                                        "Moraxellaceae", "Clade III", "Lactobacillaceae", "Oxalobacteraceae", "Bacillaceae",            
                                                        "Burkholderiaceae", "Methylacidiphilaceae", "Staphylococcaceae", "Verrucomicrobiaceae",
                                                        "Enterobacteriaceae","env.OPS 17", "Crocinitomicaceae", "Listeriaceae",
                                                        "Terrimicrobiaceae", "Cyanobiaceae", "Opitutaceae", "Enterococcaceae", "Rubritaleaceae",
                                                        "Saprospiraceae", "Rhodobacteraceae", "<5% abund."))
filtered_data <- glom_melt[glom_melt$Family == "<5% abund.", ]
ggplot(filtered_data, aes(x = time_point, y = Abundance)) +
  geom_bar(stat = "identity", fill = "gray") +
  labs(title = "Rare Taxa: <5% Abundance",
       x = "Samples", y = "Relative Abundance") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
glom_plot <- ggplot(data = glom_melt, aes(x = time_point, y=Abundance, fill = Family, color = Family)) +
  geom_bar(aes(fill = Family, color = Family), stat ="identity", position ="fill") +
  scale_fill_manual(values = c('turquoise','darkseagreen2', 'tan', 'orange', 'cornflowerblue', 
                'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 
                'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3',
                "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse', 
                'seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',
                "darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro", "gold2", "chocolate", "darkgreen", "dodgerblue", "burlywood3", 'hotpink')) +
  scale_color_manual(values = c('turquoise','darkseagreen2', 'tan', 'orange', 'cornflowerblue', 
                'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 
                'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3',
                "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse', 
                'seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',
                "darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro", "gold2", "chocolate", "darkgreen", "dodgerblue", "burlywood3", 'hotpink')) +
  xlab("Sampling Time Point") + theme(legend.position="bottom") + guides(fill=guide_legend(nrow=5))
glom_plot
dev.copy(pdf, "phyloseq analysis/figures/glom_plot.pdf", height=8, width=15)
dev.off()
```


```{r stacked bar plots}
### rarefied
ASV_rel_abund500 <- transform_sample_counts(PS.500, function(x) x/sum(x))
ASV_dataframe500 <- psmelt(ASV_rel_abund500)
Abund500 <- aggregate(Abundance ~ sequence_ID + time_point + lake + organism_water + phy_group + sample_type +
                     Phylum + Class + Order + Family +
                     Genus+Species, data=ASV_dataframe500, FUN=mean)
sum.abund <- summaryBy(Abundance ~ sequence_ID*Class, data = Abund500, FUN = sum)


ASV_rel_abund_zoops <- transform_sample_counts(PS.zoops, function(x) x/sum(x))
ASV_dataframe_zoops <- psmelt(ASV_rel_abund_zoops)
#ASV_dataframe1000$Abundance <- ifelse(ASV_dataframe_zoops$Abundance < 1e-2, 0, ASV_dataframe_zoops$Abundance)
Abund_zoops <- aggregate(Abundance ~ sequence_ID + time_point + lake + organism_water + phy_group + 
                     Phylum + Class + Order + Family +
                     Genus+Species, data=ASV_dataframe_zoops, FUN=mean)


ASV_rel_abund_water <- transform_sample_counts(PS.water, function(x) x/sum(x))
ASV_dataframe_water <- psmelt(ASV_rel_abund_water)
#ASV_dataframe1000$Abundance <- ifelse(ASV_dataframe_zoops$Abundance < 1e-2, 0, ASV_dataframe_zoops$Abundance)
Abund_water <- aggregate(Abundance ~ time_point + lake + organism_water + 
                     Phylum + Class + Order + Family +
                     Genus+Species, data=ASV_dataframe_water, FUN=mean)


# testing to see if all the relative abundances of each ASV within sample 1 add up to 1
length(which(ASV_dataframe500$Sample == "322_16S"))
sample322 <- ASV_dataframe500$Sample == "322_16S"
test <- as.matrix(ASV_dataframe500[sample322, "Abundance"])
sum(test)


# plots

Abund.fa <- Abund[(Abund$Family=="Burkholderiales"),]


phy_group_order <- ggplot(Abund500, aes(x = phy_group, y = Abundance, fill= Order, color = Order)) +
  geom_bar(aes(fill = Order, color = Order), stat ="identity", position ="fill") +
  xlab("Sampling Time Point") +
  scale_fill_manual(values = c('turquoise','darkseagreen2', 'tan', 'orange', 'cornflowerblue', 
                'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 
                'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3',
                "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse', 
                'seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',
                "darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro", "gold2", "chocolate", "darkgreen", "dodgerblue", "burlywood3", 'hotpink', 'yellow3', 'blue', 'pink', 'brown3')) +
  scale_color_manual(values = c('turquoise','darkseagreen2', 'tan', 'orange', 'cornflowerblue', 
                'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 
                'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3',
                "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse', 
                'seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',
                "darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro", "gold2", "chocolate", "darkgreen", "dodgerblue", "burlywood3", 'hotpink', 'yellow3', 'blue', 'pink', 'brown3')) +
  theme_bw() + theme(axis.text.x=element_text(size=17)) +
  theme(axis.text.y=element_text(size=10)) +
  theme(strip.text.y = element_text(size = 15))

phy_group_family <- ggplot(Abund500, aes(x = time_point, y = Abundance, fill= Family, color = Family)) +
  geom_bar(aes(fill = Family, color = Family), stat ="identity", position ="fill") +
  xlab("Sampling Time Point") +
  theme_bw() +
  scale_fill_manual(values = c('turquoise','darkseagreen2', 'tan', 'orange', 'cornflowerblue', 
                'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 
                'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3',
                "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse', 
                'seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',
                "darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro", "gold2", "chocolate", "darkgreen", "dodgerblue", "burlywood3", 'hotpink', 'yellow3', 'blue', 'pink', 'brown3')) +
  scale_color_manual(values = c('turquoise','darkseagreen2', 'tan', 'orange', 'cornflowerblue', 
                'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 
                'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3',
                "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse', 
                'seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',
                "darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro", "gold2", "chocolate", "darkgreen", "dodgerblue", "burlywood3", 'hotpink', 'yellow3', 'blue', 'pink', 'brown3'))
phy_group_gen
dev.copy(pdf, "phyloseq analysis/figures/phy_group_gen.pdf", height=8, width=7)
dev.off()

Facet <- ggplot(Abund, aes(x = time_point, y = Abundance, fill= Class)) +
  geom_bar(aes(fill = Class), stat ="identity", position ="fill") +
  facet_grid(cols=vars(phy_group), rows=vars(lake)) + xlab("Sampling Time Point")
Facet
dev.copy(pdf, "phyloseq analysis/figures/Facet.pdf", height=7, width=10)
dev.off()

Facet500 <- ggplot(Abund500, aes(x = time_point, y = Abundance, fill= Class, color = Class)) +
  geom_bar(aes(fill = Class), stat ="identity", position = "fill") +
  facet_grid(rows = vars(phy_group), cols = vars(lake)) + xlab("Sampling Time Point") +
  theme_bw() + theme(axis.text.x=element_text(size=12)) +
  theme(axis.text.y=element_text(size=10)) +
  theme(strip.text.y = element_text(size = 15)) +
  scale_fill_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "seagreen4", "lightskyblue", "navy", "skyblue4", "royalblue3", "tan3", "purple3", "black", "antiquewhite2")) +
  scale_color_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "seagreen4", "lightskyblue", "navy", "skyblue4", "royalblue3", "tan3", "purple3", "black", "antiquewhite2"))
Facet500
dev.copy(pdf, "phyloseq analysis/figures/Facet500.pdf", height=6, width=10)
dev.off()

Facet_sampletype <- ggplot(Abund500, aes(x = time_point, y = Abundance, fill= Class)) +
  geom_bar(aes(fill = Class), stat ="identity", position ="fill") +
  facet_grid(cols = vars(sample_type)) + xlab("Sampling Time Point") +
  theme_bw() + theme(axis.text.x=element_text(size=17)) +
  theme(strip.text.y = element_text(size = 20)) +
  theme(axis.text.y=element_text(size=10)) +
  theme(strip.text.y = element_text(size = 10)) +
  scale_fill_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "seagreen4", "lightskyblue", "navy", "skyblue4", "royalblue3", "tan3", "purple3", "black", "antiquewhite2"))
Facet_sampletype
dev.copy(pdf, "phyloseq analysis/figures/Facet_sampletype.pdf", height=7, width=10)
dev.off()

Facet500_genus <- ggplot(Abund500, aes(x = time_point, y = Abundance, fill= Genus)) +
  geom_bar(aes(fill = Genus), stat ="identity", position ="fill") +
  xlab("Sampling Time Point") + theme_bw()
Facet500_genus
dev.copy(pdf, "phyloseq analysis/figures/Facet500_order.pdf", height=10, width=5)
dev.off()


Facet_zoops <- ggplot(Abund_zoops, aes(x = time_point, y = Abundance, fill= Class, color = Class)) +
  geom_bar(aes(fill = Class), stat ="identity", position ="fill") +
  facet_grid(rows=vars(phy_group), cols = vars(lake)) + xlab("Sampling Time Point") +
  scale_fill_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "seagreen4", "lightskyblue", "navy", "skyblue4", "purple3", "royalblue3", "tan3", "black", "antiquewhite2")) +
  scale_color_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "seagreen4", "lightskyblue", "navy", "skyblue4", "purple3", "royalblue3", "tan3", "black", "antiquewhite2"))
Facet_zoops
dev.copy(pdf, "phyloseq analysis/figures/Facet_zoops.pdf", height=5, width=10)
dev.off()

zoops_order <- ggplot(Abund_zoops, aes(x = time_point, y = Abundance, fill= Order, color = Order)) +
  geom_bar(aes(fill = Order, color = Order), stat ="identity", position ="fill") +
  facet_grid(rows = vars(phy_group)) +
  xlab("Sampling Time Point") +
  theme_bw() +
  scale_fill_manual(values = c('turquoise','darkseagreen2', 'tan', 'orange', 'cornflowerblue', 
                'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 
                'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3',
                "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse', 
                'seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',
                "darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro", "gold2", "chocolate", "darkgreen", "dodgerblue", "burlywood3", 'hotpink', 'yellow3', 'blue', 'pink', 'brown3')) +
  scale_color_manual(values = c('turquoise','darkseagreen2', 'tan', 'orange', 'cornflowerblue', 
                'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 
                'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3',
                "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse', 
                'seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',
                "darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro", "gold2", "chocolate", "darkgreen", "dodgerblue", "burlywood3", 'hotpink', 'yellow3', 'blue', 'pink', 'brown3'))

water_order <- ggplot(Abund_water, aes(x = time_point, y = Abundance, fill= Order, color = Order)) +
  geom_bar(aes(fill = Order, color = Order), stat ="identity", position ="fill") +
  xlab("Sampling Time Point") +
  theme_bw() +
  scale_fill_manual(values = c('turquoise','darkseagreen2', 'tan', 'orange', 'cornflowerblue', 
                'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 
                'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3',
                "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse', 
                'seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',
                "darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro", "gold2", "chocolate", "darkgreen", "dodgerblue", "burlywood3", 'hotpink', 'yellow3', 'blue', 'pink', 'brown3')) +
  scale_color_manual(values = c('turquoise','darkseagreen2', 'tan', 'orange', 'cornflowerblue', 
                'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 
                'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3',
                "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse', 
                'seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',
                "darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro", "gold2", "chocolate", "darkgreen", "dodgerblue", "burlywood3", 'hotpink', 'yellow3', 'blue', 'pink', 'brown3'))

Facet_zoops_general <- ggplot(Abund_zoops, aes(x = time_point, y = Abundance, fill= Class, color = Class)) +
  geom_bar(aes(fill = Class, color = Class), width = 0.89, stat ="identity", position ="fill") +
  xlab("Sampling Time Point") + theme_bw() +
  scale_fill_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "seagreen4", "lightskyblue", "navy", "skyblue4", "purple3", "royalblue3", "tan3", "black", "antiquewhite2")) +
  scale_color_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "seagreen4", "lightskyblue", "navy", "skyblue4", "purple3", "royalblue3", "tan3", "black", "antiquewhite2")) +
  ggtitle("Zooplankton microbiome compositions")
Facet_zoops_general
dev.copy(pdf, "phyloseq analysis/figures/Facet_zoops_general.pdf", height=7, width=10)
dev.off()


# by lake and time_point
stacked1 <- ggplot(data=Abund_zoops, aes(x = time_point, y = Abundance), fill = Class) +
  geom_bar(aes(fill = Class), stat ="identity", position ="fill") +
  scale_fill_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "seagreen4", "lightskyblue", "navy", "skyblue4", "purple3", "royalblue3", "tan3", "black", "antiquewhite2")) +
  facet_grid(col=vars(lake), row=vars(phy_group), scales="free") + theme(axis.text.x=element_text(size=20)) +
  theme(strip.text.x = element_text(size = 15)) + xlab("Sampling Time Point") + theme_bw() + ggtitle("Zooplankton microbiome compositions")
stacked1
dev.copy(pdf, "phyloseq analysis/figures/stacked1.pdf", height=7, width=12)
dev.off() 

# facet_grid(col=vars(time_point), scales="free")
stacked.phy_group <- ggplot(data=Abund_zoops, aes(x = time_point, y = Abundance), fill = Class, color = Class) +
  geom_bar(aes(fill = Class, color = Class), stat ="identity", position ="fill", width = 0.75) +
  facet_grid(rows = vars(phy_group)) + xlab("Sampling Time Point") +
  scale_fill_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "seagreen4", "lightskyblue", "navy", "skyblue4", "purple3", "royalblue3", "tan3", "black", "antiquewhite2")) +
  scale_color_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "seagreen4", "lightskyblue", "navy", "skyblue4", "purple3", "royalblue3", "tan3", "black", "antiquewhite2")) +
  theme(axis.text.x=element_text(size=20)) + theme(strip.text.x = element_text(size = 20)) + theme_bw() + ggtitle("Cladocera and Copepoda microbiome compositions")
stacked.phy_group
dev.copy(pdf, "phyloseq analysis/figures/stacked.phy_group.pdf", height=7, width=12)
dev.off() 
###

stacked.phy_group <- ggplot(data=Abund_zoops, aes(x = time_point, y = Abundance), fill = Class, color = Class) +
  geom_bar(aes(fill = Class, color = Class), stat ="identity", position ="fill", width = 0.75) +
  facet_grid(rows = vars(phy_group)) + xlab("Sampling Time Point") +
  scale_fill_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "seagreen4", "lightskyblue", "navy", "skyblue4", "purple3", "royalblue3", "tan3", "black", "antiquewhite2")) +
  scale_color_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "seagreen4", "lightskyblue", "navy", "skyblue4", "purple3", "royalblue3", "tan3", "black", "antiquewhite2")) +
  theme(axis.text.x=element_text(size=20)) + theme(strip.text.x = element_text(size = 20)) + theme_bw() + ggtitle("Cladocera and Copepoda microbiome compositions")
stacked.phy_group
dev.copy(pdf, "phyloseq analysis/figures/stacked.phy_group.pdf", height=7, width=12)
dev.off() 

water.only <- ggplot(Abund_water, aes(x = time_point, y = Abundance, fill = Class, color = Class)) +
  geom_bar(aes(fill = Class, color = Class), width = 0.89, stat ="identity", position ="fill") +
  xlab("Sampling Time Point") +
  scale_fill_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "olivedrab3", "seagreen4", "lightskyblue", "navy", "skyblue4", "purple3", "royalblue3", "violetred", "tan3", "darkorange4", "grey60", "black", "antiquewhite2")) +
  scale_color_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "olivedrab3", "seagreen4", "lightskyblue", "navy", "skyblue4", "purple3", "royalblue3", "violetred", "tan3", "darkorange4", "grey60", "black", "antiquewhite2")) +
  xlab("Sampling Time Point") + theme(axis.text.x=element_text(size=20)) + theme_bw() + ggtitle("Free-living microbial communities")
water.only
dev.copy(pdf, "phyloseq analysis/figures/water.only.pdf", height=6, width=12)
dev.off() 


combined.stacked.plot <- ggarrange(Facet_zoops_general + theme(axis.title.y=element_text(size=8)),
                           water.only + theme(axis.title.y=element_text(size=8)),
                           labels = c("A", "B"),
                           common.legend = TRUE, legend="bottom",
                           align="hv", font.label = list(size=10, color="black", family=NULL, position = "top"))
combined.stacked.plot
dev.print(pdf, "phyloseq analysis/figures/combined.stacked.plot.pdf")
dev.off

zoops.v.water <- ggplot(Abund500, aes(x = sample_type, y = Abundance), fill = Class, color = Class) +
  geom_bar(aes(fill = Class, color = Class), stat ="identity", position ="fill", width = 0.85) +
  scale_fill_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "seagreen4", "lightskyblue", "navy", "skyblue4", "royalblue3", "tan3", "purple3", "black", "antiquewhite2")) +
  scale_color_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "seagreen4", "lightskyblue", "navy", "skyblue4", "royalblue3", "tan3", "purple3", "black", "antiquewhite2")) +
  xlab("Sample Type") + theme(axis.text.x=element_text(size=20)) + theme(strip.text.x = element_text(size = 20)) + theme_bw() 
zoops.v.water
dev.copy(pdf, "phyloseq analysis/figures/zoops.v.water.pdf", height=5, width=5)
dev.off()


#################
# time point 1
stacked2 <- ggplot(data=Abund[Abund$time_point=="1",], aes(x = lake, y = Abundance), fill = Phylum) +
  geom_bar(aes(color = Phylum, fill = Phylum, ), stat ="identity", position ="fill") +
  theme(legend.position="bottom") + guides(fill=guide_legend(nrow=7)) + 
  facet_grid(~time_point, scales="free") + theme(axis.text.x = element_text(angle = 90))
stacked2
dev.copy(pdf, "phyloseq analysis/figures/stacked2.pdf", height=7, width=10)
dev.off() 

# by time_point for Blue
stacked3 <- ggplot(data=Abund[Abund$lake =="Blue",], aes(x = lake, y = Abundance), fill = Phylum) +
  geom_bar(aes(color = Phylum, fill = Phylum, ), stat ="identity", position ="fill") +
  facet_grid(~time_point, scales="free")
stacked3
dev.copy(pdf, "phyloseq analysis/figures/stacked3.pdf", height=7, width=10)
dev.off() 

```
