---
title: "zoops_mb_analysis"
output: html_document
date: "2023-02-13"
---
```{r loading packages}

install.packages("ggcorrplot")
install.packages("FactoMineR")
install_github("kassambara/factoextra")
remotes::install_github("microbiome/microbiome")
devtools::install_github("microbiome/mia")
BiocManager::install("DESeq2")
install.packages("corrr")
install.packages("patchwork")
install.packages("picante")



library(ggplot2)
library(tidyverse)
library(phyloseq)
library(vegan)
library(cowplot)
library(gridExtra)
library(ggpubr)
library(dplyr)
library(doBy)
library(RColorBrewer)
library(glue)
library(ecodist)
library(geosphere)
library(data.table)
library(patchwork)
library(ggpubr)
library(car)
library(ggcorrplot)
library(FactoMineR)
library(devtools)
library(factoextra)
library(knitr)
library(magrittr)
library(BiocManager)
library(decontam)
library(plyr)
library(multcompView)
library(lattice)
library(Hmisc)
library(sciplot)
library(reshape)
library(doBy)
library(car)
library(readr)
library(corrr)
library(picante)
library(factoextra)
library(microbiome)
library(mia)
library(MASS)
library(ade4)
library(fossil)

```


```{r loading in data and modifying metadata}

# load in CSV files for phyloseq object
metadata <- read.csv('phyloseq analysis/metadata_mp.csv')
ASV_table <- read.csv('phyloseq analysis/otu_table.csv')
tax_table <- read.csv('phyloseq analysis/tax_table.csv')  

# modify metadata
str(metadata)
table(metadata$organism)

# set row names
row.names(metadata) <- metadata$X

metadata$date <- as.Date(metadata$date)

# factors
metadata$sample_ID <- as.factor(metadata$sample_ID)
metadata$sequencing_ID <- as.factor(metadata$sequencing_ID)
metadata$sample_type <- as.factor(metadata$sample_type)
metadata$organism_water <- as.factor(metadata$organism_water)
metadata$time_point <- as.factor(metadata$time_point)
metadata$date <- as.factor(metadata$date)
metadata$lake <- as.factor(metadata$lake)

# numerics
metadata$number <- as.numeric(metadata$number)
metadata$latitude <- as.numeric(metadata$latitude)
metadata$longitude <- as.numeric(metadata$longitude)
metadata$elevation_m <- as.numeric(metadata$elevation_m)
metadata$temp_C <- as.numeric(metadata$temp_C)
metadata$chla_ug.L <- as.numeric(metadata$chla_ug.L)
metadata$pH <- as.numeric(metadata$pH)
metadata$DO_perc <- as.numeric(metadata$DO_perc)
metadata$spc <- as.numeric(metadata$spc)
metadata$cond <- as.numeric(metadata$cond)
metadata$DOC <- as.numeric(metadata$DOC)
metadata$TDN <- as.numeric(metadata$TDN)
metadata$TDP <- as.numeric(metadata$TDP)

#check
class(metadata$sample_ID)
class(metadata$elevation_m)

# make new column for cladocerans, copepoda, water
metadata$phy_group <- ifelse(metadata$organism == "Calanoid" | metadata$organism == "Cyclopoid" | metadata$organism =="Large Calanoid", "copepoda", ifelse(metadata$organism == "Water", "water", "cladocera"))

#make new column for elevation category
metadata$elev.cat <- as.factor(ifelse(metadata$elevation_m < 2900, "Below", "Above"))

# make a new column for basin category
metadata$basin <- as.factor(ifelse(metadata$lake == "Eastern Brook" | metadata$lake == "Serene", "South", ifelse(metadata$lake == "Cooney" | metadata$lake == "Blue" | metadata$lake == "Virginia", "North", "Middle")))

#check that levels are right
levels(as.factor(metadata$phy_group))
levels(as.factor(metadata$elev.cat))

metadata$phy_group <- as.factor(metadata$phy_group)
metadata$elev.cat <- as.factor(metadata$elev.cat)


```


```{r making phyloseq objects}

PS.fin<-
  read_phyloseq(
    otu.file = "phyloseq analysis/otu_table.csv",
    taxonomy.file = "phyloseq analysis/tax_table.csv",
    metadata.file = "phyloseq analysis/metadata_mp.csv",
    type = c("simple"),
    sep = ","
  ) 

# make sure to remove all mitochondria and chloroplast taxonomic IDs
PS.fin <- subset_taxa(PS.fin, Family!= "Mitochondria" | 
                                is.na(Family) & Class!="Chloroplast" | is.na(Class))
sample_data(PS.fin)$phy_group <- ifelse(sample_data(PS.fin)$organism_water == "Calanoid" | sample_data(PS.fin)$organism_water == "Cyclopoid" | sample_data(PS.fin)$organism_water =="Large Calanoid", "copepoda", 
                                        ifelse(sample_data(PS.fin)$organism_water == "Daphnia" | sample_data(PS.fin)$organism_water == "Ceriodaphnia" | sample_data(PS.fin)$organism_water == "Bosmina" | sample_data(PS.fin)$organism_water == "Holopedium", "cladocera", "water"))
sample_data(PS.fin)$basin <- as.factor(ifelse(sample_data(PS.fin)$lake == "Eastern Brook" | sample_data(PS.fin)$lake == "Serene" | sample_data(PS.fin)$lake == "Convict", "South", "North"))
view(sample_data(PS.fin))
str(PS.fin)
table(sample_data(PS.fin)$phy_group)
nsamples(PS.fin)
ntaxa(PS.fin)
head(taxa_sums(PS.fin))

# change sample names to match 
sample_data(PS.fin) <- metadata
sample_names(PS.fin) <- metadata$sequence_ID
rownames(metadata) <- metadata$sequence_ID


# Make a data frame with a column for the read counts of each sample
sample_data(PS.fin)$read_depth <- sample_sums(PS.fin)
sample_sum_df <- data.frame(read.sum = sample_sums(PS.fin))
View(sample_sum_df)
median(sample_sum_df$read.sum)

########### now we have final phyloseq object with correct metadata 
saveRDS(PS.fin, file="phyloseq analysis/output/PS.fin.rds")
PS.fin <- readRDS("phyloseq analysis/output/PS.fin.rds")

# make row names the sample names
colnames(sample_sum_df) <- "read.sum"
sample_sum_df$sequence_ID <- rownames(sample_sum_df)

# merge in the reads
run.metaD <- merge(metadata, sample_sum_df, by="sequence_ID", all.y=TRUE)
write.csv(run.metaD, "phyloseq analysis/output/run.metaD.final.csv")

mean(run.metaD$read.sum)
min(run.metaD$read.sum)
max(run.metaD$read.sum)
summaryBy(read.sum ~ sample_type, data = run.metaD, FUN = c(mean,sd))

```


```{r top average taxa compositions}

# finding average abundance of each class of bacteria for samples overall
glom <- tax_glom(PS.hell, taxrank = 'Class')
C.abund <- transform_sample_counts(glom, function(x) x / sum(x))
C.abund <- psmelt(C.abund)
C.abund$Class <- as.character(C.abund$Class)
avg.abund <- summaryBy(Abundance ~ Class*sample_type, data = C.abund, FUN = mean)

# finding the unique taxa names
taxa_table <- tax_table(PS.fin)
taxa_order <- taxa_table[, "Order"]
unique_order <- as.data.frame(unique(taxa_class))
unique_classes <- as.data.frame(get_taxa_unique(PS.hell, "Family"))


# another way, based on frequency not averages
all_class <- table(tax_table(PS.hell)[,"Class"])
sorted_class <- sort(all_class, decreasing = T)
top_class <- names(sorted_class)[1:5]

```


```{r total reads by species boxplots}

# Box plots showing spread of total reads by species

reads <- ggplot(sample_data(PS.fin), aes(x=organism_water, y=read_depth)) + geom_boxplot() +
  theme(axis.text.x = element_text(angle = 90)) + labs(y= "Read Depth", x = "Organism")
reads
dev.copy(pdf, "phyloseq analysis/figures/reads.sample.pdf", height=4, width=7)
dev.off() 


log.reads <- ggplot(sample_data(PS.fin), aes(x=organism_water, y=log(read_depth))) + geom_boxplot() +
  theme(axis.text.x = element_text(angle = 90)) + labs(y= "log(Read Depth)", x = "Organism")
log.reads
dev.copy(pdf, "phyloseq analysis/figures/log.reads.sample.pdf", height=4, width=7)
dev.off() 

```


```{r Read depth plots and rarefaction curves}

readcount = data.table(as(sample_data(PS.fin), "data.frame"),
                       TotalReads = sample_sums(PS.fin),
                       keep.rownames = T)
ggplot(readcount, aes(TotalReads)) + geom_histogram() + ggtitle("Sequencing Depth")


# Histogram of sample read counts
hist.depth <- ggplot(sample_sum_df, aes(x = read.sum)) + 
  geom_histogram(color = "black", fill = "gold2", binwidth = 2500) +
  ggtitle("Distribution of sample sequencing depth") + 
  xlab("Read counts") +
  theme(axis.title.y = element_blank()) + theme_classic() + geom_vline(xintercept=1000, lty=2)
hist.depth
dev.copy(pdf, "phyloseq analysis/figures/hist.depth.pdf", height=4, width=5)
dev.off() 

# a different view
read_depth_violin <- ggplot(sample_sum_df, aes(x=1, y=read.sum)) +
  geom_violin() + ggtitle("Distribution of sample sequencing depth") + scale_y_log10() +
  xlab("Samples") + ylab("Read Depth")

read_depth_violin <- read_depth_violin + geom_boxplot(width=0.1)
read_depth_violin
dev.copy(pdf, "phyloseq analysis/figures/read_depth_violin.pdf", height=4, width=5)
dev.off()

# Rarefaction curve
count_table_filt <- as.data.frame(otu_table(PS.fin))
rarecurve((count_table_filt), step=50, cex=0.5, xlim=c(0,100000), ylab = "ASVs", label=FALSE)
abline(v = 5000, lty = "dotted", col="red", lwd=2)

dev.copy(pdf, "phyloseq analysis/figures/rare.raw.pdf", height=4, width=5)
dev.off() 
```


```{r exploring what low read count samples do and what to rarefy to}

ord <- ordinate(PS.fin, method = 'PCoA', distance = 'bray')
plot_ordination(PS.fin, ord, color = "sample_type")
sample_data(PS.fin)$read_depth <- sample_sums(PS.fin)
plot_ordination(PS.fin, ord, shape = "sample_type") +
  geom_point(aes(color = read_depth > 500))

ord <- ordinate(PS.rare, method = 'PCoA', distance = 'bray')
plot_ordination(PS.rare, ord, shape = "sample_type") +
  geom_point(aes(color = read_depth))
sample_data(PS.rare)$read_depth <- sample_sums(PS.rare)

ord <- ordinate(PS.rare, method = 'PCoA', distance = 'bray')
plot_ordination(PS.rare, ord, color = "time_point") +
  geom_point() +
  stat_ellipse(level=0.9, linetype = 2)
```


```{r rarefy}

table(sample_data(PS.fin)$sample_type)

# rarefying to 500 reads (both the zoops and water)
set.seed(10000)
PS.500 <- rarefy_even_depth(PS.fin, rngseed=1e4, sample.size = 500)
sort(rowSums(otu_table(t(PS.500))))

# filtering metadata by sample names with 500 reads or more, 49 samples lost
metadata_500 <- as.data.frame(sample_data(PS.500))
metadata_500$sequence_ID <- rownames(PS.500) 
table(sample_data(PS.500)$sample_type)

# rarefying to 500 reads for just zoops
set.seed(10000)
PS.zoops <- rarefy_even_depth(PS.zoops, rngseed=1e4, sample.size = 500)
table(sample_data(PS.zoops)$phy_group, sample_data(PS.zoops)$time_point, sample_data(PS.zoops)$lake)
table(sample_data(PS.zoops)$sample_type)


```


```{r Hellinger transformation (instead of rarefy)}
# Hellinger (square root) transform the data!
PS.hell = transform_sample_counts(PS.fin, function(x) x^0.5)
set.seed(10000)

# Bray Curtis dissimilarity matrix
bc.hell <- phyloseq::distance(PS.hell, method = "bray")

# making NMDS ordination
set.seed(10000)
nmds.hell <- metaMDS(bc.hell,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)
stress_result <- nmds.hell$stress
species.scores.hell <- as.data.frame(scores(nmds.hell))
nmds.merge.hell <- merge(sample_data(PS.hell), species.scores.hell, by = 'row.names', all = TRUE)


# estimate the actual statistical significance of differences in the observed community composition between two groups of samples.
set.seed(10000)
hell.model <- adonis2(bc.hell ~ lake*time_point + phy_group, data = nmds.merge.hell, permutations = 1000)
hell.model

set.seed(10000)
hell.model.env <- adonis2(bc.hell ~ temp_C , data = nmds.merge.hell, permutations = 1000)
hell.model.env


install.packages('devtools')
library(devtools)
install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis") 
library(pairwiseAdonis)

# post hoc pairwise comparison to see what's driving the significance:
pairwise.adonis2(bc.hell ~ time_point, data = nmds.merge.hell)
pairwise.adonis2(bc.hell ~ phy_group, data = nmds.merge.hell)
pairwise.adonis2(bc.hell ~ lake, data = nmds.merge.hell)

# beta dispersion test
permutest(betadisper(bc.hell, nmds.merge.hell$lake), pairwise = T)
permutest(betadisper(bc.hell, nmds.merge.hell$phy_group), pairwise = T)
permutest(betadisper(bc.hell, nmds.merge.hell$temp_C), pairwise = T)
disp <- permutest(betadisper(bc.hell, nmds.merge.hell$time_point), pairwise = T)
# time is significant in the betadisper, sample_type and phy_group are not

# testing this anosim function out since time did not pass homogeneity test
anosim(bc.hell, nmds.merge.hell$time_point, permutations = 1000)

```


```{r alpha diversity plots - plot_richness}

#richness by Lake for zoops only
richness.plot.location <- plot_richness(PS.fin, x="lake", measures=c("Observed", "Shannon")) + 
  labs(y= "Alpha Diversity Measure", x = "Lake") + theme_bw() + geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90)) +
  theme(axis.text.x=element_text(size=9)) + theme(strip.text.x = element_text(size = 12))
richness.plot.location
dev.copy(pdf, "phyloseq analysis/figures/richness.plot.location.pdf", height=5, width=7)
dev.off()


#richness by organism (or sample type i.e, water)
richness.plot.organism <- plot_richness(PS.fin, x="sample_type", measures=c("Observed", "Shannon")) + 
  labs(y= "Alpha Diversity Measure", x = "Sample Type") + theme_bw() + geom_boxplot() +
  theme(axis.text.x=element_text(size=9)) + theme(strip.text.x = element_text(size = 12))
richness.plot.organism
dev.copy(pdf, "phyloseq analysis/figures/richness.plot.organism.pdf", height=5, width=7)
dev.off() 


#richness by phy_group (or sample type i.e, water)
richness.plot.phy <- plot_richness(PS.fin, x="phy_group", measures=c("Observed", "Shannon")) +
  labs(y= "Alpha Diversity Measure", x = "Sample Type") + theme_bw() + geom_boxplot() +
  theme(axis.text.x=element_text(size=9)) + theme(strip.text.x = element_text(size = 12))
richness.plot.phy
dev.copy(pdf, "phyloseq analysis/figures/richness.plot.phy.pdf", height=5, width=7)
dev.off() 

#richness by Time.point
richness.plot.time.point <- plot_richness(PS.fin, x="time_point", measures=c("Observed", "Shannon")) + 
  labs(y= "Alpha Diversity Measure", x = "Time Point") + theme_bw() + geom_boxplot() +
  theme(axis.text.x=element_text(size=9)) + theme(strip.text.x = element_text(size = 12))
richness.plot.time.point
dev.copy(pdf, "phyloseq analysis/figures/richness.plot.time.point.pdf", height=5, width=7)
dev.off() 

```


```{r Env. effects and averages of ASV Richness and Diversity}

######### zoops AND water
# measure richness and shannon diversity

a.diversity <- estimate_richness(PS.fin)
row.names(a.diversity) <- row.names(sample_data(PS.fin))
a.diversity <- cbind(a.diversity, sample_data(PS.fin)[,c(3,4,7,8,9,10:19,22:24)])
view(a.diversity)


# bell curve/ normally distributed, so we can use anova
hist(a.diversity$Shannon, main="Shannon diversity", xlab="", breaks=10)
shapiro.test(a.diversity$shannon)

# Testing effects on Shannon diversity index
anova(lm(Shannon ~ lake + time_point + sample_type, data = a.diversity))
shan.anova <- aov(lm(Shannon ~ lake + time_point + sample_type + phy_group + basin, data = a.diversity))
TukeyHSD(shan.anova) # shows that Serene is highly different from all the other lakes and is driving the significant difference

# Testing effects on observed richness
anova(lm(Observed ~ lake + time_point + sample_type, data = a.diversity))
rich.anova <- aov(lm(Observed ~ lake + time_point + sample_type + phy_group + basin, data = a.diversity))
TukeyHSD(rich.anova) # again, Serene is driving the significant difference

# Running multiple linear regressions of multiple environmental variables on Shannon and Observed richness
summary(lm(Shannon ~  temp_C + DOC + elevation_m + chla_ug.L + pH + DO_perc + cond, data = a.diversity))
summary(lm(Observed ~  temp_C + DOC + elevation_m + chla_ug.L + pH + DO_perc + cond, data = a.diversity))



a.div.mod <- lm(Shannon ~ temp_C + DOC + elevation_m + chla_ug.L + pH + DO_perc + cond, data = a.diversity)
a.div.AIC <- MASS::stepAIC(a.div.mod, direction = "both") # just chlorophyll


rich.mod <- lm(Observed ~ temp_C + DOC + elevation_m + chla_ug.L + pH + DO_perc + cond + basin, data = a.diversity)
rich.AIC <- MASS::stepAIC(rich.mod, direction = "both")
rich.mod <- summary(lm(Observed ~ chla_ug.L + DO_perc, data = a.diversity))


# summaryBy for averages and lm for regression model to conduct anova for significance
sam_div <- summaryBy(Shannon ~ sample_type, data = a.diversity, FUN = c(mean, se))
sam_rich <- summaryBy(Observed ~ sample_type, data = a.diversity, FUN = c(mean, se))

time_div <- summaryBy(Shannon ~ time_point, data = a.diversity, FUN = c(mean, se))
time_rich <- summaryBy(Observed ~ time_point, data = a.diversity, FUN = c(mean, se))

# plots

bargraph.CI(sample_type, shannon, data = rare.a.diversity,
                     xlab = "Sample Type", ylab = "Shannon Diversity", cex.lab = 1.5, x.leg = 2,
                     col = "black", angle = 45, cex.names = 1.25,
                     density = c(0,20), legend = TRUE)

bargraph.CI(lake, shannon, data = rare.a.diversity,
                     xlab = "Lake", ylab = "Shannon Diversity", cex.lab = 1, x.leg = 2,
                     col = "black", angle = 45, cex.names = 1,
                     density = c(0,20), legend = TRUE)

rich.bar.sample <- bargraph.CI(sample_type, observed, data = rare.a.diversity,
                     xlab = "Sample Type", ylab = "Species Richness", cex.lab = 1.5, x.leg = 2,
                     col = "gray", cex.names = 1.25, space=c(0.5), ylim = c(0,250),
                     density = c(0,20), legend = TRUE)
rich.bar.sample
dev.print(pdf, "phyloseq analysis/figures/rich.bar.sample.pdf", height=10, width=15)
dev.off


rich.bar.lake <- bargraph.CI(lake, observed, data = rare.a.diversity,
                     xlab = "Lake", ylab = "Species Richness", cex.lab = 1.5, x.leg = 2,
                     col = "gray", ylim = c(0,600),
                     density = c(0,20), legend = TRUE)
rich.bar.lake
dev.print(pdf, "phyloseq analysis/figures/rich.bar.lake.pdf", height=10, width=15)
dev.off


rare.a.div.thru.time <- ggplot(rare.sum_div, aes(x=lake, y=shannon.mean)) + geom_boxplot()
  ylim(1,5) +
  geom_point(size = 1.2) +
  xlab('Sampling Time Point') +
  ylab('Average Shannon Diversity') +
  geom_errorbar(aes(ymin=shannon.mean - shannon.sd, ymax=shannon.mean + shannon.sd), width=.05,
                 position=position_dodge(0.05)) +
  theme(strip.text.y = element_text(size = 8, hjust = 0.5,
          vjust = 0.5, face = 'bold'))
rare.a.div.thru.time
dev.copy(pdf, "phyloseq analysis/figures/rare.a.div.thru.time.pdf", height=6, width=8)
dev.off() 

rare.rich.thru.time <- ggplot(sum_rich, aes(x=lake, y=observed.mean))+
  ylim(10,150) +
  geom_point(size = 1.2) +
  geom_line(size = 0.3) +
  xlab('Sampling Time Point') +
  ylab('Average Richness') +
  geom_errorbar(aes(ymin=observed.mean - observed.sd, ymax=observed.mean + observed.sd), width=.05,
                 position=position_dodge(0.05)) +
  theme(strip.text.y = element_text(size = 8, hjust = 0.5,
          vjust = 0.5, face = 'bold')) +
  stat_compare_means(method = "anova")
rare.rich.thru.time
dev.copy(pdf, "phyloseq analysis/figures/rare.rich.thru.time.pdf", height=6, width=8)
dev.off() 

```


```{r testing effects of env. variables on rel. abund. and richness of dominant bacteria}
# Extract the abundance matrix
abundance_matrix <- as.data.frame(otu_table(PS.hell))

# Extract the metadata
metaD <- as.data.frame(sample_data(PS.hell))

# Extract the taxa table
taxa_table <- as.data.frame(tax_table(PS.hell))

# Subset the abundance matrix to include only the class of interest
class_abundance <- abundance_matrix[taxa_table$Class == "Alphaproteobacteria",]
alphaproteobacteria <- subset(taxa_table, Class == "Alphaproteobacteria")
merged_data <- merge(alphaproteobacteria, t(abundance_matrix), metaD, by = "row.names")


Bacteroidia <- subset(my_class, Class == "Bacteroidia")
Bacteroidia <- summaryBy(Abundance ~ sequence_ID, data = Bacteroidia, FUN = sum)
Bacteroidia <- data.frame(Bacteroidia, row.names = Bacteroidia[,1])
Bacteroidia <- merge(Bacteroidia, sample_data(PS.fin), by = "row.names")
view(Bacteroidia)
Anova(lm(Abundance.sum ~ lake + time_point + phy_group, data = Bacteroidia))
FitAll.bacter <- lm(Abundance.sum ~ temp_C + DO_perc + pH + DOC + elevation_m + cond + chla_ug.L, data = Bacteroidia)
step(FitAll.bacter, direction = "both")
# nothing significant

gammaproteobacteria <- subset(my_class, Class == "Gammaproteobacteria")
gammaproteobacteria <- summaryBy(Abundance ~ sequence_ID, data = gammaproteobacteria, FUN = sum)
gammaproteobacteria <- data.frame(gammaproteobacteria, row.names = gammaproteobacteria[,1])
gammaproteobacteria <- merge(gammaproteobacteria, sample_data(PS.fin), by = "row.names")
Anova(lm(Abundance.sum ~ lake + time_point + phy_group, data = gammaproteobacteria))
FitAll.gamma <- lm(Abundance.sum ~ temp_C + DO_perc + pH + DOC + elevation_m + cond + chla_ug.L, data = gammaproteobacteria)
step(FitAll.gamma, direction = "both") # chlorophyll
summary(lm(Abundance.sum ~ chla_ug.L, data = gammaproteobacteria))

alphaproteobacteria <- subset(my_class, Class == "Alphaproteobacteria")
alphaproteobacteria <- summaryBy(Abundance ~ sequence_ID, data = alphaproteobacteria, FUN = sum)
alphaproteobacteria <- data.frame(alphaproteobacteria, row.names = alphaproteobacteria[,1])
alphaproteobacteria <- merge(alphaproteobacteria, sample_data(PS.fin), by = "row.names")
Anova(lm(Abundance.sum ~ lake + time_point + phy_group, data = alphaproteobacteria))
FitAll.alpha <- lm(Abundance.sum ~ temp_C + DO_perc + pH + DOC + elevation_m + cond + chla_ug.L, data = alphaproteobacteria)
step(FitAll.alpha, direction = "both") # elevation
summary(lm(Abundance.sum ~ elevation_m, data = alphaproteobacteria)) #nah

actinobacteria <- subset(my_class, Class == "Actinobacteria")
actinobacteria <- summaryBy(Abundance ~ sequence_ID, data = actinobacteria, FUN = sum)
actinobacteria <- data.frame(actinobacteria, row.names = actinobacteria[,1])
actinobacteria <- merge(actinobacteria, sample_data(PS.fin), by = "row.names")
Anova(lm(Abundance.sum ~ lake + time_point + phy_group, data = actinobacteria))
act.aov <- aov(lm(Abundance.sum ~ lake + time_point + phy_group, data = actinobacteria))
TukeyHSD(act.aov)
# lake is important
FitAll.actino <- lm(Abundance.sum ~ temp_C + DO_perc + pH + DOC + elevation_m + cond + chla_ug.L, data = actinobacteria)
step(FitAll.actino, direction = "both") # DO
summary(lm(Abundance.sum ~ DO_perc, data = actinobacteria))


cyanobacteriia <- subset(my_class, Class == "Cyanobacteriia")
cyanobacteriia <- summaryBy(Abundance ~ sequence_ID, data = cyanobacteriia, FUN = sum)
cyanobacteriia <- data.frame(cyanobacteriia, row.names = cyanobacteriia[,1])
cyanobacteriia <- merge(cyanobacteriia, sample_data(PS.fin), by = "row.names")
Anova(lm(Abundance.sum ~ lake + time_point + phy_group, data = cyanobacteriia))
FitAll.cyano <- lm(Abundance.sum ~ temp_C + DO_perc + pH + DOC + elevation_m + cond + chla_ug.L, data = cyanobacteriia)
step(FitAll.cyano, direction = "both") # chlorophyll
summary(lm(Abundance.sum ~ chla_ug.L, data = cyanobacteriia)) # nah


```


```{r testing effects of env. on comm. comp. and plotting}

all_class <- table(tax_table(PS.zoops)[,"Class"])
sorted_class <- sort(all_class, decreasing = T)
top_class <- names(sorted_class)[1:5]

# Subset taxa table
taxa_subset <- prune_taxa(names(c("Gammaproteobacteria", "Bacteroidia", "Alphaproteobacteria", "Cyanobacteriia", "Veruucomicrobiae")), PS.zoops)

# Extract abundance data for selected taxa
taxa_abundance <- otu_table(taxa_subset)

# Calculate correlation matrix
cor_matrix <- cor(species.scores3, taxa_abundance)

```


```{r correlations}

# cor.test(metadata$elevation_m, metadata$DOC)

ggscatter(sample_data(PS.hell), x = "elevation_m", y = "DOC",
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "elevation (m)", ylab = "DOC")

#significant
ggscatter(sample_data(PS.hell), x = "elevation_m", y = "chla_ug.L", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "elevation (m)", ylab = "chla")

ggscatter(sample_data(PS.hell), x = "elevation_m", y = "temp_C", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "elevation (m)", ylab = "Temperature (C)")

ggscatter(sample_data(PS.hell), x = "latitude", y = "temp_C",
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "latitude", ylab = "Temperature")

ggscatter(sample_data(PS.hell), x = "latitude", y = "DOC", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "latitude", ylab = "DOC")

#significant
ggscatter(sample_data(PS.hell), x = "latitude", y = "chla_ug.L", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "latitude", ylab = "chla")

ggscatter(sample_data(PS.hell), x = "DOC", y = "chla_ug.L", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "DOC", ylab = "chla")

#significant
ggscatter(sample_data(PS.hell), x = "temp_C", y = "pH", color = "lake",
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "temp_C", ylab = "pH")

```


```{r environmental effects on bacterial comm. dissimilarities}

colSums(is.na(metadata))
# As stated early in the article, PCA only works with numerical values
numerical_data <- metadata[,10:19]
data_normalized <- scale(numerical_data)
head(data_normalized)
corr_matrix <- cor(data_normalized)
ggcorrplot(corr_matrix)
data.pca <- princomp(corr_matrix)
summary(data.pca)
fviz_eig(data.pca, addlabels = TRUE)
fviz_pca_var(data.pca, col.var = "black")

```


```{r Bray curtis distance matrix}

raw.asv.table <- as.data.frame(t(otu_table(PS.fin)))
str(raw.asv.table)
str(ASV_table)

# 3 methods for calculating bray curtis dissimilarity

set.seed(10000)
avg_bray_500 <- avgdist(raw.asv.table, sample = 500)

set.seed(10000)
bc_vegdist <- vegdist(t(otu_table(PS.fin)), method = "bray")

# will use these moving forward:
set.seed(10000)
bc <- phyloseq::distance(PS.fin, method = "bray")
set.seed(10000)
bc_500 <- phyloseq::distance(PS.500, method = "bray")
set.seed(10000)
bc_zoops <- phyloseq::distance(PS.zoops, method = "bray")
set.seed(10000)
bc_water <- phyloseq::distance(PS.water, method = "bray")

```


```{r metaMDS, scores, and PERMANOVAS from BC distance matrices}

set.seed(10000)
nmds1 <- metaMDS(bc,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)

species.scores1 <- as.data.frame(scores(nmds1))
nmds.merge1 <- merge(sample_data(PS.fin), species.scores1, by = 'row.names', all = TRUE)
nmds.merge1$basin <- as.factor(ifelse(nmds.merge1$lake == "Eastern Brook" | nmds.merge1$lake == "Serene" | nmds.merge1$lake == "Convict", "South", "North"))

adonis2(bc ~ lake*time_point*sample_type, data = nmds.merge1, permutations = 1000)

#################
# rarefied
set.seed(10000)
nmds2 <- metaMDS(bc_500,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)

NMDS1=nmds2$points[,1]
NMDS2=nmds2$points[,2]


species.scores2 <- as.data.frame(scores(nmds2))
nmds.merge2 <- merge(sample_data(PS.500), species.scores2, by = 'row.names', all = TRUE)
nmds.merge2$basin <- as.factor(ifelse(nmds.merge2$lake == "Eastern Brook" | nmds.merge2$lake == "Serene" | nmds.merge2$lake == "Convict", "South", "North"))

broad.model2 <- adonis2(bc_500 ~ lake*time_point*sample_type, data = nmds.merge2, permutations = 1000)
broad.model2
env.model2 <- adonis2(bc_500 ~ latitude + longitude + elevation_m + temp_C + pH + cond + DOC + chla_ug.L + DO_perc, data = nmds.merge2, permutations = 1000)
env.model2

#test for beta dispersion
anova(betadisper(bc_500, nmds.merge2$time_point)) # not significant
anova(betadisper(bc_500, nmds.merge2$sample_type)) # not significant


##################
# zoops only
set.seed(10000)
nmds3 <- metaMDS(bc_zoops,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)

species.scores3 <- as.data.frame(scores(nmds3))
nmds.merge3 <- merge(sample_data(PS.zoops), species.scores3, by = 'row.names', all = TRUE)
nmds.merge3$basin <- as.factor(ifelse(nmds.merge3$lake == "Eastern Brook" | nmds.merge3$lake == "Serene" | nmds.merge3$lake == "Convict", "South", "North"))

broad.model3 <- adonis2(bc_zoops ~ time_point + lake + phy_group, data = nmds.merge3, permutations = 1000)
broad.model3
env.model3 <- adonis2(bc_zoops ~ latitude + longitude + elevation_m + temp_C + pH + cond + DOC + chla_ug.L + DO_perc, data = nmds.merge3, permutations = 1000)
env.model3

anova(betadisper(bc_zoops, nmds.merge3$time_point)) # significant


#######################
# water only
set.seed(10000)
nmds4 <- metaMDS(bc_water,
          distance = "bray",
          k = 3,
          maxit = 999, 
          wascores = TRUE)

species.scores4 <- as.data.frame(scores(nmds4))
nmds.merge4 <- merge(sample_data(PS.water), species.scores4, by = 'row.names', all = TRUE)
nmds.merge4$region <- as.factor(ifelse(nmds.merge4$lake == "Eastern Brook" | nmds.merge4$lake == "Serene" | nmds.merge4$lake == "Convict", "South", "North"))

# something weird going on here: no p-values?
broad.model4 <- adonis2(bc_water ~ time_point*lake + region, data = nmds.merge4, permutations = 1000)
broad.model4
env.model4 <- adonis2(bc_water ~ latitude + longitude + elevation_m + temp_C + pH + cond + DOC + chla_ug.L + DO_perc, data = nmds.merge4, permutations = 1000)
env.model4

anova(betadisper(bc_water, nmds.merge4$time_point))
```


```{r combining NMDS and PCA}
#Take environmental data, run PCA to compress to a single variable (PC1), and examine
#relationship between PC1 and NMDS1, as a way to examine beta diversity x environment. 
#Output here are testing NMDS relationship against PC1 and habitat using linear models.

numerical_data <- as.matrix(sample_data(PS.hell)[,c(12:19)])
env.PCA <- prcomp(numerical_data, center = TRUE, scale= TRUE)
summary(env.PCA)
pc_scores <- scores(env.PCA)
pc_scores <- as.data.frame(pc_scores)
pc_loadings <- env.PCA$rotation
NMDSxPCA_data <- data.frame(NMDS1 = species.scores.hell$NMDS1, PC1 = pc_scores[, 1], time = sample_data(PS.hell)$time_point, lake = sample_data(PS.hell)$lake)
nmdsXpca <- ggplot(NMDSxPCA_data, aes(x = PC1, y = NMDS1, color = lake)) +
  geom_point() + stat_ellipse() +
  #geom_smooth(aes(color=lake, fill=lake), method = "lm", alpha = .2) +
  xlab("PC1 (Environment)") + ylab("NMDS1 (Bray-Curtis)") +
  theme_bw()
nmdsXpca 
dev.copy(pdf, "phyloseq analysis/figures/nmdsXpca .pdf", height=4, width=10)
dev.off() 

anova(lm(NMDS1 ~ PC1*lake, data = NMDSxPCA_data))
```


```{r final envfit}

# subset continuous environmental data
env.hell <- sample_data(PS.hell)[ ,c(10:16,18,19)]

# run the envfit function
fit.env.hell <- envfit(nmds.hell, env.hell, na.rm = T)


# colors and groups for plots

lake_colors <- c("blue","chocolate1", "purple", "hotpink", "chartreuse3",  "gold")
lake.group <- nmds.merge.hell$lake



### make plot by habitat with vectors for environment
environm_plot <- ordiplot(nmds.merge.hell[,c(27,28)], type="n", main=substitute(paste("")), cex.main=1, display="sites", xlim=c(-0.25, 0.8), cex.lab=0.8, cex.axis=0.8)
abline(h = 0, lty = "dotted")
abline(v = 0, lty = "dotted")
points(nmds.hell, "sites", cex=0.8, pch=16, col = "gray62")
ordihull(environm_plot, groups=nmds.merge.hell$lake, draw="polygon", alpha=80, col = unique(lake_colors))
legend("topright", legend=levels(lake.group), cex=1, pch=16, col=unique(lake_colors), pt.cex=1, bty="n")
par.new=T
plot(fit.env.hell, col="black", p.max=0.05, cex=0.9, lwd=1)
dev.copy(pdf, "phyloseq analysis/figures/environm_plot.pdf", height=5.5, width=10)
dev.off() 



```


```{r envfit attempts w/ ggplot}

#Fit environmental vectors to ordination to see which environmental variables are correlated with the ordination

#environmental variables, continuous
env500 <- sample_data(PS.500)[ ,c(10,11,14,16,18)]
envzoops <- sample_data(PS.zoops)[ ,c(1, 10:19)]
envwater <- sample_data(PS.water)[ ,c(10:19)]
env.hell <- sample_data(PS.hell)[ ,c(10:19)]

# factors, categorical
facs500 <- sample_data(PS.500)[,c(7,9)]
facszoops <- sample_data(PS.zoops)[,c(7,9)]

# run envfit functions
fit.env <- envfit(nmds1, env, na.rm = T)
fit.env500 <- envfit(nmds2, env500, na.rm = T)
fit.envzoops <- envfit(nmds3, envzoops, na.rm = T)
fit.envwater <- envfit(nmds4, envwater, na.rm = T)
fit.env.hell <- envfit(nmds.hell, env.hell, na.rm = T)

NMDS.plot.df=data.frame(NMDS1=NMDS1,NMDS2=NMDS2, metadata_500[,6])
times <- NMDS.plot.df$time_point
levels(NMDS.plot.df$time_point)



data.scores <- as.data.frame(scores(nmds1))
data.scores.df <- cbind(facs, data.scores)
data.scores500 <- as.data.frame(scores(nmds2))
data.scores.df500 <- cbind(facs500, data.scores500)


en_coord_cont = as.data.frame(scores(fit.env, "vectors")) * ordiArrowMul(fit.env)
en_coord_cat = as.data.frame(scores(fit.env, "factors")) * ordiArrowMul(fit.env)
en_coord_cont500 = as.data.frame(scores(fit.env500, "vectors")) * ordiArrowMul(fit.env500)
en_coord_cat500 = as.data.frame(scores(fit.env500, "factors")) * ordiArrowMul(fit.env500)


envfit_plot = ggplot(data = data.scores.df, aes(x = NMDS1, y = NMDS2)) + 
     geom_point(data = data.scores.df, aes(colour = lake), size = 3, alpha = 0.5) + 
     scale_colour_manual(values = c("orange", "steelblue", "maroon", "mediumseagreen", "coral", "pink"))  + 
     geom_segment(aes(x = 0, y = 0, xend = NMDS1, yend = NMDS2), 
       data = en_coord_cont, size = 1, arrow = arrow(length = unit(0.25, "cm")),
       alpha = 0.5, colour = "black") +
     geom_point(data = en_coord_cat, aes(x = NMDS1, y = NMDS2), 
       shape = "diamond", size = 4, alpha = 0.6, colour = "navy") +
     geom_text(data = en_coord_cat, aes(x = NMDS1, y = NMDS2+0.04), 
       label = row.names(en_coord_cat), colour = "navy", fontface = "bold") + 
     geom_text(data = en_coord_cont, aes(x = NMDS1, y = NMDS2), colour = "black", 
       fontface = "bold", label = row.names(en_coord_cont)) + 
     stat_ellipse(aes(colour = lake)) +
     theme(axis.title = element_text(size = 10, face = "bold", colour = "black"), 
       panel.background = element_blank(), panel.border = element_rect(fill = NA, colour = "black"), 
       axis.ticks = element_blank(), axis.text = element_blank(), legend.key = element_blank(), 
       legend.title = element_text(size = 10, face = "bold", colour = "black"), 
       legend.text = element_text(size = 9, colour = "black")) + 
     labs(colour = "lake")
     
envfit_plot
dev.copy(pdf, "phyloseq analysis/figures/envfit_plot.pdf", height=4, width=5)
dev.off() 


env_vect <- ggplot(nmds.merge2)

envfit_plot500 = ggplot(data = data.scores.df500, aes(x = NMDS1, y = NMDS2)) + 
     geom_point(data = data.scores.df500, aes(colour = lake), size = 3, alpha = 0.5) + 
     scale_colour_manual(values = c("orange", "steelblue", "maroon", "mediumseagreen", "coral", "pink"))  + 
     geom_segment(aes(x = 0, y = 0, xend = NMDS1, yend = NMDS2), 
       data = en_coord_cont500, size = 1, arrow = arrow(length = unit(0.25, "cm")),
       alpha = 0.5, colour = "black") +
     geom_point(data = en_coord_cat500, aes(x = NMDS1, y = NMDS2), 
       shape = "diamond", size = 4, alpha = 0.6, colour = "navy") +
     geom_text(data = en_coord_cat500, aes(x = NMDS1, y = NMDS2+0.04), 
       label = row.names(en_coord_cat500), colour = "navy", fontface = "bold") + 
     geom_text(data = en_coord_cont500, aes(x = NMDS1, y = NMDS2), colour = "black", 
       fontface = "bold", label = row.names(en_coord_cont500)) + 
     stat_ellipse(aes(colour = lake)) +
     theme(axis.title = element_text(size = 10, face = "bold", colour = "black"), 
       panel.background = element_blank(), panel.border = element_rect(fill = NA, colour = "black"), 
       axis.ticks = element_blank(), axis.text = element_blank(), legend.key = element_blank(), 
       legend.title = element_text(size = 10, face = "bold", colour = "black"), 
       legend.text = element_text(size = 9, colour = "black")) + 
     labs(colour = "lake")
     
envfit_plot500
dev.copy(pdf, "phyloseq analysis/figures/envfit_plot500.pdf", height=4, width=5)
dev.off() 

```


```{r map}
gps <- as.data.frame(cbind(unique(sample_data(PS.hell)$longitude), unique(sample_data(PS.hell)$latitude)))
rownames(gps) <- unique(sample_data(PS.hell)$lake)


install.packages("leaflet")
library(leaflet)
map <- leaflet() %>%
  setView(lng = -119, lat = 38.25, zoom = 10) %>%
  addTiles() %>%
  fitBounds(-118.75, 37.42, -119, 38.06) %>%
  addMarkers(data = gps, lng = ~V1, lat = ~V2, popup = ~row.names(gps)) %>%
  addMarkers(
    lng = -118.7426, lat = 37.43153,
    label = "Eastern Brook (3,155 m)",
    labelOptions = labelOptions(noHide = T, direction = "left",
      style = list(
        "color" = "hotpink",
        "font-family" = "serif",
        "box-shadow" = "3px 3px rgba(0,0,0,0.25)",
        "font-size" = "12px",
        "border-color" = "rgba(0,0,0,0.5)"))) %>%
  addMarkers(
    lng = -118.7444, lat = 37.43839,
    label = "Serene (3,124 m)",
    labelOptions = labelOptions(noHide = T, direction = "right",
      style = list(
        "color" = "green",
        "font-family" = "sanserif",
        "font-style" = "bold",
        "box-shadow" = "3px 3px rgba(0,0,0,0.25)",
        "font-size" = "12px",
        "border-color" = "rgba(0,0,0,0.5)"))) %>%
  addMarkers(
    lng = -118.8574, lat = 37.59009,
    label = "Convict (2,315 m)",
    labelOptions = labelOptions(noHide = T, direction = "right",
      style = list(
        "color" = "orange",
        "font-family" = "serif",
        "font-style" = "bold",
        "box-shadow" = "3px 3px rgba(0,0,0,0.25)",
        "font-size" = "12px",
        "border-color" = "rgba(0,0,0,0.5)"))) %>%
   addMarkers(
    lng = -119.2651, lat = 38.04698,
    label = "Virginia (2,983 m)",
    labelOptions = labelOptions(noHide = T, direction = "bottom",
      style = list(
        "color" = "gold",
        "font-family" = "serif",
        "font-style" = "bold",
        "box-shadow" = "3px 3px rgba(0,0,0,0.25)",
        "font-size" = "12px",
        "border-color" = "rgba(0,0,0,0.5)"))) %>%
  addMarkers(
    lng = -119.2703, lat = 38.05095,
    label = "Blue (3,005 m)",
    labelOptions = labelOptions(noHide = T, direction = "right",
      style = list(
        "color" = "blue",
        "font-family" = "serif",
        "font-style" = "bold",
        "box-shadow" = "3px 3px rgba(0,0,0,0.25)",
        "font-size" = "12px",
        "border-color" = "rgba(0,0,0,0.5)"))) %>%
  addMarkers(
    lng = -119.2770, lat = 38.04806,
    label = "Cooney (3,116 m)",
    labelOptions = labelOptions(noHide = T, direction = "left",
      style = list(
        "color" = "purple",
        "font-family" = "serif",
        "font-style" = "bold",
        "box-shadow" = "3px 3px rgba(0,0,0,0.25)",
        "font-size" = "12px",
        "border-color" = "rgba(0,0,0,0.5)")))
map


```


```{r mantel test, geographic distance}
# to calculate geographic distances between lakes
# geographic distance dissim matrix - adapted from weinstein et al 
gps <- as.data.frame(cbind(sample_data(PS.hell)$longitude, sample_data(PS.hell)$latitude))
spatial_dist <- distm(gps, fun = distGeo)
rownames(spatial_dist) <- sample_data(PS.hell)$lake
colnames(spatial_dist) <- sample_data(PS.hell)$lake

# converting distances to km 
spatial_dist_km <- spatial_dist/1000
dim(spatial_dist_km)
spatial_dist <- as.dist(spatial_dist)
spatial_dist_km <- as.dist(spatial_dist_km) # make distance matrix

# run the Mantel test
vegan::mantel(spatial_dist_km, bc.hell)
# observation value in the output is the observed correlation (r) and shows sign (pos or neg corr.)

# plotting
dist.df <- as.data.frame(cbind(spatial_dist_km, bc.hell))

dist_plot <- ggplot(dist.df, aes(x=spatial_dist_km, y=bc.hell)) +
  geom_point(size=0.8,alpha=0.5) +
  geom_smooth(method = "glm", formula = y~x,
              method.args = list(family = gaussian(link = 'log')), lwd=0.7, se = T)+ 
  ggtitle("Bacterial Community Distances")+ xlab("Geographic distance (km) between lakes") + 
  ylab("Bray Curtis dissimilarities")
dist_plot
dev.copy(pdf, "phyloseq analysis/figures/dist_plot.pdf", height=4, width=5)
dev.off() 



# Attempting to also run a Procrustes test
norm_distances <- decostand(spatial_dist_km, method = "normalize")
norm_dissimilarities <- decostand(bc.hell, method = "normalize")
std_distances <- decostand(norm_distances, method = "standardize")
std_dissimilarities <- decostand(norm_dissimilarities, method = "standardize")
# Perform Procrustes analysis
procrustes_results <- protest(std_distances, std_dissimilarities)
# Extract transformed coordinates
transformed_coordinates <- procrustes_results$points


```


```{r MDS plots by time point}
# Time 1
T1 <- subset_samples(PS.hell, time_point=="1")
ORD.T1 <- ordinate(T1, method='MDS', distance='bray')

MDS.ord.T1 <- plot_ordination(
  physeq = T1,                                                   
  ordination = ORD.T1) +                                                
  geom_point(aes(color = lake, shape= phy_group), size = 3) +    
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  scale_color_brewer(palette = "Dark2") +
  scale_fill_brewer(palette = "Dark2") + ggtitle("Time1") + theme_classic()
MDS.ord.T1
dev.copy(pdf, "phyloseq analysis/figures/MDS.ord.T1.pdf", height=6, width=7)
dev.off() 

# Time 2
T2<- subset_samples(PS.hell, time_point=="2")
ORD.T2 <- ordinate(T2, method='MDS', distance='bray')
MDS.ord.T2<-plot_ordination(
  physeq = T2,                                                   
  ordination = ORD.T2) +                                                
  geom_point(aes(color = lake, shape= phy_group), size = 3) +    
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  scale_color_brewer(palette = "Dark2") +
  scale_fill_brewer(palette = "Dark2") + ggtitle("Time2") +
  theme_classic()     
MDS.ord.T2
dev.copy(pdf, "phyloseq analysis/figures/MDS.ord.T2.pdf", height=6, width=7)
dev.off() 

# Time 3
T3<- subset_samples(PS.hell, time_point=="3")
ORD.T3 <- ordinate(T3, method='MDS', distance='bray')
MDS.ord.T3<-plot_ordination(
  physeq = T3,                                                   
  ordination = ORD.T3) +                                                
  geom_point(aes(color = lake, shape= phy_group), size = 3) +    
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  scale_color_brewer(palette = "Dark2") +
  scale_fill_brewer(palette = "Dark2") + ggtitle("Time3") +
  theme_classic()     
MDS.ord.T3
dev.copy(pdf, "phyloseq analysis/figures/MDS.ord.T3.pdf", height=6, width=7)
dev.off() 

# Time 4
T4<- subset_samples(PS.hell, time_point=="4")
ORD.T4 <- ordinate(T4, method='MDS', distance='bray')
MDS.ord.T4<-plot_ordination(
  physeq = T4,                                                   
  ordination = ORD.T4) +                                                
  geom_point(aes(color = lake, shape= phy_group), size = 3) +    
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  scale_color_brewer(palette = "Dark2") +
  scale_fill_brewer(palette = "Dark2") + ggtitle("Time4") +
  theme_classic()     
MDS.ord.T4
dev.copy(pdf, "phyloseq analysis/figures/MDS.ord.T4.pdf", height=6, width=7)
dev.off() 

# Time 5
T5 <- subset_samples(PS.hell, time_point=="5")
ORD.T5 <- ordinate(T5, method='MDS', distance='bray')
MDS.ord.T5 <- plot_ordination(
  physeq = T5,                                                   
  ordination = ORD.T5) +                                                
  geom_point(aes(color = lake, shape= phy_group), size = 3) +    
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  scale_color_brewer(palette = "Dark2") +
  scale_fill_brewer(palette = "Dark2") + ggtitle("Time5") +
  theme_classic()     
MDS.ord.T5
dev.copy(pdf, "phyloseq analysis/figures/MDS.ord.T5.pdf", height=6, width=7)
dev.off() 


# now put them all together
combined.mds <- ggarrange(MDS.ord.T1 + theme(axis.title.y=element_text(size=8)),
          MDS.ord.T2 + theme(axis.title.y=element_text(size=8)),
          MDS.ord.T3 + theme(axis.title.y=element_text(size=8)),
          MDS.ord.T4 + theme(axis.title.y=element_text(size=8)),
          MDS.ord.T5 + theme(axis.title.y=element_text(size=8)),
          plot_layout(guides="collect"), labels=NULL, common.legend = TRUE, 
          legend="bottom", align="hv", 
          font.label = list(size=10, color="black", family=NULL, position="top", labels="auto"))
combined.mds
dev.print(pdf, "phyloseq analysis/figures/combined.mds.pdf", height=10, width=15)
dev.off
 
```


```{r NMDS plots broken down by lakes and times}
centroid <- nmds.merge2 %>% 
  group_by(time_point) %>%
  summarize(NMDS1 = mean(NMDS1), NMDS2 = mean(NMDS2))
str(nmds2.merge)



nmds.convict <- ggplot(nmds.merge.hell[nmds.merge.hell$lake == "Convict", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point), size = 2.5) +
  stat_ellipse(level=0.9, lwd = 0.8, linetype = 2, aes(color = time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) +
  theme_bw() + ggtitle("Convict")
nmds.convict
dev.print(pdf, "phyloseq analysis/figures/nmds.convict.pdf", height=10, width=15)
dev.off

nmds.east.brook <- ggplot(nmds.merge.hell[nmds.merge.hell$lake == "Eastern Brook", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point), size = 2.5) +
  stat_ellipse(level=0.9, lwd = 0.8, linetype = 2, aes(color = time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) +
  theme_bw() + ggtitle("Eastern Brook")
nmds.east.brook
dev.print(pdf, "phyloseq analysis/figures/nmds.east.brook.pdf", height=10, width=15)
dev.off

nmds.serene <- ggplot(nmds.merge.hell[nmds.merge.hell$lake == "Serene", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point), size = 2.5) +
  stat_ellipse(level=0.9, lwd = 0.8, linetype = 2, aes(color = time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) +
  theme_bw() + ggtitle("Serene")
nmds.serene
dev.print(pdf, "phyloseq analysis/figures/nmds.serene.pdf", height=10, width=15)
dev.off

nmds.cooney <- ggplot(nmds.merge.hell[nmds.merge.hell$lake == "Cooney", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point), size = 2.5) +
  stat_ellipse(level=0.9, lwd = 0.8, linetype = 2, aes(color = time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) +
  theme_bw() + ggtitle("Cooney")
nmds.cooney
dev.print(pdf, "phyloseq analysis/figures/nmds.cooney.pdf", height=10, width=15)
dev.off

nmds.blue <- ggplot(nmds.merge.hell[nmds.merge.hell$lake == "Blue", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point), size = 2.5) +
  stat_ellipse(level=0.9, lwd = 0.8, linetype = 2, aes(color = time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) +
  theme_bw() + ggtitle("Blue")
nmds.blue
dev.print(pdf, "phyloseq analysis/figures/nmds.blue.pdf", height=10, width=15)
dev.off

nmds.virginia <- ggplot(nmds.merge.hell[nmds.merge.hell$lake == "Virginia", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point), size = 2.5) +
  stat_ellipse(level=0.9, lwd = 0.8, linetype = 2, aes(color = time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) +
  theme_bw() + ggtitle("Virginia")
nmds.virginia
dev.print(pdf, "phyloseq analysis/figures/nmds.virginia.pdf", height=10, width=15)
dev.off

combined.nmds.lakes <- ggarrange(nmds.convict + theme(axis.title.y=element_text(size=8)),
          nmds.east.brook + theme(axis.title.y=element_text(size=8)),
          nmds.serene + theme(axis.title.y=element_text(size=8)),
          nmds.cooney + theme(axis.title.y=element_text(size=8)),
          nmds.blue + theme(axis.title.y=element_text(size=8)),
          nmds.virginia + theme(axis.title.y=element_text(size=8)),
          plot_layout(guides="collect"), labels=NULL, common.legend = TRUE, 
          legend="bottom", align="hv", 
          font.label = list(size=10, color="black", family=NULL, position="top", labels="auto"))
combined.nmds.lakes
dev.print(pdf, "phyloseq analysis/figures/combined.nmds.lakes.pdf", height=10, width=15)
dev.off

nmds.cladocera.time <- ggplot(nmds.merge.hell[nmds.merge.hell$phy_group == "cladocera", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point), size = 4) +
  stat_ellipse(level=0.9, lwd = 1, linetype = 2, aes(color = time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) + 
  theme(axis.text.x= element_text(size = 15)) +
  theme(axis.text.y= element_text(size = 15)) +
  theme_bw() +
  theme(legend.key.size = unit(1, 'cm'), #change legend key size
        legend.key.height = unit(1, 'cm'), #change legend key height
        legend.key.width = unit(1, 'cm'), #change legend key width
        legend.title = element_text(size=20), #change legend title font size
        legend.text = element_text(size=18)) +#change legend text font size 
  ggtitle("Cladocera")
nmds.cladocera.time
dev.print(pdf, "phyloseq analysis/figures/nmds.cladocera.time.pdf", height=10, width=15)
dev.off

nmds.copepoda.time <- ggplot(nmds.merge.hell[nmds.merge.hell$phy_group == "copepoda", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point), size = 4) +
  stat_ellipse(level=0.9, lwd= 1, linetype = 2, aes(color = time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) + 
  theme_bw() +
  theme(legend.key.size = unit(1, 'cm'), #change legend key size
        legend.key.height = unit(1, 'cm'), #change legend key height
        legend.key.width = unit(1, 'cm'), #change legend key width
        legend.title = element_text(size=20), #change legend title font size
        legend.text = element_text(size=18)) + #change legend text font size
  ggtitle("Copepoda")
nmds.copepoda.time
dev.print(pdf, "phyloseq analysis/figures/nmds.copepoda.time.pdf", height=10, width=15)
dev.off

nmds.water.time <- ggplot(nmds.merge.hell[nmds.merge.hell$sample_type == "Water", ], aes(x=NMDS1, y=NMDS2)) +
  geom_point(aes(color = time_point), size = 3) +
  stat_ellipse(level=0.9, lwd =1, linetype = 2, aes(color = time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) + 
  theme(legend.key.size = unit(1, 'cm'), #change legend key size
        legend.key.height = unit(1, 'cm'), #change legend key height
        legend.key.width = unit(1, 'cm'), #change legend key width
        legend.title = element_text(size=20), #change legend title font size
        legend.text = element_text(size=18)) + #change legend text font size
  theme_bw() +
  ggtitle("Water")
nmds.water.time
dev.print(pdf, "phyloseq analysis/figures/nmds.water.time.pdf", height=10, width=15)
dev.off

nmds.zoops.time.basin <- ggplot(nmds.merge3, aes(x=NMDS1, y=NMDS2, color = time_point)) +
  geom_point(aes(color = basin)) +
  stat_ellipse(level=0.9, linetype = 2) +
  theme_bw() +
  ggtitle("Zoops")
nmds.zoops.time.basin
  

```


```{r General NMDS}

nmds.time <- ggplot(nmds.merge.hell, aes(x=NMDS1, y=NMDS2, color = time_point)) +
  geom_point(aes(color = time_point), size = 2.3) + 
  stat_ellipse(level=0.9, lwd = 1, linetype = 2, aes(color=time_point)) +
  scale_color_manual(values = c("orangered", "turquoise2", "navy", "violetred", "lightslateblue")) + 
  theme_bw() +
  theme(legend.key.size = unit(1, 'cm'), #change legend key size
        legend.key.height = unit(1, 'cm'), #change legend key height
        legend.key.width = unit(1, 'cm'), #change legend key width
        legend.title = element_text(size=20), #change legend title font size
        legend.text = element_text(size=18)) #change legend text font size
nmds.time
dev.print(pdf, "phyloseq analysis/figures/nmds.time.pdf", height=10, width=15)
dev.off

nmds.lake <- ggplot(nmds.merge.hell, aes(x=NMDS1, y=NMDS2, color = lake)) +
  geom_point(aes(color = lake), size = 3) + 
  stat_ellipse(level=0.9, lwd = 1, linetype = 2, aes(color=lake)) +
  scale_color_manual(values = c("blue","chocolate1", "purple", "hotpink", "chartreuse3",  "gold2")) +
  theme_bw() +
    theme(legend.key.size = unit(1, 'cm'), #change legend key size
        legend.key.height = unit(1, 'cm'), #change legend key height
        legend.key.width = unit(1, 'cm'), #change legend key width
        legend.title = element_text(size=20), #change legend title font size
        legend.text = element_text(size=18)) #change legend text font size
nmds.lake
dev.print(pdf, "phyloseq analysis/figures/nmds.lake.pdf", height=10, width=15)
dev.off

nmds.basin <- ggplot(nmds.merge.hell, aes(x=NMDS1, y=NMDS2, color = basin)) +
  geom_point(aes(color = basin), size = 3) + 
  stat_ellipse(level=0.9, linetype = 2, aes(color=basin)) +
  theme_bw()
nmds.basin
dev.print(pdf, "phyloseq analysis/figures/nmds.basin.pdf", height=10, width=15)
dev.off

nmds.elevation <- ggplot(nmds.merge.hell, aes(x=NMDS1, y=NMDS2, color = elev.cat)) +
  geom_point(aes(color = elev.cat), size = 3) + 
  stat_ellipse(level=0.9, linetype = 2, aes(color=elev.cat)) +
  theme_bw()
nmds.elevation
dev.print(pdf, "phyloseq analysis/figures/nmds.elevation.pdf", height=10, width=15)
dev.off
  
nmds.sample_type <- ggplot(nmds.merge.hell, aes(x=NMDS1, y=NMDS2, color = sample_type)) +
  geom_point(aes(color = sample_type), size = 3) + 
  stat_ellipse(level=0.9, linetype = 2, aes(color=sample_type)) +
  theme_bw()
nmds.sample_type
dev.print(pdf, "phyloseq analysis/figures/nmds.sample_type.pdf", height=10, width=15)
dev.off

nmds.phy_group <- ggplot(nmds.merge.hell, aes(x=NMDS1, y=NMDS2, color = phy_group)) +
  geom_point(aes(color = phy_group), size = 3) + 
  stat_ellipse(level=0.9, lwd = 1, linetype = 2, aes(color=phy_group)) +
  theme_bw() +
  scale_color_manual(values = c("orangered", "violet", "royalblue3")) + 
  theme(legend.key.size = unit(1, 'cm'), #change legend key size
        legend.key.height = unit(1, 'cm'), #change legend key height
        legend.key.width = unit(1, 'cm'), #change legend key width
        legend.title = element_text(size=20), #change legend title font size
        legend.text = element_text(size=18)) #change legend text font size
nmds.phy_group
dev.print(pdf, "phyloseq analysis/figures/nmds.phy_group.pdf", height=10, width=15)
dev.off

nmds.phy_group <- ggplot(nmds.merge.hell, aes(x=NMDS1, y=NMDS2, color = chla_ug.L)) +
  geom_point(size = 3) + 
  stat_ellipse(aes(color = chla_ug.L), level=0.9, lwd = 1, linetype = 2) +
  theme_bw() +
  theme(legend.key.size = unit(1, 'cm'), #change legend key size
        legend.key.height = unit(1, 'cm'), #change legend key height
        legend.key.width = unit(1, 'cm'), #change legend key width
        legend.title = element_text(size=20), #change legend title font size
        legend.text = element_text(size=18)) #change legend text font size
nmds.phy_group
dev.print(pdf, "phyloseq analysis/figures/nmds.phy_group.pdf", height=10, width=15)
dev.off

```


```{r capscale and dbRDA}
# Trying to find environmental variables that signiificantly explained variation in the 
# 16S community composition

# Spatial variables were calculated using trend-surface analysis
# calculated second-order orthogonal polynomial terms (latitude, longitude, latitude  × longitude, 
# latitude2, longitude2) from UTM coordinates for  each lake using the “poly” function in R.
library(sp)
install.packages('rgdal')
library(rgdal)
gps <- as.data.frame(cbind(sample_data(PS.500)$longitude, sample_data(PS.500)$latitude))
coordinates(gps) <- c("V1", "V2")
# Define the coordinate reference system (CRS) for the latitude and longitude data
proj4string(gps) <- CRS("+proj=longlat +datum=WGS84")
# Convert latitude and longitude to UTM coordinates
utm_coords <- spTransform(gps, CRS("+proj=utm +zone=11 +datum=WGS84"))
# Extract the UTM coordinates from the transformed object
utm_easting <- utm_coords@coords[, 1]
utm_northing <- utm_coords@coords[, 2]
# Create a data frame for the UTM coordinates
coord_df <- data.frame(cbind(utm_easting, utm_northing))
# Calculate second-order orthogonal polynomial terms for latitude and longitude
poly_terms <- poly(coord_df$utm_easting, coord_df$utm_northing, degree = 2)
# Extract the polynomial terms from the matrix
coord_df$latitude_2 <- poly_terms[, 1]
coord_df$longitude_2 <- poly_terms[, 2]
coord_df$latitude_longitude <- poly_terms[, 3]
coord_df$latitude_squared <- poly_terms[, 4]
coord_df$longitude_squared <- poly_terms[, 5]


# CAPSCALE
set.seed(10000)
capscale_result <- capscale(bc.hell ~ cond + chla_ug.L + DO_perc + pH + temp_C + DOC + elevation_m, dfun = "distance", data = nmds.merge.hell, distance = "bray")
capscale_model <- capscale(bc.hell ~ 1, data = nmds.merge.hell, distance = "bray")
# fitting global model, then performing stepwise selection with ordiRstep
mod <- ordiR2step(capscale_model, scope = capscale_result)
Anova(mod) # no matter the order of the explanatory variables in the original formula, the output of this shows that temp. is most important
summary(mod)
plot(mod)

# most parsimonious dbRDA model
set.seed(10000)
capscale_model <- capscale(bc.hell ~ temp_C, data = nmds.merge.hell, dfun = "distance", distance = "bray")
capscale_model_an <- Anova(capscale_model)
plot(capscale_model)


mod_scores <- as.data.frame(scores(mod, choices = c(1,2), display = "sites"))
merge.dbrda <- merge(mod_scores, nmds.merge.hell, by = "row.names")



cap_scores <- as.data.frame(scores(capscale_model, display = "sites"))
nmds.merge.hell$CAP1 <- cap_scores$CAP1
nmds.merge.hell$MDS1 <- cap_scores$MDS1
CAPxMDS_data <- data.frame(CAP1 = cap_scores$CAP1, MDS1 = cap_scores$MDS1, time = sample_data(PS.hell)$time_point, lake = sample_data(PS.hell)$lake, temp = sample_data(PS.hell)$temp_C)
CAPxMDS <- ggplot(CAPxMDS_data, aes(x = CAP1, y = MDS1, color = lake)) +
  geom_point() + stat_ellipse() +
  #geom_smooth(aes(color=lake, fill=lake), method = "lm", alpha = .2) +
  xlab("CAP1 (Environment)") + ylab("MDS1 (Bray-Curtis)") +
  theme_bw()
CAPxMDS

ordiplot(CAPxMDS_data[,c(1:2)], type="n", main=substitute(paste("")), cex.main=1, display="sites", xlim=c(-0.25, 0.8), cex.lab=0.8, cex.axis=0.8)
abline(h = 0, lty = "dotted")
abline(v = 0, lty = "dotted")
points(capscale_model, "sites", cex=0.8, pch=16, col = "gray62")
ordihull(environm_plot, groups=nmds.merge.hell$lake, draw="polygon", alpha=80, col = unique(lake_colors))
legend("topright", legend=levels(lake.group), cex=1, pch=16, col=unique(lake_colors), pt.cex=1, bty="n")
par.new=T




#dbRDA
set.seed(10000) 
upr <- dbrda(bc.hell ~ temp_C + elevation_m + cond + chla_ug.L + DOC + DO_perc + pH, data = nmds.merge.hell)
lwr <- dbrda(bc.hell ~ 1, data = nmds.merge.hell, na.action = 'na.omit')
# fitting global model, then performing stepwise selection with ordiRstep
mod_selec <- ordiR2step(lwr, scope = upr)
anova(mod_selec)
anova(mod_selec, by = "axis", permu = 200)
summary(mod_selec)
RsquareAdj(mod_selec)
screeplot(mod_selec)


dbRDA <- dbrda(formula = bc.hell ~ DOC, data = nmds.merge3, na.action = "na.omit")
plot(dbRDA)
```


```{r bioenv}

bioenv(bc.hell, env.hell)

```


```{r bar plot for rare (<0.01 species)}

ASV_rel_abund <- transform_sample_counts(PS.hell, function(x) x/sum(x))
ASV_dataframe <- psmelt(ASV_rel_abund)
rare_subset <- subset(ASV_dataframe, ASV_dataframe$Abundance < 0.01)

Abund_rare <- aggregate(Abundance ~ sequence_ID + time_point + lake + organism_water + phy_group + 
                     Phylum + Class + Order + Family +
                     Genus + Species, data=rare_subset, FUN=mean)

rare_genus <- ggplot(Abund.hell, aes(x = phy_group, y = Abundance, fill= Genus, color = Genus)) +
  geom_bar(aes(fill = Genus, color = Genus), stat ="identity", position ="fill") +
  xlab("Sampling Time Point") +
  theme_bw() + theme(axis.text.x=element_text(size=17)) +
  theme(axis.text.y=element_text(size=10)) +
  theme(strip.text.y = element_text(size = 15)) 
rare_genus
dev.copy(pdf, "phyloseq analysis/figures/rare_genus.pdf", height=8, width=7)
dev.off()

rare_class <- ggplot(Abund_rare, aes(x = phy_group, y = Abundance, fill= Class, color = Class)) +
  geom_bar(aes(fill = Class, color = Class), stat ="identity", position ="fill") +
  xlab("Sampling Time Point") +
  theme_bw() + theme(axis.text.x=element_text(size=17)) +
  theme(axis.text.y=element_text(size=10)) +
  theme(strip.text.y = element_text(size = 15)) 
rare_class


```


```{r another way to categorize & plot rare taxa as a category with the rest}
# testing something out from online
physeq2 = filter_taxa(PS.500, function(x) mean (x) > 0.05, T)
physeq3 = transform_sample_counts(physeq2, function(x) x/sum(x))
abundance_df <- psmelt(physeq3)
glom <- tax_glom(physeq3, taxrank = 'Family')
glom_melt <- psmelt(glom)
glom_melt$Family <- as.character(glom_melt$Family)
glom_melt$Family[glom_melt$Abundance < 0.05] <- "<5% abund."
rare_taxa <- subset(glom_melt, Family == "<5% abund.")
num_unique_families <- length(unique(glom_melt$Family))
unique_families <- unique(glom_melt$Family)
num_rare <- length(unique(glom_melt$Family[glom_melt$Abundance < 0.05]))
cat("Number of unique families:", num_unique_families, "\n")
cat("Unique family names:", unique_families, "\n")
glom_melt$Family <- factor(glom_melt$Family, levels = c("Candidatus Hepatincola", "Comamonadaceae", "Flavobacteriaceae", "Aeromonadaceae", 
                                                        "Chitinibacteraceae", "Pseudomonadaceae", "Methylophilaceae", "Sporichthyaceae", "Spirosomaceae",
                                                        "Chitinophagaceae", "Weeksellaceae", "T34", "Mycobacteriaceae", "Exiguobacteraceae", 
                                                        "Moraxellaceae", "Clade III", "Lactobacillaceae", "Oxalobacteraceae", "Bacillaceae",            
                                                        "Burkholderiaceae", "Methylacidiphilaceae", "Staphylococcaceae", "Verrucomicrobiaceae",
                                                        "Enterobacteriaceae","env.OPS 17", "Crocinitomicaceae", "Listeriaceae",
                                                        "Terrimicrobiaceae", "Cyanobiaceae", "Opitutaceae", "Enterococcaceae", "Rubritaleaceae",
                                                        "Saprospiraceae", "Rhodobacteraceae", "<5% abund."))
filtered_data <- glom_melt[glom_melt$Family == "<5% abund.", ]
ggplot(filtered_data, aes(x = time_point, y = Abundance)) +
  geom_bar(stat = "identity", fill = "gray") +
  labs(title = "Rare Taxa: <5% Abundance",
       x = "Samples", y = "Relative Abundance") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
glom_plot <- ggplot(data = glom_melt, aes(x = time_point, y=Abundance, fill = Family, color = Family)) +
  geom_bar(aes(fill = Family, color = Family), stat ="identity", position ="fill") +
  scale_fill_manual(values = c('turquoise','darkseagreen2', 'tan', 'orange', 'cornflowerblue', 
                'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 
                'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3',
                "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse', 
                'seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',
                "darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro", "gold2", "chocolate", "darkgreen", "dodgerblue", "burlywood3", 'hotpink')) +
  scale_color_manual(values = c('turquoise','darkseagreen2', 'tan', 'orange', 'cornflowerblue', 
                'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 
                'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3',
                "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse', 
                'seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',
                "darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro", "gold2", "chocolate", "darkgreen", "dodgerblue", "burlywood3", 'hotpink')) +
  xlab("Sampling Time Point") + theme(legend.position="bottom") + guides(fill=guide_legend(nrow=5))
glom_plot
dev.copy(pdf, "phyloseq analysis/figures/glom_plot.pdf", height=8, width=15)
dev.off()
```


```{r stacked bar plots}


ASV_rel_abund_hell <- transform_sample_counts(PS.hell, function(x) x/sum(x))
ASV_dataframe_hell <- psmelt(ASV_rel_abund_hell)
Abund.hell <- aggregate(Abundance ~ sequence_ID + time_point + lake + organism_water + phy_group + sample_type +
                     Phylum + Class + Order + Family +
                     Genus+Species, data=ASV_dataframe_hell, FUN=mean)

# remove rare
common_subset <- subset(ASV_dataframe_hell, ASV_dataframe_hell$Abundance > 0.01)
Abund.class <- aggregate(Abundance ~ Class, data=common_subset, FUN=mean)


# trying something
ASV_rel_abund_hell <- transform_sample_counts(PS.fin, function(x) x/sum(x))
pm <- psmelt(ASV_rel_abund_hell)
abund_plot <- pm %>%
  dplyr::group_by(sample_type) %>%
  mutate(tot = sum(Abundance)) %>%
  group_by(Class, sample_type) %>%
  summarise(rel_abund = sum(Abundance)/tot)

########### plotting relative abundance, help from Megan

ASV_rel_abund <- transform_sample_counts(PS.fin, function(x) x/sum(x))
pm <- psmelt(ASV_rel_abund)
Abund <- aggregate(Abundance ~ phy_group + lake + time_point + Class, data=pm, FUN=sum)
sam <- aggregate(Abundance ~ phy_group + lake + time_point , data=pm, FUN=sum) # this should include everything Abund has except for Class

colnames(sam)[ncol(sam)] <- "tot.abund" # renaming the last column 'tot.abund' so that we can differentiate that from rel.abund

Abund.final <- dplyr::full_join(Abund, sam)
Abund.final$rel_abund <- Abund.final$Abundance/Abund.final$tot.abund
View(Abund.final)

Abund.final$Class[Abund.final$rel_abund <= 0.05] <- "Other (<5%)" # classes that have rel.abund less than 5% will be named as other


# Help from Dr. Jackrel

my_class <- PS.fin %>%
  tax_glom(taxrank = "Class") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.05) %>%                         # Filter out low abundance taxa
  arrange(Class)  
view(my_class)

my_order <- PS.fin %>%
  tax_glom(taxrank = "Order") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.05) %>%                         # Filter out low abundance taxa
  arrange(Order)  
view(my_order)

# plot
overall.comp <- ggplot(my_order, aes(x=phy_group, y=Abundance, fill=Order, color=Order)) +
  geom_bar(aes(fill=Order, color=Order), stat="identity", position="fill") +
  ylab("Relative Abundance (%)") + theme(axis.text.x=element_text(size=12)) +
  theme(axis.text.y=element_text(size=10)) +
  theme(strip.text.y = element_text(size = 15)) + theme_bw() +
  scale_fill_manual(values = c('turquoise','pink2', 'tan', 'orange', 'cornflowerblue', 
                'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 
                'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3',
                "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse', 
                'seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',
                "darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro", "gold2", "chocolate", "darkgreen", "dodgerblue", "burlywood3", 'hotpink', 'yellow3', 'blue', 'pink', 'brown3')) +
  scale_color_manual(values = c('turquoise','pink2', 'tan', 'orange', 'cornflowerblue', 
                'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 
                'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3',
                "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse', 
                'seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',
                "darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro", "gold2", "chocolate", "darkgreen", "dodgerblue", "burlywood3", 'hotpink', 'yellow3', 'blue', 'pink', 'brown3'))
overall.comp
dev.copy(pdf, "phyloseq analysis/figures/overall.comp.pdf", height=6, width=10)
dev.off()


Facet <- ggplot(my_class, aes(x = time_point, y = Abundance, fill= Class, color = Class)) +
  geom_bar(width = 0.8, aes(fill = Class, color = Class), stat ="identity", position = "fill") +
  facet_grid(cols = vars(lake), rows = vars(phy_group)) + xlab("Sampling Time Point") +  ylab("Relative Abundance (%)") +
  theme_bw() + theme(axis.text.x=element_text(size=12)) +
  theme(axis.text.y=element_text(size=10)) +
  theme(strip.text.y = element_text(size = 15)) +
  scale_fill_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "seagreen4", "lightskyblue", "skyblue4", "navy", "tan3", "royalblue3", "black", "antiquewhite2", "chartreuse3", "chocolate", "orchid", "lightpink")) +
  scale_color_manual(values = c("red4", "orangered", "gold", "lightgreen", "turquoise3", "seagreen4", "lightskyblue", "skyblue4", "navy", "tan3", "royalblue3", "black", "antiquewhite2", "chartreuse3", "chocolate", "orchid", "lightpink"))
Facet
dev.copy(pdf, "phyloseq analysis/figures/Facet.pdf", height=6, width=10)
dev.off()

# calculate
sum.abund <- summaryBy(rel_abund ~ Class*phy_group, data = Abund.final, FUN = mean)



# keep these plots for the hell of it

phy_group_order <- ggplot(Abund500, aes(x = phy_group, y = Abundance, fill= Order, color = Order)) +
  geom_bar(aes(fill = Order, color = Order), stat ="identity", position ="fill") +
  xlab("Sample Type") + theme_bw() +
  scale_fill_manual(values = c('turquoise','pink2', 'tan', 'orange', 'cornflowerblue', 
                'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 
                'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3',
                "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse', 
                'seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',
                "darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro", "gold2", "chocolate", "darkgreen", "dodgerblue", "burlywood3", 'hotpink', 'yellow3', 'blue', 'pink', 'brown3')) +
  scale_color_manual(values = c('turquoise','pink2', 'tan', 'orange', 'cornflowerblue', 
                'magenta', 'darkolivegreen4', 'indianred1', 'tan4', 'darkblue', 
                'mediumorchid1','firebrick4',  'yellowgreen', 'lightsalmon', 'tan3',
                "tan1",'darkgray', 'wheat4', '#DDAD4B', 'chartreuse', 
                'seagreen1', 'moccasin', 'mediumvioletred', 'seagreen','cadetblue1',
                "darkolivegreen1" ,"tan2" ,   "tomato3" , "#7CE3D8","gainsboro", "gold2", "chocolate", "darkgreen", "dodgerblue", "burlywood3", 'hotpink', 'yellow3', 'blue', 'pink', 'brown3'))
phy_group_order            
dev.copy(pdf, "phyloseq analysis/figures/phy_group_family .pdf", height=8, width=7)
dev.off()


Facet500_genus <- ggplot(Abund.hell, aes(x = time_point, y = Abundance, fill= Genus)) +
  geom_bar(aes(fill = Genus), stat ="identity", position ="fill") +
  xlab("Sampling Time Point") + theme_bw()
Facet500_genus
dev.copy(pdf, "phyloseq analysis/figures/Facet500_order.pdf", height=10, width=5)
dev.off()



################# subsetting
# time point 1
stacked2 <- ggplot(data=Abund[Abund$time_point=="1",], aes(x = lake, y = Abundance), fill = Phylum) +
  geom_bar(aes(color = Phylum, fill = Phylum, ), stat ="identity", position ="fill") +
  theme(legend.position="bottom") + guides(fill=guide_legend(nrow=7)) + 
  facet_grid(~time_point, scales="free") + theme(axis.text.x = element_text(angle = 90))
stacked2
dev.copy(pdf, "phyloseq analysis/figures/stacked2.pdf", height=7, width=10)
dev.off() 

# by time_point for Blue
stacked3 <- ggplot(data=Abund[Abund$lake =="Blue",], aes(x = lake, y = Abundance), fill = Phylum) +
  geom_bar(aes(color = Phylum, fill = Phylum, ), stat ="identity", position ="fill") +
  facet_grid(~time_point, scales="free")
stacked3
dev.copy(pdf, "phyloseq analysis/figures/stacked3.pdf", height=7, width=10)
dev.off() 

Abund.fa <- Abund[(Abund$Family=="Burkholderiales"),]

```
