---
title: "zoops_mb_analysis"
output: html_document
date: "2023-02-13"
---
```{r loading packages}

library("tidyverse")
library("ggplot2")
library("phyloseq")
library("vegan")
library("cowplot")
library("gridExtra")
library("ggpubr")
library("dplyr")
library("doBy")
library(RColorBrewer)
library(glue)
library(ecodist)
library(geosphere)

# use pacman to load CRAN packages missing
pacman::p_load('knitr', 'tidyverse', 'knitr', 'magrittr', 'effects', 'devtools',
               'stringi', 'dplyr', "ggplot2", "gridExtra", "dada2", "phyloseq", "vegan",
               "cowplot", "decontam","BiocManager", "dada2", "microbiome", "BiocManager")


remotes::install_github("microbiome/microbiome")
library("microbiome")

devtools::install_github("microbiome/mia")
library("mia")

if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("DESeq2")
```


```{r loading in data and modifying metadata}

# load in CSV files for phyloseq object
metadata <- read.csv('phyloseq analysis/metadata_mp.csv')
ASV_table <- read.csv('phyloseq analysis/otu_table.csv')
tax_table <- read.csv('phyloseq analysis/tax_table.csv')  


# modify metadata
str(metadata)
table(metadata$organism)

# set row names
row.names(metadata) <- metadata$X

metadata$date <- as.Date(metadata$date)

# factors
metadata$sample_ID <- as.factor(metadata$sample_ID)
metadata$sequencing_ID <- as.factor(metadata$sequencing_ID)
metadata$sample_type <- as.factor(metadata$sample_type)
metadata$organism_water <- as.factor(metadata$organism_water)
metadata$time_point <- as.factor(metadata$time_point)
metadata$date <- as.factor(metadata$date)
metadata$lake <- as.factor(metadata$lake)

# numerics
metadata$number <- as.numeric(metadata$number)
metadata$latitude <- as.numeric(metadata$latitude)
metadata$longitude <- as.numeric(metadata$longitude)
metadata$elevation_m <- as.numeric(metadata$elevation_m)
metadata$temp_C <- as.numeric(metadata$temp_C)
metadata$chla_ug.L <- as.numeric(metadata$chla_ug.L)
metadata$pH <- as.numeric(metadata$pH)
metadata$DO_perc <- as.numeric(metadata$DO_perc)
metadata$spc <- as.numeric(metadata$spc)
metadata$cond <- as.numeric(metadata$cond)
metadata$DOC <- as.numeric(metadata$DOC)
metadata$TDN <- as.numeric(metadata$TDN)
metadata$TDP <- as.numeric(metadata$TDP)

#check
class(metadata$sample_ID)
class(metadata$elevation_m)
class(metadata$spc)
str(metadata)


# make new column for cladocerans, copepoda, water
metadata$phy_group <- ifelse(metadata$organism == "Calanoid" | metadata$organism == "Cyclopoid" | metadata$organism =="Large Calanoid", "copepoda", ifelse(metadata$organism == "Water", "water", "cladocera"))

#make new column for elevation category
metadata$elev.cat <- as.factor(ifelse(metadata$elevation_m < 2900, "Below", "Above"))



#check that the new column called phy_group worked
levels(as.factor(metadata$phy_group))
levels(as.factor(metadata$elev.cat))

metadata$phy_group <- as.factor(metadata$phy_group)


```


```{r making physeq object}

PS.fin<-
  read_phyloseq(
    otu.file = "phyloseq analysis/otu_table.csv",
    taxonomy.file = "phyloseq analysis/tax_table.csv",
    metadata.file = "phyloseq analysis/metadata_mp.csv",
    type = c("simple"),
    sep = ","
  ) 
str(PS.fin)

PS.fin <- subset_taxa(PS.fin, Family!= "Mitochondria" | 
                                is.na(Family) & Class!="Chloroplast" | is.na(Class))

# change sample names to match 
sample_names(PS.fin) <- metadata$sequence_ID
rownames(metadata) <- metadata$sequence_ID
sample_data(PS.fin) <- metadata


########### let's inspect
df.fin <- as.data.frame(sample_data(PS.fin))
df.fin$LibrarySize <- sample_sums(PS.fin) # this is the # of reads
df.fin$Index <- seq(nrow(df.fin))

########### now we have final phyloseq object with correct metadata 
saveRDS(PS.fin, file="phyloseq analysis/output/PS.fin.rds")
PS.fin <- readRDS("phyloseq analysis/output/PS.fin.rds")


# Make a data frame with a column for the read counts of each sample
sample_data(PS.fin)$read_depth <- sample_sums(PS.fin)
sample_sum_df <- data.frame(read.sum = sample_sums(PS.fin))

# make row names the sample names
colnames(sample_sum_df) <- "read.sum"
sample_sum_df$sequence_ID <- rownames(sample_sum_df)

# merge in the reads
run.metaD <- merge(metadata, sample_sum_df, by="sequence_ID", all.y=TRUE)
write.csv(run.metaD, "phyloseq analysis/output/run.metaD.final.csv")
```


```{r exploratory richness plots}

#richness by Lake
richness.plot.location <- plot_richness(PS.rare, x="lake", measures=c("Observed", "Shannon")) + 
  labs(y= "Alpha Diversity Measure", x = "Lake") + theme_bw() + geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90)) + stat_compare_means(method = "anova")
richness.plot.location
dev.copy(pdf, "phyloseq analysis/figures/richness.plot.location.pdf", height=4, width=10)
dev.off() 

#richness by organism (or sample type i.e, water)
richness.plot.organism <- plot_richness(PS.rare, x="organism_water", measures=c("Observed", "Shannon")) + 
  labs(y= "Alpha Diversity Measure", x = "Organism") + theme_bw() + geom_boxplot() + 
  theme(axis.text.x = element_text(angle = 90)) + stat_compare_means(method = "anova")
richness.plot.organism
dev.copy(pdf, "phyloseq analysis/figures/richness.plot.organism.pdf", height=4, width=12)
dev.off() 


#richness by phy_group (or sample type i.e, water)
richness.plot.phy <- plot_richness(PS.rare, x="phy_group", measures=c("Observed", "Shannon")) +
  labs(y= "Alpha Diversity Measure", x = "Sample Type") + theme_bw() + geom_boxplot() +
  theme(axis.text.x = element_text(angle = 90)) + stat_compare_means(method = "anova")
richness.plot.phy
dev.copy(pdf, "phyloseq analysis/figures/richness.plot.phy.pdf", height=4, width=12)
dev.off() 


#richness by Time.point
richness.plot.time.point <- plot_richness(PS.rare, x="time_point", measures=c("Observed", "Shannon")) + 
  labs(y= "Alpha Diversity Measure", x = "Time Point") + theme_bw() + geom_boxplot() +
  stat_compare_means(method = "anova")

richness.plot.time.point
dev.copy(pdf, "phyloseq analysis/figures/richness.plot.time.point.pdf", height=4, width=10)
dev.off() 


```


```{r total reads by species boxplots}

# Box plots showing spread of total reads by species

box1 <- ggplot(sample_data(PS.fin), aes(x=organism_water, y=read_depth)) + geom_boxplot() +
  theme(axis.text.x = element_text(angle = 90)) + labs(y= "Read Depth", x = "Organism")
box1
dev.copy(pdf, "phyloseq analysis/figures/reads.sample.pdf", height=4, width=7)
dev.off() 

box2 <- ggplot(sample_data(PS.fin), aes(x=organism_water, y=log(read_depth))) + geom_boxplot() +
  theme(axis.text.x = element_text(angle = 90)) + labs(y= "log(Read Depth)", x = "Organism")
box2
dev.copy(pdf, "phyloseq analysis/figures/reads.sample.pdf", height=4, width=7)
dev.off() 

```


```{r Read counts and rarefaction curves}

# Histogram of sample read counts
hist.depth <- ggplot(sample_sum_df, aes(x = read.sum)) + 
  geom_histogram(color = "black", fill = "gold2", binwidth = 2500) +
  ggtitle("Distribution of sample sequencing depth") + 
  xlab("Read counts") +
  theme(axis.title.y = element_blank()) + theme_classic() + geom_vline(xintercept=1000, lty=2)
hist.depth
dev.copy(pdf, "phyloseq analysis/figures/hist.depth.pdf", height=4, width=5)
dev.off() 


# zoom in a bit on the mode of the histogram
zoom.hist.depth<-ggplot(sample_sum_df, aes(x = read.sum)) + 
  geom_histogram(color = "black", fill = "gold2", binwidth = 1000) + coord_cartesian(xlim=c(0, 5000)) +
  ggtitle("Distribution of sample sequencing depth") + 
  xlab("Read counts") +
  theme(axis.title.y = element_blank()) + theme_classic() + geom_vline(xintercept=1000, lty=2)
zoom.hist.depth
dev.copy(pdf, "phyloseq analysis/figures/zoom.hist.depth.pdf", height=4, width=5)
dev.off() 

# a different view
read_depth_violin <- ggplot(sample_sum_df, aes(x=1, y=read.sum)) +
  geom_violin() + ggtitle("Distribution of sample sequencing depth") + scale_y_log10()
read_depth_violin
dev.copy(pdf, "phyloseq analysis/figures/read_depth_violin.pdf", height=4, width=5)
dev.off()

# Rarefaction curve
count_table_filt <- as.data.frame(otu_table(PS.fin))
rarecurve((count_table_filt), step=50, cex=0.5, xlim=c(0,100000), ylab = "ASVs", label=FALSE)
abline(v = 5000, lty = "dotted", col="red", lwd=2)

dev.copy(pdf, "phyloseq analysis/figures/rare.raw.pdf", height=4, width=5)
dev.off() 
```


```{r exploring what low read count samples do and what to rarefy to}

ord <- ordinate(PS.fin, method = 'PCoA', distance = 'bray')
plot_ordination(PS.fin, ord, color = "sample_type")
sample_data(PS.fin)$read_depth <- sample_sums(PS.fin)
plot_ordination(PS.fin, ord, shape = "sample_type") +
  geom_point(aes(color = read_depth > 500))


ord <- ordinate(PS.rare, method = 'PCoA', distance = 'bray')
plot_ordination(PS.rare, ord, shape = "sample_type") +
  geom_point(aes(color = read_depth))
sample_data(PS.rare)$read_depth <- sample_sums(PS.rare)

```

```{r rarefy}
# rarefying to 500 reads

set.seed(10000)
PS.rare <- rarefy_even_depth(PS.fin, rngseed=1e4, sample.size = 500)
sort(rowSums(otu_table(t(PS.rare))))

# filtering metadata by sample names with 500 reads or more 
metadata_rare <- as.data.frame(sample_data(PS.rare))
metadata_rare$sequence_ID <- rownames(PS.rare) 


```




```{r INCORRECT ASV Richness and Diversity Plots}
rich <- richness(ASV_table, index="observed")
div <- diversity(ASV_table, index="shannon")
div <- cbind(rich, div)
div$sequence_ID <- rownames(div)

div <- merge(div, metadata[,c(1,7,8,9,20)], by="sequence_ID", all=T)
sum_div <- summaryBy(shannon~phy_group*lake*time_point, data = div, FUN = c(mean, sd))

m1 <- lm(shannon~time_point*lake*phy_group, data=div)
anova(m1)

ggplot(sum_div, aes(x=time_point, y=shannon.mean, fill=lake, color=lake))+
  facet_grid(rows="phy_group")+
  geom_point()+
  geom_line(wt=2)


######### now using the rarefied phyloseq object

rich_rare <- richness(physeq.rare500, index="observed")
div_rare <- diversity(physeq.rare500, index="shannon")
div_rare <- cbind(rich_rare, div_rare)
div_rare$sequence_ID <- rownames(div_rare)

div_rare <- merge(div_rare, metadata_500[,c(1,7,8,9,20)], by="sequence_ID", all=T)
div_agg2 <- summaryBy(shannon~phy_group*lake*time_point, data = div, FUN = mean)

m1 <- lm(shannon~time_point*lake*phy_group, data=div_rare)
anova(m1)

ggplot(div, aes(x=time_point, y=shannon, fill=phy_group, color=phy_group))+
  facet_grid(rows="lake")+
  geom_point()+
  geom_line(wt=2)



```


```{r prevelance of ASVs by phylum}
# Make a prevalence column so we can see how common these ASVs are 
# Compute prevalence of each feature, store as data.frame

ASV_table[1,]
prevdf = apply(X = otu_table(PS.fin),
               MARGIN = ifelse(taxa_are_rows(PS.fin), yes = 1, no = 2), 
               function(x){sum(x > 0)})

# Add taxonomy and total read counts to this data.frame
prevdf = data.frame(Prevalence = prevdf,
                    TotalAbundance = taxa_sums(PS.fin),
                    tax_table(PS.fin))

plyr::ddply(prevdf, "Phylum", function(df1){cbind(mean(df1$Prevalence),sum(df1$Prevalence))})
```

```{r correlations}


cor.test(metadata$elevation_m, metadata$DOC)
ggscatter(metadata, x = "elevation_m", y = "DOC", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "elevation (m)", ylab = "DOC")


cor.test(metadata$elevation_m, metadata$temp_C)
ggscatter(metadata, x = "elevation_m", y = "temp_C", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "elevation (m)", ylab = "Temperature (C)")

cor.test(metadata$chla_ug.L, metadata$DOC)
ggscatter(metadata, x = "chla_ug.L", y = "DOC", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "chla", ylab = "DOC")
```


```{r T tests}
temps <- unique(metadata$temp_C)
t.test(x = temps)

```




```{r Bray curtis distance matrix}

raw.asv.table <- as.data.frame(otu_table(PS.fin))
grep('NA', metadata)

str(raw.asv.table)
str(ASV_table)

set.seed(10000)
avg_bray_500 <- avgdist(t(raw.asv.table), sample = 500, iterations = 1e3)
str(avg_bray_500) # should be type 'dist'

set.seed(10000)
bc_vegdist <- vegdist(t(otu_table(PS.rare)), method = "bray")


set.seed(10000)
bc <- phyloseq::distance(PS.rare, method = "bray")

```



```{r PERMANOVAS from distance matrix}


set.seed(10000)
nmds <- metaMDS(bc,
          distance = "bray",
          k = 3,
          maxit = 999, 
          trymax = 500,
          wascores = TRUE)

scores(nmds)
species.scores <- as.data.frame(scores(nmds, display="site"))

nmds2.merge <- merge(sample_data(PS.rare), species.scores, by = 'row.names', all = TRUE)
stressplot(nmds)


# running all sorts of PERMANOVAs
# adonis2 has been found to be less reliable with more variable data/ distances or for uneven variation
broad.model <- adonis2(bc ~ lake*time_point*phy_group, data = nmds2.merge, permutations = 1000)
broad.model

capture.output()
anova(betadisper(bc, nmds2.merge$time_point))
anova(betadisper(bc, nmds2.merge$phy_group))

factors.model <- adonis2(bc ~ lake + time_point + date + phy_group + organism_water + sample_type
                         , data = nmds2.merge, permutations = 1000)
factors.model

# model for the factors 
model.lake <- adonis2(bc ~ lake, data = nmds2.merge) # not significant :(
model.lake

model.time <- adonis2(bc ~ nmds2.merge$time_point) # significant 
model.time 

model.date <- adonis2(bc ~ nmds2.merge$date) # significant
model.date

model.phy <- adonis2(bc ~ nmds2.merge$phy_group) # not significant
model.phy

model.group <- adonis2(bc ~ nmds2.merge$organism_water) # not significant 
model.group

model.type <- adonis2(bc ~ nmds2.merge$sample_type, permutations = 1000) # not significant :(
model.type

# models for the numerics

model.DO <- adonis2(bc ~ nmds2.merge$DO_perc)  # not significant :(
model.DO

model.temp <-adonis2(bc ~ nmds2.merge$temp_C) # not significant :(
model.temp

model.DOC <- adonis2(bc ~ nmds2.merge$DOC) # significant
model.DOC

model.chla <- adonis2(bc ~ nmds2.merge$chla_ug.L) # not significant :(
model.chla

model.pH <- adonis2(bc ~ nmds2.merge$pH) # not significant :(
model.pH

model.spc <- adonis2(bc ~ nmds2.merge$spc) # not significant
model.spc

model.elev <- adonis2(bc ~ nmds2.merge$elevation_m) # not significant
model.elev

model.elevcat <- adonis2(bc ~ nmds2.merge$elev.cat) # not significant
model.elevcat



#interactions
model.phy.by.DOC <- adonis2(bc ~ nmds2.merge$DOC*nmds2.merge$time_point) # not significant
model.phy.by.DOC

model.phy.by.DOC <- adonis2(bc ~ nmds2.merge$DOC*nmds2.merge$phy_group) # not significant
model.phy.by.DOC

model.phy.by.time <- adonis2(bc ~ nmds2.merge$phy_group*nmds2.merge$time_point) # not significant
model.phy.by.time

model.phy.by.temp <- adonis2(bc ~ nmds2.merge$phy_group*nmds2.merge$temp_C)  #significant
model.phy.by.temp

model.phy.by.elev <- adonis2(bc ~ nmds2.merge$phy_group*nmds2.merge$elevation_m) # not significant :(
model.phy.by.elev

model.chla.by.elev <- adonis2(bc ~ nmds2.merge$chla_ug.L*nmds2.merge$elevation_m) # not significant
model.chla.by.elev

model.temp.by.elev <- adonis2(bc ~ nmds2.merge$temp_C*nmds2.merge$elevation_m*nmds2.merge$temp_C) # not significant
model.temp.by.elev

model.DOC.by.elev <- adonis2(bc ~ nmds2.merge$DOC*nmds2.merge$elevation_m) # not significant :(
model.DOC.by.elev

model.DOC.by.lake <- adonis2(bc ~ nmds2.merge$DOC*nmds2.merge$lake) # not significant :(
model.DOC.by.lake

model.DOC.by.phy <- adonis2(bc ~ nmds2.merge$phy_group*nmds2.merge$DOC) # not significant
```


```{r ordiR2step}
# capscale is partial distance-based redundancy analysis

# bray-curtis distance matrix


# define scope, fit global model
dbrda <- capscale(bc_vegdist ~ time_point + lake + phy_group + DOC + temp_C, data = nmds2.merge, distance = "euclidean")
set.seed(1999) 
upr <- dbrda(bc ~ . + I(temp_C)^2 + I(elevation_m)^2 + I(chla_ug.L)^2 + 
               I(DOC)^2 + I(longitude)^2 + I(latitude)^2 + I(DO_perc)^2 +
               I(spc)^2 + I(cond)^2 + I(pH)^2, data = nmds2.merge)
lwr <- dbrda(bc_vegdist ~ 1, data = global_df, na.action = 'na.omit')
mod_selec <- ordiR2step(lwr, scope = formula(upr))
anova(mod_selec)# selecting site as most important factor - good stuff honestly simple semi strong model here 
anova(mod_selec, by = "axis")
RsquareAdj(mod_selec)
screeplot(mod_selec)

# performing stepwise selection with ordiRstep
ordiR2step(dbrda, )

```

```{r AIC}

AIC()
```


```{r mantel test, geographic distance}
install.packages("fossil")
library(fossil)

# to calculate geographic distances between lakes

gps <- as.data.frame(cbind(sample_data(PS.fin)$latitude, sample_data(PS.fin)$longitude))
geodists_km <- earth.dist(gps[,1:2], dist = T)
str(geodists_km)
mat_geodists_km <- as.matrix(geodists_km)
str(mat_geodists_km)
rownames(mat_geodists_km) <- rownames(sample_data(PS.fin))
colnames(mat_geodists_km) <- t(rownames(sample_data(PS.fin)))


# to calculate bray curtis distance
study_bc_mat <- as.matrix(vegdist(t(raw.asv.table), "bray"))
str(study_bc_mat)
all_otu.dist <- phyloseq::distance(sample_data(PS.fin), "bray")

# mantel distance decay
# spatial effects on community dissimilarity were tested using distance between samples in meters and the bray curtis dissimilarity matrix
#To run a Mantel test, we will need to generate two distance matrices: one containing spatial 
#distances and one containing Bray-Curtis distances between samples at the given points.  In the 
#spatial distance matrix, entries for pairs of points that are close together are lower than for 
#pairs of points that are far apart.  In the BC matrix, entries for pairs of locations 
#with similar outcomes are lower than for pairs of points with dissimilar outcomes.  We do this 
#using the dist function.  The Mantel test function will require objects of this "distance" class.
mantel_test <- mantel(study_bc_mat ~ mat_geodists_km)

Study.df <- data.frame(Distance = mat_geodists_km[lower.tri(mat_geodists_km)],
           BrayCurtis = study_bc_mat[lower.tri(study_bc_mat)])
dim(mat_geodists_km)
dim(study_bc_mat)
rownames(mat_geodists_km)
rownames(study_bc_mat)
colnames(mat_geodists_km[lower.tri(mat_geodists_km)])

Study_dist_plot <- ggplot(Study.df, aes(x=Distance, y=BrayCurtis)) +
  geom_point(size=0.8,alpha=0.5) +
  geom_smooth(method = "glm", formula = y~x) +
  scale_y_continuous(name="Bray-Curtis Dissimilarity",breaks=seq(0,1,0.25),limits=c(0,1)) +
  scale_x_continuous(name="Distance (km)",breaks=seq(0,8,2),limits=c(0,8)) +
  theme(plot.title = element_text(size=10, hjust = 0.5),
        axis.text.x=element_text(colour="black",size=7), 
        axis.title.x=element_blank(),
        axis.text.y=element_text(colour="black",size=7),
        axis.title=element_text(colour="black",size=9),
        legend.title=element_text(colour="black",size=7,face="bold"),
        legend.text=element_text(colour="black",size=7),
        panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
        panel.background=element_blank())

mantel.mod<-lm(BrayCurtis~log(Distance+1), data=Study.df)
```




```{r MDS plots by time point}
T1 <- subset_samples(PS.rare, time_point=="1")
ORD.T1 <- ordinate(T1, method='MDS', distance='bray')

MDS.ord.T1 <- plot_ordination(
  physeq = T1,                                                   
  ordination = ORD.T1) +                                                
  geom_point(aes(color = lake, shape= phy_group), size = 3) +    
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  scale_color_brewer(palette = "Dark2") +
  scale_fill_brewer(palette = "Dark2") + ggtitle("Time1") + theme_classic()

MDS.ord.T1
dev.copy(pdf, "phyloseq analysis/figures/MDS.ord.T1.pdf", height=6, width=7)
dev.off() 





T2<- subset_samples(PS.rare, time_point=="2")
ORD.T2 <- ordinate(T2, method='MDS', distance='bray')

MDS.ord.T2<-plot_ordination(
  physeq = T2,                                                   
  ordination = ORD.T2) +                                                
  geom_point(aes(color = lake, shape= phy_group), size = 3) +    
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  scale_color_brewer(palette = "Dark2") +
  scale_fill_brewer(palette = "Dark2") + ggtitle("Time2") +
  theme_classic()     

MDS.ord.T2
dev.copy(pdf, "phyloseq analysis/figures/MDS.ord.T2.pdf", height=6, width=7)
dev.off() 






T3<- subset_samples(PS.rare, time_point=="3")
ORD.T3 <- ordinate(T3, method='MDS', distance='bray')

MDS.ord.T3<-plot_ordination(
  physeq = T3,                                                   
  ordination = ORD.T3) +                                                
  geom_point(aes(color = lake, shape= phy_group), size = 3) +    
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  scale_color_brewer(palette = "Dark2") +
  scale_fill_brewer(palette = "Dark2") + ggtitle("Time3") +
  theme_classic()     

MDS.ord.T3
dev.copy(pdf, "phyloseq analysis/figures/MDS.ord.T3.pdf", height=6, width=7)
dev.off() 





T4<- subset_samples(PS.rare, time_point=="4")
ORD.T4 <- ordinate(T4, method='MDS', distance='bray')

MDS.ord.T4<-plot_ordination(
  physeq = T4,                                                   
  ordination = ORD.T4) +                                                
  geom_point(aes(color = lake, shape= phy_group), size = 3) +    
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  scale_color_brewer(palette = "Dark2") +
  scale_fill_brewer(palette = "Dark2") + ggtitle("Time4") +
  theme_classic()     

MDS.ord.T4
dev.copy(pdf, "phyloseq analysis/figures/MDS.ord.T4.pdf", height=6, width=7)
dev.off() 





T5<- subset_samples(PS.rare, time_point=="5")
ORD.T5 <- ordinate(T5, method='MDS', distance='bray')

MDS.ord.T5<-plot_ordination(
  physeq = T5,                                                   
  ordination = ORD.T5) +                                                
  geom_point(aes(color = lake, shape= phy_group), size = 3) +    
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  scale_color_brewer(palette = "Dark2") +
  scale_fill_brewer(palette = "Dark2") + ggtitle("Time5") +
  theme_classic()     

MDS.ord.T5
dev.copy(pdf, "phyloseq analysis/figures/MDS.ord.T5.pdf", height=6, width=7)
dev.off() 
```



```{r NMDS time_point}
centroid <- nmds2.merge %>% 
  group_by(time_point) %>%
  summarize(NMDS1 = mean(NMDS1), NMDS2 = mean(NMDS2))

  
test_nmds <- ggplot(nmds2.merge, aes(x=NMDS1, y=NMDS2), color = time_point) +
  geom_point(aes(color = time_point), size = 3) + 
  geom_point(data = centroid, size = 5, shape = 21, color = "black", aes(fill = time_point), show.legend = F) +
  stat_ellipse(level=0.9, linetype = 2, aes(color=time_point), show.legend = F) +
  theme_bw()
test_nmds


```


```{r NMDS phy_group by time point}
centroid <- nmds2.merge %>% 
  group_by(time_point) %>%
  summarize(NMDS1 = mean(NMDS1), NMDS2 = mean(NMDS2))
str(nmds2.merge)

  
test_nmds1 <- ggplot(nmds2.merge[(nmds2.merge$time_point == "1"), ], aes(x=NMDS1, y=NMDS2, color = phy_group)) +
  geom_point(aes(color = phy_group), size = 3) + 
  geom_point(data = centroid, size = 5, shape = 21, color = "black", aes(fill = time_point), show.legend = F) +
  stat_ellipse(level=0.9, linetype = 2, aes(color=phy_group)) +
  theme_bw()
test_nmds1

test_nmds2 <- ggplot(nmds2.merge[(nmds2.merge$time_point == "2"), ], aes(x=NMDS1, y=NMDS2, color = phy_group)) +
  geom_point(aes(color = phy_group), size = 3) + 
   geom_point(data = centroid, size = 5, shape = 21, color = "black", aes(fill = time_point), show.legend = F) +
  stat_ellipse(level=0.9, linetype = 2, aes(color=phy_group)) +
  theme_bw()
test_nmds2

test_nmds3 <- ggplot(nmds2.merge[(nmds2.merge$time_point == "3"), ], aes(x=NMDS1, y=NMDS2, color = phy_group)) +
  geom_point(aes(color = phy_group), size = 3) + 
   geom_point(data = centroid, size = 5, shape = 21, color = "black", aes(fill = time_point), show.legend = F) +
  stat_ellipse(level=0.9, linetype = 2, aes(color=phy_group)) +
  theme_bw()
test_nmds3

test_nmds4 <- ggplot(nmds2.merge[(nmds2.merge$time_point == "4"), ], aes(x=NMDS1, y=NMDS2, color = phy_group)) +
  geom_point(aes(color = phy_group), size = 3) + 
   geom_point(data = centroid, size = 5, shape = 21, color = "black", aes(fill = time_point), show.legend = F) +
  stat_ellipse(level=0.9, linetype = 2, aes(color=phy_group)) +
  theme_bw()
test_nmds4

test_nmds5 <- ggplot(nmds2.merge[(nmds2.merge$time_point == "5"), ], aes(x=NMDS1, y=NMDS2, color = phy_group)) +
  geom_point(aes(color = phy_group), size = 3) + 
   geom_point(data = centroid, size = 5, shape = 21, color = "black", aes(fill = time_point), show.legend = F) +
  stat_ellipse(level=0.9, linetype = 2, aes(color=phy_group)) +
  theme_bw()
test_nmds5


```
```{r NMDS lake}
centroid <- nmds2.merge %>% 
  group_by(lake) %>%
  summarize(NMDS1 = mean(NMDS1), NMDS2 = mean(NMDS2))
  
test_nmds <- ggplot(nmds2.merge, aes(x=NMDS1, y=NMDS2, color = lake)) +
  geom_point(aes(color = lake), size = 3) + 
   geom_point(data = centroid, size = 5, shape = 21, color = "black", aes(fill = lake), show.legend = F) +
  stat_ellipse(level=0.9, linetype = 2, aes(color=lake)) +
  theme_bw()
test_nmds
```


```{r NMDS elevation}
centroid <- nmds2.merge %>% 
  group_by(elev.cat) %>%
  summarize(NMDS1 = mean(NMDS1), NMDS2 = mean(NMDS2))
str(nmds2.merge)

test_nmds1 <- ggplot(nmds2.merge, aes(x=NMDS1, y=NMDS2, color = elev.cat)) +
  geom_point(aes(shape = phy_group), size = 3) + 
  geom_point(data = centroid, size = 5, shape = 21, color = "black", aes(fill = elev.cat), show.legend = F) +
  stat_ellipse(level=0.9, linetype = 2, aes(color=elev.cat)) +
  theme_bw()
test_nmds1
```





```{r dbRDA}

Temp. <- nmds2.merge$temp_C
Elevation <- nmds2.merge$elevation_m
Chlorophyll <- nmds2.merge$chla_ug.L
DOC <- nmds2.merge$DOC


my_dbRDA <- dbrda(avg_bray_500 ~ Temp. + Elevation + DOC, dist="bray")
anova(my_dbRDA) # overall significance of the analysis
anova(my_dbRDA, by="axis", perm.max=500) #tests axes for significance
my_anova <- anova(my_dbRDA, by="terms", permu=200) # tests for significance of environ. variables
save.image(file = "my_anova.RData")
plot(my_dbRDA)



```


```{r PCoA}
all_pcoa <- ordinate(
  physeq = PS.rare, 
  method = "PCoA", 
  distance = "bray"
)

PCoA.ord.plot <- plot_ordination(
  physeq = PS.rare,                                                       
  ordination = all_pcoa, color = "time_point") +                                                
  geom_point(aes(color = phy_group), size = 3) +  
  stat_ellipse(type = "norm", linetype = 2, aes(color=time_point)) +
  scale_shape_manual(values = c(21, 22, 23, 24, 25)) +
  theme_classic() +                                                      
  theme(                             
    legend.text = element_text(size = 12),                               
    legend.title = element_blank(),                                      
    legend.background = element_rect(fill = "white", color = "NA"))+  
  theme(axis.text.y.left = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12))+
  guides(fill = guide_legend(override.aes = list(shape = 21))) 

PCoA.ord.plot
dev.copy(pdf, "phyloseq analysis/figures/PCoA.pdf", height=7, width=10)
dev.off() 

```
```{r stacked bar phylum - incorrect}

# make colors for sites
library(RColorBrewer)
qual_col_pals = brewer.pal.info[brewer.pal.info$category == 'qual',]
col_vector = unlist(mapply(brewer.pal, qual_col_pals$maxcolors, rownames(qual_col_pals)))

# merge all to phylum level

ps.rare_phylum <- physeq.rare500 %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa
  arrange(Phylum) 

phylum_colors<-col_vector

# Plot 
Bac.Phyla.stacked.bar<-
  ggplot(ps.rare_phylum, aes(x = phy_group, y = Abundance, fill = Phylum)) + 
  #facet_grid(site~.) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = phylum_colors) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Site")+
  ggtitle("Phylum Composition of Bacterial Communities by Site") +
  theme(axis.text.x=element_text(angle = 30))


Bac.Phyla.stacked.bar
ggsave(filename = "figures/Bac.Phyla.stacked.bar.pdf", plot = Bac.Phyla.stacked.bar, height=5.5, width=10)
dev.copy(pdf, "figures/Bac.Phyla.stacked.bar.pdf", height=5.5, width=10)
dev.off() 





ps.rare_phylum1 <- ps.rare %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(Phylum) 


Bac.Phyla.stacked.bar1<-
  ggplot(ps.rare_phylum1, aes(x = location, y = Abundance, fill = Phylum)) +
  #facet_grid(site~.) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = phylum_colors) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance") +
  xlab("Site")+
  ggtitle("Phylum Composition of Bacterial Communities by Site") +
  theme(axis.text.x=element_text(angle = 30))


Bac.Phyla.stacked.bar1
ggsave(filename = "figures/Bac.Phyla.stacked.bar1.pdf", plot = Bac.Phyla.stacked.bar1, height=5.5, width=10)
```




```{r stacked bar Phylum - correct?}

ASV_rel_abund <- transform_sample_counts(PS.rare,function(x) x/sum(x))
ASV_dataframe <- psmelt(ASV_rel_abund)
ASV_dataframe$Abundance<- ifelse(ASV_dataframe$Abundance < 1e-2, 0, ASV_dataframe$Abundance)
Abund <- aggregate(Abundance ~ time_point + lake + organism_water + phy_group + 
                     elevation_m + temp_C + Phylum + Class + Order + Family +
                     Genus+Species, data=ESV_dataframe, FUN=mean)

# testing to see if all the relative abundances of each ASV within sample 1 add up to 1
length(which(ASV_dataframe$Sample == "98_16S"))
sample98 <- ASV_dataframe$Sample == "98_16S"
test <- as.matrix(ASV_dataframe[sample98, "Abundance"])
sum(test)

Abund.fa <- Abund[(Abund$Family=="Burkholderiales"),]
p1 <- ggplot(Abund.fa, aes(x=time_point, y=Abundance, fill= Family)) +
  geom_bar(aes(fill = Family), stat ="identity", position ="fill")+
  facet_grid(cols=vars(phy_group), rows=vars(lake))


p1 <- ggplot(Abund, aes(x = time_point, y = Abundance, fill= Class)) +
  geom_bar(aes(fill = Class), stat ="identity", position ="fill") +
  facet_grid(cols=vars(phy_group), rows=vars(lake))



# by lake and time_point
stacked1 <- ggplot(data=Abund, aes(x = lake, y = Abundance), fill = Class) +
  geom_bar(aes(fill = Class), stat ="identity", position ="fill") +
  facet_grid(~time_point, scales="free") + theme(axis.text.x = element_text(angle = 90))

stacked1
dev.copy(pdf, "phyloseq analysis/figures/stacked1.pdf", height=7, width=10)
dev.off() 

###

# time point 1
stacked2 <- ggplot(data=Abund[Abund$time_point=="1",], aes(x = lake, y = Abundance), fill = Phylum) +
  geom_bar(aes(color = Phylum, fill = Phylum, ), stat ="identity", position ="fill") +
  theme(legend.position="bottom") + guides(fill=guide_legend(nrow=7)) + 
  facet_grid(~time_point, scales="free") + theme(axis.text.x = element_text(angle = 90))
stacked2
dev.copy(pdf, "phyloseq analysis/figures/stacked2.pdf", height=7, width=10)
dev.off() 

# by time_point for Blue
stacked3 <- ggplot(data=Abund[Abund$lake =="Blue",], aes(x = lake, y = Abundance), fill = Phylum) +
  geom_bar(aes(color = Phylum, fill = Phylum, ), stat ="identity", position ="fill") +
  facet_grid(~time_point, scales="free")
stacked3
dev.copy(pdf, "phyloseq analysis/figures/stacked3.pdf", height=7, width=10)
dev.off() 

# by time_point for Eastern Brook
stacked <- ggplot(data=Abund[Abund$lake =="Eastern Brook",], aes(x = lake, y = Abundance), fill = Phylum) +
  geom_bar(aes(color = Phylum, fill = Phylum, ), stat ="identity", position ="fill") +
  facet_grid(~time_point, scales="free")
stacked
dev.copy(pdf, "phyloseq analysis/figures/stacked.pdf", height=7, width=10)
dev.off() 

# time_point by phy_group
stacked4 <- ggplot(data=Abund[Abund$time_point=="1",], aes(x = phy_group, y = Abundance), fill = Class) +
  geom_bar(aes(color = Class, fill = Class, ), stat ="identity", position ="fill") +
  facet_grid(~time_point, scales="free") + theme(axis.text.x = element_text(angle = 90))

stacked4
dev.copy(pdf, "phyloseq analysis/figures/stacked4.pdf", height=7, width=10)
dev.off() 

stacked5 <- ggplot(data=Abund[Abund$time_point=="2",], aes(x = phy_group, y = Abundance), fill = Class) +
  geom_bar(aes(color = Class, fill = Class, ), stat ="identity", position ="fill") +
  facet_grid(~time_point, scales="free") + theme(axis.text.x = element_text(angle = 90))

stacked5
dev.copy(pdf, "phyloseq analysis/figures/stacked5.pdf", height=7, width=10)
dev.off() 

stacked6 <- ggplot(data=Abund[Abund$time_point=="3",], aes(x = phy_group, y = Abundance), fill = Class) +
  geom_bar(aes(color = Class, fill = Class, ), stat ="identity", position ="fill") +
  facet_grid(~time_point, scales="free") + theme(axis.text.x = element_text(angle = 90))

stacked6
dev.copy(pdf, "phyloseq analysis/figures/stacked6.pdf", height=7, width=10)
dev.off() 

stacked7 <- ggplot(data=Abund[Abund$time_point=="4",], aes(x = phy_group, y = Abundance), fill = Class) +
  geom_bar(aes(color = Class, fill = Class, ), stat ="identity", position ="fill") +
  facet_grid(~time_point, scales="free") + theme(axis.text.x = element_text(angle = 90))

stacked7
dev.copy(pdf, "phyloseq analysis/figures/stacked7.pdf", height=7, width=10)
dev.off() 


stacked8 <- ggplot(data=Abund[Abund$time_point=="5",], aes(x = phy_group, y = Abundance), fill = Class) +
  geom_bar(aes(color = Class, fill = Class, ), stat ="identity", position ="fill") +
  facet_grid(~time_point, scales="free") + theme(axis.text.x = element_text(angle = 90))

stacked8
dev.copy(pdf, "phyloseq analysis/figures/stacked8.pdf", height=7, width=10)
dev.off()


plot_composition(PS.fin)

```

```{r}
ASV_abund <- transform_sample_counts(PS.fin,function(x) x)
ASV_reads_DF <- psmelt(ASV_abund)

Reads<-aggregate(Abundance~time_point+lake+organism_water+phy_group+elevation_m+
                   temp_C+Phylum+Class+Order+Family+Genus+Species, data=ESV_reads_DF, FUN=sum)
#Reads<-aggregate(Abundance~time_point+lake+organism_water+phy_group+elevation_m+temp_C+Phylum, data=ESV_reads_DF, FUN=sum)

Reads$samp_tot<-summaryBy(Abundance~time_point + lake + organsim_water + phy_group, 
                          data=Reads, FUN="sum")

library(doBy)
x<-summaryBy(Abundance~lake+time_point, data=Abund, FUN="mean")
```



```{r Using physeq.rare to make community comp figure }



# aggregating ASVs by class 
class_counts_tab <- otu_table(tax_glom(PS.fin, taxrank="Class"))
class_tax_vec <- as.vector(tax_table(tax_glom(PS.fin, taxrank="Class"))[,"Class"]) 
rownames(class_counts_tab) <- as.vector(class_tax_vec)

# below is aggregating ASVs by phylum 
phyla_counts_tab <- otu_table(tax_glom(PS.fin, taxrank="Phylum")) 
# making rownames phyla 
phyla_tax_vec <- as.vector(tax_table(tax_glom(PS.fin, taxrank="Phylum"))[,"Phylum"]) 
rownames(phyla_counts_tab) <- as.vector(phyla_tax_vec)

# now we have a dataframe with the number of reads corresponding for each phylum and class in each sample 

# making a relative abundance df at the phylum level 
phylum_proportions <- apply(phyla_counts_tab, 2, function(x) x/sum(x))

phylum_proportions <- as.data.frame(phylum_proportions) # making df
phylum_proportions$Phyla <- row.names(phylum_proportions) # column of phyla
# check they all add up to 1
sum(phylum_proportions[,1])





# making data long - sample column, phyla column, relative abundance column 
phylum_prop_long <- phylum_proportions %>% 
  pivot_longer(cols = colnames(phylum_proportions[,1:320]), 
                                    names_to = "sequence_ID", 
                                    values_to = "rel_abund")
phylum_prop_long <- as.data.frame(phylum_prop_long)
str(phylum_prop_long)

# now we have long df with each sample and the rel abundance of phyla 
```


```{r merging with metadata - specifically site level and elev.cat level }
# manipulating metadata file: 

comp.metadata <- as.data.frame(metadata[, c(1,4,5,7,9,10,20)])

# merging metadata with phyla df 
phyla_comp <-  inner_join(phylum_prop_long, comp.metadata, by = "sequence_ID")
```


```{r plotting Phylum composition }
# basic figure of mean relative abundance for above and below treeline
phyla_comp %>% 
  group_by(time_point, Phyla) %>% 
  summarize(mean_rel_abund = 100*mean(rel_abund)) %>% 
  ggplot(aes(x=time_point, y=mean_rel_abund, fill=Phyla)) + 
  geom_col() + 
  theme_classic() + 
  scale_fill_discrete(name = NULL) + 
  labs(y = "Mean Relative Abundance (%)", x=NULL) + 
  theme(legend.text = element_text(face = "italic"), 
        legend.key.size = unit(10, "pt"))
ggsave("figures/mean_phyla_bar_raw.tiff", width = 5, height = 4)

# we have the basic figure - now we need to make it more appealing 

# making an "other" category and improving figure based on Pat Schloss suggestion

# seeing maximum mean relative abundance of each phyla in above/below groups 
phyla_mean_rel_abund <- phyla_comp %>% 
  group_by(time_point, Phyla) %>% 
  summarize(mean_rel_abund = 100*mean(rel_abund))

max.phyla <- phyla_mean_rel_abund %>% 
  group_by(Phyla) %>% 
  summarize(max(mean_rel_abund)) 
max.phyla <- max.phyla[order(max.phyla$'max(mean_rel_abund)'), decreasing = FALSE]

# we can see our most abundant phyla: Firmicutes, Fusobacteriota, Proteobacteria, Cyanobacteria, Actinobacteriota (maybe)

# pooling everything over 5% using a logical 
phyla_pool <- phyla_mean_rel_abund %>% 
  group_by(Phyla) %>% 
  summarize(pool = max(mean_rel_abund) < 5, 
            mean = mean(mean_rel_abund)) # this creates a logical of if a phyla meets the condition (is less than 5%)

# join with df and plot! 
inner_join(phyla_mean_rel_abund, phyla_pool, by = "Phyla") %>% 
  mutate(Phyla = if_else(pool, "Other", Phyla)) %>%
  group_by(time_point, Phyla) %>% 
  summarize(mean_rel_abund = sum(mean_rel_abund),
            mean = sum(mean)) %>% 
  mutate(Phyla = factor(Phyla), 
         Phyla = fct_reorder(Phyla, mean, .desc = TRUE), 
         Phyla = fct_shift(Phyla, n=1)) %>%
  ggplot(aes(x=time_point, y=mean_rel_abund, fill=Phyla)) + 
  geom_col() + 
  theme_classic() + 
  scale_fill_manual(name = NULL, 
                    breaks = c("Cyanobacteria", "Firmicutes", "Fusobacteriota", 
                               "Proteobacteria", "Other"), 
                    values = c(brewer.pal(4, "Set1"), "gray")) +
  labs(y = "Mean Relative Abundance (%)", x=NULL) + 
  theme(legend.text = element_text(face = "italic"), 
        legend.key.size = unit(12, "pt"))
ggsave("figures/phyla_comp_condensed.tiff", height = 4, width = 5)
```

